\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
\usepackage{graphicx}\setkeys{Gin}{width=0.7\textwidth}
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge \R{} documentation}} \par\bigskip{{\Large of all in \file{man}}}
\par\bigskip{\large \today}
\end{center}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{httk-package}{High-Throughput Toxicokinetics}{httk.Rdash.package}
\aliasA{httk}{httk-package}{httk}
\keyword{package}{httk-package}
%
\begin{Description}\relax
Generic models and chemical-specific data for simulation and
             statistical analysis of chemical toxicokinetics ("TK") as
             described by Pearce et al. (2017) <doi:10.18637/jss.v079.i04>.
             Chemical-specific in vitro data have been obtained from relatively
             high throughput experiments. Both physiologically-based ("PBTK")
             and empirical (for example, one compartment) "TK" models can be
             parameterized with the data provided for thousands of chemicals,
             multiple exposure routes, and various species. The models consist
             of systems of ordinary differential equations which are solved
             using compiled (C-based) code for speed. A Monte Carlo sampler is
             included, which allows for simulating human biological variability
             (Ring et al., 2017 <doi:10.1016/j.envint.2017.06.004>)
             and propagating parameter uncertainty. Calibrated methods are
             included for predicting tissue:plasma partition coefficients and
             volume of distribution
             (Pearce et al., 2017 <doi:10.1007/s10928-017-9548-7>).
             These functions and data provide a set of tools for
             in vitro-in vivo extrapolation ("IVIVE") of high throughput
             screening data (for example, Tox21, ToxCast) to real-world
             exposures via reverse dosimetry (also known as "RTK")
             (Wetmore et al., 2015 <doi:10.1093/toxsci/kfv171>).
\end{Description}
%
\begin{Author}\relax
John Wambaugh, Robert Pearce, Caroline Ring, Gregory Honda, Nisha
Sipes, Jimena Davis, Barbara Wetmore, Woodrow Setzer, Mark Sfeir
\end{Author}
%
\begin{SeeAlso}\relax
\Rhref{https://www.epa.gov/chemical-research/computational-toxicology-communities-practice-high-throughput-toxicokinetic-httk-r}{PowerPoint
Presentation: High-Throughput Toxicokinetics (HTTK) R package}

doi:\nobreakspace{}\Rhref{https://doi.org/10.18637/jss.v079.i04}{10.18637\slash{}jss.v079.i04}Pearce et al. (2017): httk: R
Package for High-Throughput Toxicokinetics

doi:\nobreakspace{}\Rhref{https://doi.org/10.1093/toxsci/kfv171}{10.1093\slash{}toxsci\slash{}kfv171}Wetmore et al. (2015):
Incorporating High-Throughput Exposure Predictions With Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing

doi:\nobreakspace{}\Rhref{https://doi.org/10.1093/toxsci/kfv118}{10.1093\slash{}toxsci\slash{}kfv118}Wambaugh et al. (2015):
Toxicokinetic Triage for Environmental Chemicals

doi:\nobreakspace{}\Rhref{https://doi.org/10.1007/s10928-017-9548-7}{10.1007\slash{}s10928\-017\-9548\-7}Pearce et al. (2017):
Evaluation and calibration of high-throughput predictions of chemical
distribution to tissues

doi:\nobreakspace{}\Rhref{https://doi.org/10.1016/j.envint.2017.06.004}{10.1016\slash{}j.envint.2017.06.004}Ring et al. (2017):
Identifying populations sensitive to environmental chemicals by simulating
toxicokinetic variability

doi:\nobreakspace{}\Rhref{https://doi.org/10.1021/acs.est.7b00650}{10.1021\slash{}acs.est.7b00650}Sipes et al. (2017): An
Intuitive Approach for Predicting Potential Human Health Risk with the Tox21
10k Library

doi:\nobreakspace{}\Rhref{https://doi.org/10.1093/toxsci/kfy020}{10.1093\slash{}toxsci\slash{}kfy020}Wambaugh et al. (2018):
Evaluating In Vitro-In Vivo Extrapolation of Toxicokinetics

doi:\nobreakspace{}\Rhref{https://doi.org/10.1371/journal.pone.0217564}{10.1371\slash{}journal.pone.0217564}Honda et al. (2019):
Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptionss

doi:\nobreakspace{}\Rhref{https://doi.org/10.1093/toxsci/kfz205}{10.1093\slash{}toxsci\slash{}kfz205}Wambaugh et al. (2019):
Assessing Toxicokinetic Uncertainty and Variability in Risk Prioritization

doi:\nobreakspace{}\Rhref{https://doi.org/10.1038/s41370-020-0238-y}{10.1038\slash{}s41370\-020\-0238\-y}Linakis et al. (2020):
Development and evaluation of a high throughput inhalation model for organic chemicals

\Rhref{https://www.epa.gov/chemical-research/rapid-chemical-exposure-and-dose-research}{EPA's
ExpoCast (Exposure Forecasting) Project}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{add\_chemtable}{Add a table of chemical information for use in making httk predictions.}{add.Rul.chemtable}
%
\begin{Description}\relax
This function adds chemical-specific information to the table
chem.physical\_and\_invitro.data. This table is queried by the model
parameterization functions when attempting to parameterize a model, so
adding sufficient data to this table allows additional chemicals to be
modeled.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
add_chemtable(
  new.table,
  data.list,
  current.table = NULL,
  reference = NULL,
  species = NULL,
  overwrite = FALSE,
  sig.fig = 4,
  clint.pvalue.overwrite = TRUE,
  allow.na = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{new.table}] Object of class data.frame containing one row per chemical,
with each chemical minimally described by a CAS number.

\item[\code{data.list}] This list identifies which properties are to be read from
the table. Each item in the list should point to a column in the table
new.table. Valid names in the list are: 'Compound', 'CAS', 'DSSTox.GSID'
'SMILES.desalt', 'Reference', 'Species', 'MW', 'logP', 'pKa\_Donor',
'pKa\_Accept', 'logMA', 'Clint', 'Clint.pValue', 'Funbound.plasma',
'Fgutabs', 'Rblood2plasma'.

\item[\code{current.table}] This is the table to which data are being added.

\item[\code{reference}] This is the reference for the data in the new table. This
may be omitted if a column in data.list gives the reference value for each
chemical.

\item[\code{species}] This is the species for the data in the new table. This may
be omitted if a column in data.list gives the species value for each
chemical or if the data are not species-specific (e.g., MW).

\item[\code{overwrite}] If overwrite=TRUE then data in current.table will be
replaced by any data in new.table that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.

\item[\code{sig.fig}] Sets the number of significant figures stored (defaults to 4)

\item[\code{clint.pvalue.overwrite}] If TRUE then the Cl\_int p-value is set to NA 
when the Cl\_int value is changed unless a new p-value is provided. (defaults
to TRUE)

\item[\code{allow.na}] If TRUE (default is FALSE) then NA values are written to the
table, otherwise they are ignored.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{data.frame}] A new data.frame containing the data in
current.table augmented by new.table
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}


my.new.data <- as.data.frame(c("A","B","C"),stringsAsFactors=FALSE)
my.new.data <- cbind(my.new.data,as.data.frame(c("111-11-2","222-22-0","333-33-5"),
                     stringsAsFactors=FALSE))
my.new.data <- cbind(my.new.data,as.data.frame(c("DTX1","DTX2","DTX3"),
                    stringsAsFactors=FALSE))
my.new.data <- cbind(my.new.data,as.data.frame(c(200,200,200)))
my.new.data <- cbind(my.new.data,as.data.frame(c(2,3,4)))
my.new.data <- cbind(my.new.data,as.data.frame(c(0.01,0.02,0.3)))
my.new.data <- cbind(my.new.data,as.data.frame(c(0,10,100)))
colnames(my.new.data) <- c("Name","CASRN","DTXSID","MW","LogP","Fup","CLint")

chem.physical_and_invitro.data <- add_chemtable(my.new.data,
                                  current.table=chem.physical_and_invitro.data,
                                  data.list=list(
                                  Compound="Name",
                                  CAS="CASRN",
                                  DTXSID="DTXSID",
                                  MW="MW",
                                  logP="LogP",
                                  Funbound.plasma="Fup",
                                  Clint="CLint"),
                                  species="Human",
                                  reference="MyPaper 2015")
parameterize_steadystate(chem.name="C")  
calc_css(chem.name="B")                                


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{age\_dist\_smooth}{Smoothed age distributions by race and gender.}{age.Rul.dist.Rul.smooth}
\keyword{data}{age\_dist\_smooth}
\keyword{httk-pop}{age\_dist\_smooth}
%
\begin{Description}\relax
Distributions of ages in months, computed from NHANES data smoothed using
survey::svysmooth(), for each combination of race/ethnicity and gender.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
age_dist_smooth
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table object with three variables: \begin{description}

\item[\code{gender}] Gender: Male or Female
\item[\code{reth}] Race/ethnicity\item[\code{smth}] A list of
\code{svysmooth} objects, each encoding a weighted smoothed distribution of
ages.
\end{description}

\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{age\_draw\_smooth}{Draws ages from a smoothed distribution for a given gender/race combination}{age.Rul.draw.Rul.smooth}
\keyword{httk-pop}{age\_draw\_smooth}
%
\begin{Description}\relax
Draws ages from a smoothed distribution for a given gender/race combination
\end{Description}
%
\begin{Usage}
\begin{verbatim}
age_draw_smooth(g, r, nsamp, agelim_months)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{g}] Gender. Either 'Male' or 'Female'.

\item[\code{r}] Race/ethnicity. One of 'Mexican American', 'Other Hispanic',
'Non-Hispanic Black', 'Non-Hispanic White', 'Other'.

\item[\code{nsamp}] Number of ages to draw.

\item[\code{agelim\_months}] Two-element numeric vector giving the minimum and
maximum ages in months to include.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A named list with members 'ages\_months' and 'ages\_years', each
numeric of length \code{nsamp}, giving the sampled ages in months and years.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{armitage\_estimate\_sarea}{Estimate well surface area}{armitage.Rul.estimate.Rul.sarea}
%
\begin{Description}\relax
Estimate geometry surface area of plastic in well plate based on well plate
format suggested values from Corning.  option.plastic == TRUE (default) give
nonzero surface area (sarea, m\textasciicircum{}2) option.bottom == TRUE (default) includes
surface area of the bottom of the well in determining sarea.  Optionally
include user values for working volume (v\_working, m\textasciicircum{}3) and surface area.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
armitage_estimate_sarea(
  tcdata = NA,
  this.well_number = 384,
  this.cell_yield = NA,
  this.v_working = NA
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{tcdata}] A data table with well\_number corresponding to plate format,
optionally include v\_working, sarea, option.bottom, and option.plastic

\item[\code{this.well\_number}] For single value, plate format default is 384, used
if is.na(tcdata)==TRUE

\item[\code{this.cell\_yield}] For single value, optionally supply cell\_yield,
otherwise estimated based on well number

\item[\code{this.v\_working}] For single value, optionally supply working volume,
otherwise estimated based on well number (m\textasciicircum{}3)
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data table composed of any input data.table \emph{tcdata}
with only the following columns either created or altered by this function:  

\Tabular{ccc}{
\strong{Column Name} & \strong{Description} & \strong{Units} \\{}
well\_number & number of wells on plate & \\{}
sarea & surface area & m\textasciicircum{}2 \\{}
cell\_yield & number of cells & cells \\{} 
v\_working & working (filled) volume of each well & uL \\{}
v\_total & total volume of each well & uL \\{}
}
\end{Value}
%
\begin{Author}\relax
Greg Honda
\end{Author}
%
\begin{References}\relax
Armitage, J. M., Arnot, J. A., Wania, F., \& Mackay, D. (2013). Development 
and evaluation of a mechanistic bioconcentration model for ionogenic organic 
chemicals in fish. Environmental toxicology and chemistry, 32(1), 115-128.
\end{References}
\inputencoding{utf8}
\HeaderA{armitage\_eval}{Evaluate the updated Armitage model}{armitage.Rul.eval}
%
\begin{Description}\relax
Evaluate the Armitage model for chemical distributon in vitro. Takes input
as data table or vectors of values. Outputs a data table. Updates over
the model published in Armitage et al. 2014 include binding to plastic walls
and lipid and protein compartments in cells.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
armitage_eval(
  casrn.vector = NA_character_,
  nomconc.vector = 1,
  this.well_number = 384,
  this.FBSf = NA_real_,
  tcdata = NA,
  this.sarea = NA_real_,
  this.v_total = NA_real_,
  this.v_working = NA_real_,
  this.cell_yield = NA_real_,
  this.Tsys = 37,
  this.Tref = 298.15,
  this.option.kbsa2 = F,
  this.option.swat2 = F,
  this.pseudooct = 0.01,
  this.memblip = 0.04,
  this.nlom = 0.2,
  this.P_nlom = 0.035,
  this.P_dom = 0.05,
  this.P_cells = 1,
  this.csalt = 0.15,
  this.celldensity = 1,
  this.cellmass = 3,
  this.f_oc = 1
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{casrn.vector}] For vector or single value, CAS number

\item[\code{nomconc.vector}] For vector or single value, micromolar nominal 
concentration (e.g. AC50 value)

\item[\code{this.well\_number}] For single value, plate format default is 384, used
if is.na(tcdata)==TRUE

\item[\code{this.FBSf}] Fraction fetal bovine serum, must be entered by user.

\item[\code{tcdata}] A data.table with casrn, nomconc, MP, gkow, gkaw, gswat, sarea,
v\_total, v\_working. Otherwise supply single values to this.params.

\item[\code{this.sarea}] Surface area per well (m\textasciicircum{}2)

\item[\code{this.v\_total}] Total volume per well (m\textasciicircum{}3)

\item[\code{this.v\_working}] Working volume per well (m\textasciicircum{}3)

\item[\code{this.cell\_yield}] Number of cells per well

\item[\code{this.Tsys}] System temperature (degrees C)

\item[\code{this.Tref}] Reference temperature (degrees K)

\item[\code{this.option.kbsa2}] Use alternative bovine-serum-albumin partitioning
model

\item[\code{this.option.swat2}] Use alternative water solubility correction

\item[\code{this.pseudooct}] Pseudo-octanol cell storage lipid content

\item[\code{this.memblip}] Membrane lipid content of cells

\item[\code{this.nlom}] Structural protein conent of cells

\item[\code{this.P\_nlom}] Proportionality constant to octanol structural protein

\item[\code{this.P\_dom}] Proportionality constant to dissolve organic material

\item[\code{this.P\_cells}] Proportionality constant to octanol storage lipid

\item[\code{this.csalt}] Ionic strength of buffer, mol/L

\item[\code{this.celldensity}] Cell density kg/L, g/mL

\item[\code{this.cellmass}] Mass per cell, ng/cell

\item[\code{this.f\_oc}] 1, everything assumed to be like proteins
\end{ldescription}
\end{Arguments}
%
\begin{Value}

\Tabular{lll}{
\strong{Column} & \strong{Description} & \strong{units} \\{}
casrn & Chemical Abstracts Service Registry Number & \\{}
nomconc & Nominal Concentration & mol/L \\{}       
well\_number & Number of wells in plate & unitless \\{}   
sarea & Surface area of well & m\textasciicircum{}2 \\{}         
v\_total & Total volume of well & m\textasciicircum{}3 \\{}       
v\_working & Filled volume of well & m\textasciicircum{}3 \\{}     
cell\_yield & Number of cells & cells \\{}    
gkow & log10 octanol to water partition coefficient (PC)& log10 \\{}          
logHenry & log10 Henry's law constant '& log10 atm-m3/mol \\{}      
gswat & log10 Water solubility & log10 mol/L \\{}         
MP & Melting Point & degrees Celsius \\{}           
MW & Molecular Weight & g/mol \\{}            
gkaw & air to water PC & (mol/m3)/(mol/m3) \\{}
dsm & & \\{}           
duow & & \\{}          
duaw & & \\{}          
dumw & & \\{}          
gkmw & & \\{}          
gkcw & & \\{}          
gkbsa & & \\{}         
gkpl & & \\{}          
ksalt & & \\{}        
Tsys & & \\{}          
Tref & & \\{}          
option.kbsa2 & & \\{}  
option.swat2 & & \\{}  
FBSf & & \\{}          
pseudooct & & \\{}     
memblip & & \\{}       
nlom & & \\{}          
P\_nlom & & \\{}   
P\_dom & dissolved organic matter to water PC & Dimensionless \\{}         
P\_cells & & \\{}      
csalt & & \\{}         
celldensity & & \\{}   
cellmass & & \\{}      
f\_oc & & \\{}          
cellwat & & \\{}       
Tcor & & \\{}          
Vm & Volume of media & L \\{}            
Vwell & volume of medium (aqueous phase only) & L \\{}         
Vair & volume of head space & L \\{}          
Vcells & volume of cells/tissue& \\{}        
Valb & volume of serum albumin & \\{}         
Vslip & volume of serum lipids & \\{}         
Vdom & volume of dissolved organic matter& \\{}          
F\_ratio & & \\{}       
gs1.GSE & & \\{}       
s1.GSE & & \\{}        
gss.GSE & & \\{}       
ss.GSE & & \\{}        
kmw & & \\{}           
kow & octanol to water PC & \\{}           
kaw & the air towater PC & dimensionless \\{}           
swat & & \\{}         
kpl & & \\{}           
kcw & cell/tissue to water PC & dimensionless \\{}           
kbsa & & \\{}          
swat\_L & & \\{}        
oct\_L & & \\{}        
scell\_L & & \\{}       
cinit & Initial concentration & mol \\{}         
mtot & Total moles & mol \\{}          
cwat & Total concentration in water & mol/L \\{}          
cwat\_s & Dissolved concentration in water & mol/L \\{}        
csat & Is the solution saturated (1/0) & Boolean \\{}         
activity & & \\{}      
cair & & mol/L \\{}          
calb & & mol/L \\{}          
cslip & & mol/L \\{}         
cdom & concentration of/in dissolved organic matter& mol/L \\{}          
ccells & & mol/L \\{}        
cplastic & & mol/L \\{}      
mwat\_s & Mass dissolved in water & mols \\{}        
mair & Mass in air & mols \\{}          
mbsa & Mass bound to bovine serum albumin & mols \\{}          
mslip & Mass bound to serum lipids & mols \\{}        
mdom & Mass bound to dissolved organic matter & mols \\{}          
mcells & Mass in cells & mols \\{}        
mplastic & Mass bond to plastic & mols \\{}      
mprecip & Mass precipitated out of solution & \\{}       
xwat\_s & Fraction dissolved in water & fraction \\{}        
xair & Fraction in the air & fraction \\{}          
xbsa & Fraction bound to bovine serum albumin & fraction \\{}          
xslip & Fraction bound to serum lipids & fraction \\{}         
xdom & Fraction bound to dissolved organic matter & fraction \\{}          
xcells & Fraction within cells & fraction \\{}        
xplastic & Fraction bound to plastic & fraction \\{}     
xprecip & Fraction precipitated out of solution & fraction \\{}       
eta\_free & effective availability ratio & fraction \\{}      
\strong{cfree.invitro} & \strong{Free concentration in the in vitro media} (use for Honda1 and Honda2) & micromolar \\{}
}
\end{Value}
%
\begin{Author}\relax
Greg Honda
\end{Author}
%
\begin{References}\relax
Armitage, J. M.; Wania, F.; Arnot, J. A. Environ. Sci. Technol. 
2014, 48, 9770-9779. https://doi.org/10.1021/es501955g

Honda et al. PloS one 14.5 (2019): e0217564. https://doi.org/10.1371/journal.pone.0217564
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

library(httk)

# Check to see if we have info on the chemical:
"80-05-7" %in% get_cheminfo()

#We do:
temp <- armitage_eval(casrn.vector = c("80-05-7", "81-81-2"), this.FBSf = 0.1,
this.well_number = 384, nomconc = 10)
print(temp$cfree.invitro)

# Check to see if we have info on the chemical:
"793-24-8" %in% get_cheminfo()

# Since we don't look up phys-chem from dashboard:
cheminfo <- data.frame(
  Compound="6-PPD",
  CASRN="793-24-8",
  DTXSID="DTXSID9025114",
  logP=4.27, 
  logHenry=log10(7.69e-8),
  logWSol=log10(1.58e-4),
  MP=	99.4,
  MW=268.404
  )
  
# Add the information to HTTK's database:
chem.physical_and_invitro.data <- add_chemtable(
 cheminfo,
 current.table=chem.physical_and_invitro.data,
 data.list=list(
 Compound="Compound",
 CAS="CASRN",
  DTXSID="DTXSID",
  MW="MW",
  logP="logP",
  logHenry="logHenry",
  logWSol="logWSol",
  MP="MP"),
  species="Human",
  reference="CompTox Dashboard 31921")

# Run the Armitage et al. (2014) model:
out <- armitage_eval(
  casrn.vector = "793-24-8", 
  this.FBSf = 0.1,
  this.well_number = 384, 
  nomconc = 10)
  
print(out)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{armitage\_input}{Armitage et al. (2014) Model Inputs from Honda et al. (2019)}{armitage.Rul.input}
\keyword{data}{armitage\_input}
%
\begin{Description}\relax
Armitage et al. (2014) Model Inputs from Honda et al. (2019)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
armitage_input
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data frame with 53940 rows and 10 variables:
\begin{description}

\item[MP] 
\item[MW] 
\item[casrn] 
\item[compound\_name] 
\item[gkaw] 
\item[gkow] 
\item[gswat] 

\end{description}

\end{Format}
%
\begin{Author}\relax
Greg Honda
\end{Author}
%
\begin{Source}\relax
\url{https://www.diamondse.info/}
\end{Source}
%
\begin{References}\relax
Armitage, J. M.; Wania, F.; Arnot, J. A. Environ. Sci. Technol.
2014, 48, 9770-9779. dx.doi.org/10.1021/es501955g

Honda, Gregory S., et al. "Using the Concordance of In Vitro and
In Vivo Data to Evaluate Extrapolation Assumptions", PloS ONE 14.5 (2019): e0217564.
\end{References}
\inputencoding{utf8}
\HeaderA{augment.table}{Add a parameter value to the chem.physical\_and\_invitro.data table}{augment.table}
%
\begin{Description}\relax
This internal function is used by \code{\LinkA{add\_chemtable}{add.Rul.chemtable}} to add a single 
new parameter to the table of chemical parameters. It should not be typically
used from the command line.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
augment.table(
  this.table,
  this.CAS,
  compound.name = NULL,
  this.property,
  value,
  species = NULL,
  reference,
  overwrite = FALSE,
  sig.fig = 4,
  clint.pvalue.overwrite = TRUE,
  allow.na = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{this.table}] Object of class data.frame containing one row per chemical.

\item[\code{this.CAS}] The Chemical Abstracts Service registry number (CAS-RN)
correponding to the parameter value

\item[\code{compound.name}] A name associated with the chemical (defaults to NULL)

\item[\code{this.property}] The property being added/modified.

\item[\code{value}] The value being assigned to this.property.

\item[\code{species}] This is the species for the data in the new table. This may
be omitted if a column in data.list gives the species value for each
chemical or if the data are not species-specific (e.g., MW).

\item[\code{reference}] This is the reference for the data in the new table. This
may be omitted if a column in data.list gives the reference value for each
chemical.

\item[\code{overwrite}] If overwrite=TRUE then data in current.table will be
replaced by any data in new.table that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.

\item[\code{sig.fig}] Sets the number of significant figures stored (defaults to 4)

\item[\code{clint.pvalue.overwrite}] If TRUE then the Cl\_int p-value is set to NA 
when the Cl\_int value is changed unless a new p-value is provided. (defaults
to TRUE)

\item[\code{allow.na}] If TRUE (default is FALSE) then NA values are written to the
table, otherwise they are ignored.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{data.frame}] A new data.frame containing the data in
current.table augmented by new.table
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{available\_rblood2plasma}{Find the best available ratio of the blood to plasma concentration constant.}{available.Rul.rblood2plasma}
\keyword{Parameter}{available\_rblood2plasma}
%
\begin{Description}\relax
This function finds the best available constant ratio of the blood
concentration to the plasma concentration, using get\_rblood2plasma and
calc\_rblood2plasma.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
available_rblood2plasma(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  adjusted.Funbound.plasma = TRUE,
  suppress.messages = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the CAS number or the chemical name must be
specified.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{adjusted.Funbound.plasma}] Whether or not to use Funbound.plasma
adjustment if calculating Rblood2plasma.

\item[\code{suppress.messages}] Whether or not to display relevant warning messages
to user.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Either retrieves a measured blood:plasma concentration ratio from the
chem.physical\_and\_invitro.data table or calculates it using the red blood cell
partition coefficient predicted with Schmitt's method

If available, in vivo data (from chem.physical\_and\_invitro.data) for the
given species is returned, substituting the human in vivo value when missing
for other species.  In the absence of in vivo data, the value is calculated
with calc\_rblood2plasma for the given species. If Funbound.plasma is
unvailable for the given species, the human Funbound.plasma is substituted.
If none of these are available, the mean human Rblood2plasma from
chem.physical\_and\_invitro.data is returned.  
details than the description above \textasciitilde{}\textasciitilde{}
\end{Details}
%
\begin{Value}
The blood to plasma chemical concentration ratio -- measured if available,
calculated if not.
\end{Value}
%
\begin{Author}\relax
Robert Pearce
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

available_rblood2plasma(chem.name="Bisphenol A",adjusted.Funbound.plasma=FALSE)
available_rblood2plasma(chem.name="Bisphenol A",species="Rat")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{blood\_mass\_correct}{Find average blood masses by age.}{blood.Rul.mass.Rul.correct}
\keyword{httk-pop}{blood\_mass\_correct}
%
\begin{Description}\relax
If blood mass from \code{\LinkA{blood\_weight}{blood.Rul.weight}} is negative or very small,
then just default to the mean blood mass by age. (Geigy Scientific Tables,
7th ed.)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
blood_mass_correct(blood_mass, age_months, age_years, gender, weight)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{blood\_mass}] A vector of blood masses in kg to be replaced with
averages.

\item[\code{age\_months}] A vector of ages in months.

\item[\code{age\_years}] A vector of ages in years.

\item[\code{gender}] A vector of genders (either 'Male' or 'Female').

\item[\code{weight}] A vector of body weights in kg.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of blood masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Geigy Pharmaceuticals, "Scientific Tables", 7th Edition, 
John Wiley and Sons (1970)

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{blood\_weight}{Predict blood mass.}{blood.Rul.weight}
\keyword{httk-pop}{blood\_weight}
%
\begin{Description}\relax
Predict blood mass based on body surface area and gender, using equations
from Bosgra et al. 2012
\end{Description}
%
\begin{Usage}
\begin{verbatim}
blood_weight(BSA, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{BSA}] Body surface area in m\textasciicircum{}2. May be a vector.

\item[\code{gender}] Either 'Male' or 'Female'. May be a vector.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of blood masses in kg the same length as \code{BSA} and
\code{gender}.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Bosgra, Sieto, et al. "An improved model to predict 
physiologically based model parameters and their inter-individual variability 
from anthropometry." Critical reviews in toxicology 42.9 (2012): 751-767.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{bmiage}{CDC BMI-for-age charts}{bmiage}
\keyword{data}{bmiage}
\keyword{httk-pop}{bmiage}
%
\begin{Description}\relax
Charts giving the BMI-for-age percentiles for boys and girls ages 2-18
\end{Description}
%
\begin{Usage}
\begin{verbatim}
bmiage
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table object with variables \begin{description}

\item[\code{Sex}] 'Male' or 'Female'
\item[\code{Agemos}] Age in months
\item[\code{L},
\code{M}, \code{S}] LMS parameters; see
\url{https://www.cdc.gov/growthcharts/percentile_data_files.htm}
\item[\code{P3},
\code{P5}, \code{P10}, \code{P25}, \code{P50}, \code{P75}, \code{P85},
\code{P90}, \code{P95}, and \code{P97}] BMI percentiles
\end{description}

\end{Format}
%
\begin{Details}\relax
For children ages 2 to 18, weight class depends on the BMI-for-age percentile.
\begin{description}

\item[Underweight] <5th percentile
\item[Normal weight] 5th-85th percentile
\item[Overweight] 85th-95th percentile
\item[Obese] >=95th percentile

\end{description}

\end{Details}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{Source}\relax
\url{https://www.cdc.gov/growthcharts/percentile_data_files.htm}
\end{Source}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{body\_surface\_area}{Predict body surface area.}{body.Rul.surface.Rul.area}
\keyword{httk-pop}{body\_surface\_area}
%
\begin{Description}\relax
Predict body surface area from weight, height, and age, using Mosteller's
formula for age>18 and Haycock's formula for age<18
\end{Description}
%
\begin{Usage}
\begin{verbatim}
body_surface_area(BW, H, age_years)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{BW}] A vector of body weights in kg.

\item[\code{H}] A vector of heights in cm.

\item[\code{age\_years}] A vector of ages in years.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of body surface areas in cm\textasciicircum{}2.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Mosteller, R. D. "Simplified calculation of body surface area." 
N Engl J Med 317 (1987): 1098..

Haycock, George B., George J. Schwartz, and David H. Wisotsky. "Geometric 
method for measuring body surface area: a height-weight formula validated in 
infants, children, and adults." The Journal of pediatrics 93.1 (1978): 62-66.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{bone\_mass\_age}{Predict bone mass}{bone.Rul.mass.Rul.age}
\keyword{httk-pop}{bone\_mass\_age}
%
\begin{Description}\relax
Predict bone mass from age\_years, height, weight, gender, using logistic
equations fit to data from Baxter-Jones et al. 2011, or for infants < 1
year, using equation from Koo et al. 2000 (See Price et al. 2003)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
bone_mass_age(age_years, age_months, height, weight, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{age\_years}] Vector of ages in years.

\item[\code{age\_months}] Vector of ages in months.

\item[\code{height}] Vector of heights in cm.

\item[\code{weight}] Vector of body weights in kg.

\item[\code{gender}] Vector of genders, either 'Male' or 'Female'.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Vector of bone masses.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Baxter-Jones, Adam DG, et al. "Bone mineral accrual from 8 to 30 years of age: 
an estimation of peak bone mass." Journal of Bone and Mineral Research 26.8 
(2011): 1729-1739.

Koo, Winston WK, and Elaine M. Hockman. "Physiologic predictors of lumbar 
spine bone mass in neonates." Pediatric research 48.4 (2000): 485-489.

Price, Paul S., et al. "Modeling interindividual variation in physiological 
factors used in PBPK models of humans." Critical reviews in toxicology 33.5 
(2003): 469-503.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{brain\_mass}{Predict brain mass.}{brain.Rul.mass}
\keyword{httk-pop}{brain\_mass}
%
\begin{Description}\relax
Predict brain mass from gender and age.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
brain_mass(gender, age_years)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{gender}] Vector of genders, either 'Male' or 'Female'

\item[\code{age\_years}] Vector of ages in years.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of brain masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{calc\_analytic\_css}{Calculate the analytic steady state concentration.}{calc.Rul.analytic.Rul.css}
\keyword{Solve}{calc\_analytic\_css}
%
\begin{Description}\relax
This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing for the three compartment and 
multiple compartment PBTK models.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_analytic_css(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "human",
  daily.dose = 1,
  output.units = "uM",
  model = "pbtk",
  concentration = "plasma",
  suppress.messages = FALSE,
  tissue = NULL,
  restrictive.clearance = T,
  bioactive.free.invivo = F,
  IVIVE = NULL,
  parameterize.args = list(default.to.human = FALSE, adjusted.Funbound.plasma = TRUE,
    regression = TRUE, minimum.Funbound.plasma = 1e-04),
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk (for model = 
'pbtk'), parameterize\_3comp (for model = '3compartment), 
parameterize\_1comp(for model = '1compartment') or parameterize\_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{daily.dose}] Total daily dose, mg/kg BW.

\item[\code{output.units}] Units for returned concentrations, defaults to uM 
(specify units = "uM") but can also be mg/L.

\item[\code{model}] Model used in calculation, 'pbtk' for the multiple compartment 
model,'3compartment' for the three compartment model, '3compartmentss' for 
the three compartment steady state model, and '1compartment' for one 
compartment model.

\item[\code{concentration}] Desired concentration type, 'blood','tissue', or default 'plasma'.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{tissue}] Desired tissue conentration (defaults to whole body 
concentration.)

\item[\code{restrictive.clearance}] If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).

\item[\code{bioactive.free.invivo}] If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.

\item[\code{IVIVE}] Honda et al. (2019) identified four plausible sets of 
assumptions for \emph{in vitro-in vivo} extrapolation (IVIVE) assumptions. 
Argument may be set to "Honda1" through "Honda4". If used, this function 
overwrites the tissue, restrictive.clearance, and bioactive.free.invivo arguments. 
See Details below for more information.

\item[\code{parameterize.args}] List of arguments passed to model's associated
parameterization function, including default.to.human, 
adjusted.Funbound.plasma, regression, and minimum.Funbound.plasma. The 
default.to.human argument substitutes missing animal values with human values
if true, adjusted.Funbound.plasma returns adjusted Funbound.plasma when set 
to TRUE along with parition coefficients calculated with this value, 
regression indicates whether or not to use the regressions in calculating
partition coefficients, and minimum.Funbound.plasma is the value to which
Monte Carlo draws less than this value are set (default is 0.0001 -- half
the lowest measured Fup in our dataset).

\item[\code{...}] Additional parameters passed to parameterize function if 
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Concentrations are calculated for the specifed model with constant 
oral infusion dosing.  All tissues other than gut, liver, and lung are the 
product of the steady state plasma concentration and the tissue to plasma 
partition coefficient. 

\Tabular{lrrrr}{
& \emph{in vivo} Conc. & Metabolic Clearance & Bioactive Chemical Conc. & TK Statistic Used* \\{}
Honda1 & Veinous (Plasma) & Restrictive & Free & Mean Conc. \\{}
Honda2 & Veinous & Restrictive & Free & Max Conc. \\{}
Honda3 & Veinous & Non-restrictive & Total & Mean Conc. \\{}
Honda4 & Veinous & Non-restrictive & Total & Max Conc. \\{}
Honda5 & Target Tissue & Non-restrictive & Total & Mean Conc. \\{}
Honda6 & Target Tissue & Non-restrictive & Total & Max Conc. \\{}
}
*Assumption is currently ignored because analytical steady-state solutions are currently used by this function.
\end{Details}
%
\begin{Value}
Steady state concentration
\end{Value}
%
\begin{Author}\relax
Robert Pearce, John Wambaugh, and Greg Honda
\end{Author}
%
\begin{References}\relax
Honda, Gregory S., et al. "Using the Concordance of In Vitro and 
In Vivo Data to Evaluate Extrapolation Assumptions." 2019. PLoS ONE 14(5): e0217564.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
calc_analytic_css(chem.name='Bisphenol-A',output.units='mg/L',
                 model='3compartment',concentration='blood')

calc_analytic_css(chem.name='Bisphenol-A',tissue='liver',species='rabbit',
                 parameterize.args = list(
                                default.to.human=TRUE,
                                adjusted.Funbound.plasma=TRUE,
                                regression=TRUE,
                                minimum.Funbound.plasma=1e-4),daily.dose=2)

calc_analytic_css(chem.name="bisphenol a",model="1compartment")

calc_analytic_css(chem.cas="80-05-7",model="3compartmentss")

params <- parameterize_pbtk(chem.cas="80-05-7") 

calc_analytic_css(parameters=params,model="pbtk")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_analytic\_css\_1comp}{Calculate the analytic steady state concentration for the one compartment model.}{calc.Rul.analytic.Rul.css.Rul.1comp}
\keyword{1compartment}{calc\_analytic\_css\_1comp}
%
\begin{Description}\relax
This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_analytic_css_1comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = F,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk (for model = 
'pbtk'), parameterize\_3comp (for model = '3compartment), 
parameterize\_1comp(for model = '1compartment') or parameterize\_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.

\item[\code{hourly.dose}] Hourly dose rate mg/kg BW/h.

\item[\code{concentration}] Desired concentration type, 'blood' or default 'plasma'.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
altered hematocrit, Funbound.plasma, or Krbc2pu.

\item[\code{tissue}] Desired tissue conentration (defaults to whole body 
concentration.)

\item[\code{restrictive.clearance}] If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).

\item[\code{bioactive.free.invivo}] If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.

\item[\code{...}] Additional parameters passed to parameterize function if 
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Steady state concentration in uM units
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{calc\_analytic\_css\_3comp}{Calculate the analytic steady state concentration for model 3comp}{calc.Rul.analytic.Rul.css.Rul.3comp}
\keyword{3compartment}{calc\_analytic\_css\_3comp}
%
\begin{Description}\relax
This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_analytic_css_3comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk (for model = 
'pbtk'), parameterize\_3comp (for model = '3compartment), 
parameterize\_1comp(for model = '1compartment') or parameterize\_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.

\item[\code{hourly.dose}] Hourly dose rate mg/kg BW/h.

\item[\code{concentration}] Desired concentration type, 'blood' or default 'plasma'.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
'altered hematocrit, Funbound.plasma, or Krbc2pu.

\item[\code{tissue}] Desired tissue conentration (defaults to whole body 
concentration.)

\item[\code{restrictive.clearance}] If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).

\item[\code{bioactive.free.invivo}] If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with \code{tissue = NULL} in current implementation.

\item[\code{...}] Additional parameters passed to parameterize function if 
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Steady state concentration in uM units
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{calc\_analytic\_css\_3compss}{Calculate the analytic steady state concentration for the three compartment steady-state model}{calc.Rul.analytic.Rul.css.Rul.3compss}
\keyword{3compss}{calc\_analytic\_css\_3compss}
%
\begin{Description}\relax
This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_analytic_css_3compss(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk (for model = 
'pbtk'), parameterize\_3comp (for model = '3compartment), 
parameterize\_1comp(for model = '1compartment') or parameterize\_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.

\item[\code{hourly.dose}] Hourly dose rate mg/kg BW/h.

\item[\code{concentration}] Desired concentration type, 'blood' or default 'plasma'.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
'altered hematocrit, Funbound.plasma, or Krbc2pu.

\item[\code{tissue}] Desired tissue concentration (defaults to whole body 
concentration.)

\item[\code{restrictive.clearance}] If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).

\item[\code{bioactive.free.invivo}] If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.

\item[\code{...}] Additional parameters passed to parameterize function if 
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Steady state concentration in uM units
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{calc\_analytic\_css\_pbtk}{Calculate the analytic steady state concentration for model pbtk.}{calc.Rul.analytic.Rul.css.Rul.pbtk}
\keyword{pbtk}{calc\_analytic\_css\_pbtk}
%
\begin{Description}\relax
This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_analytic_css_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must 
be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk (for model = 
'pbtk'), parameterize\_3comp (for model = '3compartment), 
parameterize\_1comp(for model = '1compartment') or parameterize\_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.

\item[\code{hourly.dose}] Hourly dose rate mg/kg BW/h.

\item[\code{concentration}] Desired concentration type, 'blood', 'tissue', or default 'plasma'.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
'altered hematocrit, Funbound.plasma, or Krbc2pu.

\item[\code{tissue}] Desired tissue conentration (defaults to whole body 
concentration.)

\item[\code{restrictive.clearance}] If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).

\item[\code{bioactive.free.invivo}] If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.

\item[\code{...}] Additional parameters passed to parameterize function if 
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Steady state concentration in uM units
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{calc\_css}{Find the steady state concentration and the day it is reached.}{calc.Rul.css}
\keyword{Steady-State}{calc\_css}
%
\begin{Description}\relax
This function finds the day a chemical comes within the specified range of
the analytical steady state venous blood or plasma concentration(from
calc\_analytic\_css) for the multiple compartment, three compartment, and one
compartment models, the fraction of the true steady state value reached on
that day, the maximum concentration, and the average concentration at the
end of the simulation.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_css(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  f = 0.01,
  daily.dose = 1,
  doses.per.day = 3,
  days = 21,
  output.units = "uM",
  suppress.messages = FALSE,
  tissue = "plasma",
  model = "pbtk",
  default.to.human = FALSE,
  f.change = 1e-05,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  dosing = NULL,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or parameters must be
specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or parameters must be
specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk function,
overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{f}] Fractional distance from the final steady state concentration that
the average concentration must come within to be considered at steady state.

\item[\code{daily.dose}] Total daily dose, mg/kg BW.

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{days}] Initial number of days to run simulation that is multiplied on
each iteration.

\item[\code{output.units}] Units for returned concentrations, defaults to uM
(specify units = "uM") but can also be mg/L.

\item[\code{suppress.messages}] Whether or not to suppress messages.

\item[\code{tissue}] Desired tissue concentration (defaults to whole body 
concentration.)

\item[\code{model}] Model used in calculation, 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, and '1compartment' for
the one compartment model.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{f.change}] Fractional change of daily steady state concentration
reached to stop calculating.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE for model 1compartment elimination
rate.  This assumes clearance relative to amount unbound in whole blood
instead of plasma, but converted to use with plasma concentration.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{dosing}] The dosing object for more complicated scenarios. Defaults to
repeated \code{daily.dose} spread out over \code{doses.per.day}

\item[\code{...}] Additional arguments passed to model solver (default of
\code{\LinkA{solve\_pbtk}{solve.Rul.pbtk}}).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{frac}] Ratio of the mean concentration on the day steady state
is reached (baed on doses.per.day) to the analytical Css (based on infusion
dosing).\item[\code{max}] The maximum concentration of the simulation.
\item[\code{avg}] The average concentration on the final day of the simulation.
\item[\code{the.day}] The day the average concentration comes within 100 * p
percent of the true steady state concentration.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce, John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

calc_css(chem.name='Bisphenol-A',doses.per.day=5,f=.001,output.units='mg/L')

parms <- parameterize_3comp(chem.name='Bisphenol-A')
parms$Funbound.plasma <- .07
calc_css(parameters=parms,model='3compartment')

out <- solve_pbtk(chem.name = "Bisphenol A",
  days = 50,                                   
  daily.dose=1,
  doses.per.day = 3)
plot.data <- as.data.frame(out)

css <- calc_analytic_css(chem.name = "Bisphenol A")
library("ggplot2")
c.vs.t <- ggplot(plot.data,aes(time, Cplasma)) + geom_line() +
geom_hline(yintercept = css) + ylab("Plasma Concentration (uM)") +
xlab("Day") + theme(axis.text = element_text(size = 16), axis.title =
element_text(size = 16), plot.title = element_text(size = 17)) +
ggtitle("Bisphenol A")

print(c.vs.t)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_elimination\_rate}{Calculate the elimination rate for a one compartment model.}{calc.Rul.elimination.Rul.rate}
\keyword{1compartment}{calc\_elimination\_rate}
\keyword{Parameter}{calc\_elimination\_rate}
%
\begin{Description}\relax
This function calculates an elimination rate from the three compartment
steady state model where elimination is entirely due to metablism by the
liver and glomerular filtration in the kidneys.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_elimination_rate(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  suppress.messages = FALSE,
  default.to.human = FALSE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  well.stirred.correction = TRUE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the cas number or the chemical name must be
specified.

\item[\code{chem.name}] Either the chemical name or the cas number must be
specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_steadystate or
1compartment function, overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{restrictive.clearance}] In calculating elimination rate, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for -stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Elimination rate calculated by dividing the total clearance (using the
default -stirred hepatic model) by the volume of distribution. When
species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{Elimination rate}] Units of 1/h.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

calc_elimination_rate(chem.name="Bisphenol A")
calc_elimination_rate(chem.name="Bisphenol A",species="Rat")
calc_elimination_rate(chem.cas="80-05-7")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_half\_life}{Calculates the half-life for a one compartment model.}{calc.Rul.half.Rul.life}
\keyword{1compartment}{calc\_half\_life}
\keyword{Parameter}{calc\_half\_life}
%
\begin{Description}\relax
This function calculates the half life from the three compartment
steady state model where elimination is entirely due to metabolism by the
liver and glomerular filtration in the kidneys.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_half_life(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  suppress.messages = FALSE,
  default.to.human = FALSE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  well.stirred.correction = TRUE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the cas number or the chemical name must be specified.

\item[\code{chem.name}] Either the chemical name or the cas number must be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_steadystate or
1compartment function, overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{restrictive.clearance}] In calculating elimination rate, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for -stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Half life is calculated by dividing the natural-log of 2 by the elimination
rate from the one compartment model.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{Half life}] Units of h.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Sarah E. Davidson
\end{Author}
%
\begin{SeeAlso}\relax
[calc\_elimination\_rate()] for the elimination rate calculation
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}

calc_half_life(chem.name="Bisphenol A")
calc_half_life(chem.name="Bisphenol A",species="Rat")
calc_half_life(chem.cas="80-05-7")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_hepatic\_clearance}{Calculate the hepatic clearance (deprecated).}{calc.Rul.hepatic.Rul.clearance}
\keyword{Parameter}{calc\_hepatic\_clearance}
%
\begin{Description}\relax
This function is included for backward compatibility. It calls
\code{\LinkA{calc\_hep\_clearance}{calc.Rul.hep.Rul.clearance}} which
calculates the hepatic clearance in plasma for a well-stirred model
or other type if specified. Based on  Ito and Houston (2004)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_hepatic_clearance(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  default.to.human = FALSE,
  hepatic.model = "well-stirred",
  suppress.messages = FALSE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_steadystate
function, overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{hepatic.model}] Model used in calculating hepatic clearance, unscaled,
parallel tube, dispersion, or default well-stirred.

\item[\code{suppress.messages}] Whether or not to suppress the output message.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE for hepatic.model well-stirred.
This assumes clearance relative to amount unbound in whole blood instead of
plasma, but converted to use with plasma concentration.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE.

\item[\code{...}] Additional parameters passed to parameterize\_steadystate if
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{Hepatic Clearance}] Units of L/h/kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Ito, K., \& Houston, J. B. (2004). "Comparison of the use of liver models for 
predicting drug clearance using in vitro kinetic data from hepatic microsomes 
and isolated hepatocytes." Pharmaceutical Tesearch, 21(5), 785-792.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

calc_hep_clearance(chem.name="Ibuprofen",hepatic.model='unscaled')
calc_hep_clearance(chem.name="Ibuprofen",well.stirred.correction=FALSE)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_hep\_bioavailability}{Calculate first pass metabolism}{calc.Rul.hep.Rul.bioavailability}
\keyword{physiology}{calc\_hep\_bioavailability}
%
\begin{Description}\relax
For models that don't described first pass blood flow from the gut, need to
cacluate a hepatic bioavailability, that is, the fraction of chemical 
systemically available after metabolism during the first pass through the 
liver (Rowland, 1973 Equaation 29, where k21 is blood flow through the liver
and k23 is clearance from the liver in Figure 1).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_hep_bioavailability(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  restrictive.clearance = TRUE,
  flow.34 = TRUE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from the appropriate parameterization function
for the model indicated by argument model

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to 1) in 
liver clearance if FALSE.

\item[\code{flow.34}] A logical constraint
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table whose columns are the parameters of the HTTK model
specified in \code{model}.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Rowland, Malcolm, Leslie Z. Benet, and Garry G. Graham. 
"Clearance concepts in pharmacokinetics." Journal of pharmacokinetics and 
biopharmaceutics 1.2 (1973): 123-136.
\end{References}
\inputencoding{utf8}
\HeaderA{calc\_hep\_clearance}{Calculate the hepatic clearance.}{calc.Rul.hep.Rul.clearance}
\keyword{Parameter}{calc\_hep\_clearance}
%
\begin{Description}\relax
This function calculates the hepatic clearance in plasma for a well-stirred model
or other type if specified. Based on  Ito and Houston (2004)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_hep_clearance(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  default.to.human = FALSE,
  hepatic.model = "well-stirred",
  suppress.messages = FALSE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_steadystate
function, overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{hepatic.model}] Model used in calculating hepatic clearance, unscaled,
parallel tube, dispersion, or default well-stirred.

\item[\code{suppress.messages}] Whether or not to suppress the output message.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE for hepatic.model well-stirred.
This assumes clearance relative to amount unbound in whole blood instead of
plasma, but converted to use with plasma concentration.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE.

\item[\code{...}] Additional parameters passed to parameterize\_steadystate if
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{Hepatic Clearance}] Units of L/h/kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Ito, K., \& Houston, J. B. (2004). "Comparison of the use of liver models for 
predicting drug clearance using in vitro kinetic data from hepatic microsomes 
and isolated hepatocytes." Pharmaceutical Tesearch, 21(5), 785-792.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

calc_hep_clearance(chem.name="Ibuprofen",hepatic.model='unscaled')
calc_hep_clearance(chem.name="Ibuprofen",well.stirred.correction=FALSE)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_hep\_fu}{Calculate the free chemical in the hepaitic clearance assay}{calc.Rul.hep.Rul.fu}
\keyword{in-vitro}{calc\_hep\_fu}
%
\begin{Description}\relax
Method from Kilford et al. (2008) for fraction of unbound chemical in the 
hepatocyte intrinsic clearance assay
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_hep_fu(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  Vr = 0.005,
  pH = 7.4
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from the appropriate parameterization function
for the model indicated by argument model

\item[\code{Vr}] Rratio of cell volume to incubation volume. Default is taken from

\item[\code{pH}] pH of the incupation medium.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A numeric fraction between zero and one
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Kilford, Peter J., et al. "Hepatocellular binding of drugs: 
correction for unbound fraction in hepatocyte incubations using microsomal 
binding or drug lipophilicity data." Drug Metabolism and Disposition 36.7 
(2008): 1194-1197.

Wetmore, Barbara A., et al. "Incorporating high-throughput exposure 
predictions with dosimetry-adjusted in vitro bioactivity to inform chemical 
toxicity testing." Toxicological Sciences 148.1 (2015): 121-136.
\end{References}
\inputencoding{utf8}
\HeaderA{calc\_ionization}{Calculate the ionization.}{calc.Rul.ionization}
\keyword{Parameter}{calc\_ionization}
%
\begin{Description}\relax
This function calculates the ionization of a compound at a given pH. The 
pKa's are either entered as parameters or taken from a specific compound in
the package.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_ionization(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  pH = NULL,
  pKa_Donor = NULL,
  pKa_Accept = NULL
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the chemical name or the CAS number must be
specified.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from a parameterize\_MODEL function,
overrides chem.name and chem.cas.

\item[\code{pH}] pH where ionization is evaluated.

\item[\code{pKa\_Donor}] Compound H dissociation equilibirum constant(s).
Overwrites chem.name and chem.cas.

\item[\code{pKa\_Accept}] Compound H association equilibirum constant(s).
Overwrites chem.name and chem.cas.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
The arguments pKa\_Donor and pKa\_Accept may be single numbers, characters, or 
vectors. We support characters because there are many instances with multiple 
predicted values and all those values can be included by concatenating with 
commas (for example, pKa\_Donor = "8.1,8.6". Finally, pka\_Donor and pKa\_Accept 
may be vectors of characters representing different chemicals or instances of
chemical parameters to allow for uncertainty analysis.

The fractions are calculated by determining the coefficients for each
species and dividing the particular species by the sum of all three.  The
positive, negative and zwitterionic/neutral coefficients are given by:
\deqn{zwitter/netural = 1}{} \deqn{for(i in 1:pkabove) negative = negative +
10^(i * pH - pKa1 - ... - pKai)}{} \deqn{for(i in 1:pkbelow) positive =
positive + 10^(pKa1 + ... + pKai - i * pH)}{} where i begins at 1 and ends at
the number of points above(for negative) or below(for positive) the
neutral/zwitterionic range.  The neutral/zwitterionic range is either the pH
range between 2 pKa's where the number of acceptors above is equal to the
number of donors below, everything above the pKa acceptors if there are no
donors, or everything below the pKa donors if there are no acceptors.  Each
of the terms in the sums represent a different ionization.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{fraction\_neutral}] fraction of compound neutral
\item[\code{fraction\_charged}] fraction of compound charged
\item[\code{fraction\_negative}] fraction of compound negative
\item[\code{fraction\_positive}] fraction of compound positive
\item[\code{fraction\_zwitter}] fraction of compound zwitterionic
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Evaluation and calibration of
high-throughput predictions of chemical distribution to tissues." Journal of
Pharmacokinetics and Pharmacodynamics 44.6 (2017): 549-565.

Strope, Cory L., et al. "High-throughput in-silico prediction of ionization 
equilibria for pharmacokinetic modeling." Science of The Total Environment 
615 (2018): 150-160.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
# Donor pKa's 9.78,10.39 -- Should be almost all neutral at plasma pH:
out <- calc_ionization(chem.name='bisphenola',pH=7.4)
print(out)
out[["fraction_neutral"]]==max(unlist(out))

# Donor pKa's 9.78,10.39 -- Should be almost all negative (anion) at higher pH:
out <- calc_ionization(chem.name='bisphenola',pH=11)
print(out)
out[["fraction_negative"]]==max(unlist(out))

# Fictious compound, should be almost all all negative (anion):
out <- calc_ionization(pKa_Donor=8,pKa_Accept="1,4",pH=9)
print(out)
out[["fraction_negative"]]>0.9

# Donor pKa 6.54 -- Should be mostly negative (anion):
out <- calc_ionization(chem.name='Acephate',pH=7)
print(out)
out[["fraction_negative"]]==max(unlist(out))

#Acceptor pKa's "9.04,6.04"  -- Should be almost all positive (cation) at plasma pH:
out <- calc_ionization(chem.cas="145742-28-5",pH=7.4)
print(out)
out[["fraction_positive"]]==max(unlist(out))

#Fictious Zwitteron:
out <- calc_ionization(pKa_Donor=6,pKa_Accept="8",pH=7.4)
print(out)
out[["fraction_zwitter"]]==max(unlist(out))

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_krbc2pu}{Back-calculates the Red Blood Cell to Unbound Plasma Partition Coefficient}{calc.Rul.krbc2pu}
\keyword{Parameter}{calc\_krbc2pu}
%
\begin{Description}\relax
Given and observed ratio of chemial concentration in blood to plasma, this
function calculates a Red Blood Cell to unbound plasma (Krbc2pu) partition
coefficient that would be consistent with that observation.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_krbc2pu(
  Rb2p,
  Funbound.plasma,
  hematocrit = NULL,
  default.to.human = FALSE,
  species = "Human",
  suppress.messages = TRUE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{Rb2p}] The chemical blood:plasma concentration ratop

\item[\code{Funbound.plasma}] The free fraction of chemical in the presence of 
plasma protein
Rblood2plasma.

\item[\code{hematocrit}] Overwrites default hematocrit value in calculating
Rblood2plasma.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{suppress.messages}] Determine whether to display certain usage
feedback.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The red blood cell to unbound chemical in plasma partition coefficient.
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues." Journal of 
pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.

Ruark, Christopher D., et al. "Predicting passive and active tissue: plasma 
partition coefficients: interindividual and interspecies variability." 
Journal of pharmaceutical sciences 103.7 (2014): 2189-2198.
\end{References}
\inputencoding{utf8}
\HeaderA{calc\_mc\_css}{Find the monte carlo steady state concentration.}{calc.Rul.mc.Rul.css}
\keyword{Monte-Carlo}{calc\_mc\_css}
\keyword{Steady-State}{calc\_mc\_css}
%
\begin{Description}\relax
This function finds the analytical steady state plasma concentration(from
calc\_analytic\_css) using a monte carlo simulation (monte\_carlo).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_mc_css(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  samples = 1000,
  which.quantile = 0.95,
  species = "Human",
  suppress.messages = FALSE,
  model = "3compartmentss",
  httkpop = TRUE,
  invitrouv = TRUE,
  calcrb2p = TRUE,
  censored.params = list(),
  vary.params = list(),
  return.samples = FALSE,
  tissue = NULL,
  output.units = "mg/L",
  invitro.mc.arg.list = list(adjusted.Funbound.plasma = TRUE, poormetab = TRUE,
    fup.censored.dist = FALSE, fup.lod = 0.01, fup.meas.cv = 0.4, clint.meas.cv = 0.3,
    fup.pop.cv = 0.3, clint.pop.cv = 0.3),
  httkpop.generate.arg.list = list(method = "direct resampling", gendernum = NULL,
    agelim_years = NULL, agelim_months = NULL, weight_category = c("Underweight",
    "Normal", "Overweight", "Obese"), gfr_category = c("Normal", "Kidney Disease",
    "Kidney Failure"), reths = c("Mexican American", "Other Hispanic",
    "Non-Hispanic White", "Non-Hispanic Black", "Other")),
  convert.httkpop.arg.list = list(),
  parameterize.arg.list = list(default.to.human = FALSE, clint.pvalue.threshold = 0.05,
    restrictive.clearance = T, regression = TRUE),
  calc.analytic.css.arg.list = list(well.stirred.correction = TRUE,
    adjusted.Funbound.plasma = TRUE, regression = TRUE, IVIVE = NULL, tissue = tissue,
    restrictive.clearance = T, bioactive.free.invivo = FALSE)
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from the appropriate parameterization function
for the model indicated by argument model

\item[\code{samples}] Number of samples generated in calculating quantiles.

\item[\code{which.quantile}] Which quantile from Monte Carlo simulation is
requested. Can be a vector.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").  Species must be set to "Human" to run httkpop model.

\item[\code{suppress.messages}] Whether or not to suppress output message.

\item[\code{model}] Model used in calculation: 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.  This only applies when httkpop=TRUE and species="Human",
otherwise '3compartmentss' is used.

\item[\code{httkpop}] Whether or not to use population generator and sampler from
httkpop.  This is overwrites censored.params and vary.params and is only for
human physiology.  Species must also be set to 'Human'.

\item[\code{invitrouv}] Logical to indicate whether to include in vitro parameters
in uncertainty and variability analysis

\item[\code{calcrb2p}] Logical determining whether or not to recalculate the 
chemical ratio of blood to plasma

\item[\code{censored.params}] The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sublists. Each sublist is named for a parameter in
"parameters" and contains two elements: "CV" (coefficient of variation) and
"LOD" (limit of detection, below which parameter values are censored. New
values are sampled with mean equal to the value in "parameters" and standard
deviation equal to the mean times the CV.  Censored values are sampled on a
uniform distribution between 0 and the limit of detection. Not used with
httkpop model.

\item[\code{vary.params}] The parameters listed in vary.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (CV) for the normal distribution. Each
entry in the list is named for a parameter in "parameters". New values are
sampled with mean equal to the value in "parameters" and standard deviation
equal to the mean times the CV. Not used with httkpop model.

\item[\code{return.samples}] Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.

\item[\code{tissue}] Desired steady state tissue concentration.

\item[\code{output.units}] Plasma concentration units, either uM or default mg/L.

\item[\code{invitro.mc.arg.list}] List of additional parameters passed to 
\code{\LinkA{invitro\_mc}{invitro.Rul.mc}}

\item[\code{httkpop.generate.arg.list}] Additional parameters passed to 
\code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.

\item[\code{convert.httkpop.arg.list}] Additional parameters passed to the 
convert\_httkpop\_* function for the model.

\item[\code{parameterize.arg.list}] Additional parameters passed to the 
parameterize\_* function for the model.

\item[\code{calc.analytic.css.arg.list}] Additional parameters passed to 
\code{\LinkA{calc\_analytic\_css}{calc.Rul.analytic.Rul.css}}.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
All arguments after httkpop only apply if httkpop is set to TRUE and species
to "Human".

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data (volumes and flows) but substitutes human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.

Tissue concentrations are calculated for the pbtk model with a default oral
infusion dosing. All tissues other than gut, liver, and lung are the product
of the steady state plasma concentration and the tissue to plasma partition
coefficient.

The six sets of plausible \emph{in vitro-in vivo} extrapolation (IVIVE)
assumptions identified by Honda et al. (2019) are: 
\Tabular{lrrrr}{
& \emph{in vivo} Conc. & Metabolic Clearance & Bioactive Chemical
Conc. & TK Statistic Used* \\{} Honda1 & Veinous (Plasma) &
Restrictive & Free & Mean Conc. \\{} Honda2 & Veinous &
Restrictive & Free & Max Conc. \\{} Honda3 & Veinous &
Non-restrictive & Total & Mean Conc. \\{} Honda4 & Veinous &
Non-restrictive & Total & Max Conc. \\{} Honda5 & Target Tissue &
Non-restrictive & Total & Mean Conc. \\{} Honda6 & Target Tissue
& Non-restrictive & Total & Max Conc. \\{} } *Assumption is
currently ignored because analytical steady-state solutions are currently
used by this function.
\end{Details}
%
\begin{Value}
Quantiles (specified by which.quantile) of the distribution of plasma
steady-stae concentration (Css) from the Monte Carlo simulation
\end{Value}
%
\begin{Author}\relax
Caroline Ring, Robert Pearce, and John Wambaugh
\end{Author}
%
\begin{References}\relax
Wambaugh, John F., et al. "Toxicokinetic triage for 
environmental chemicals." Toxicological Sciences 147.1 (2015): 55-67.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment international 106 (2017): 105-118. 

Honda, Gregory S., et al. "Using the Concordance of In Vitro and 
In Vivo Data to Evaluate Extrapolation Assumptions." 2019. PLoS ONE 14(5): e0217564.

Rowland, Malcolm, Leslie Z. Benet, and Garry G. Graham. "Clearance concepts in 
pharmacokinetics." Journal of pharmacokinetics and biopharmaceutics 1.2 (1973): 123-136.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}


 set.seed(1234)
 calc_mc_css(chem.name='Bisphenol A',output.units='uM',
             samples=100,return.samples=TRUE)
             
 set.seed(1234)
 calc_mc_css(chem.name='2,4-d',which.quantile=.9,httkpop=FALSE,tissue='heart')

 set.seed(1234)
 calc_mc_css(chem.cas = "80-05-7", which.quantile = 0.5,
             output.units = "uM", samples = 2000,
             httkpop.generate.arg.list=list(method='vi', gendernum=NULL, 
             agelim_years=NULL, agelim_months=NULL, weight_category = 
             c("Underweight", "Normal", "Overweight", "Obese")))

 params <- parameterize_pbtk(chem.cas="80-05-7")
 set.seed(1234)
 calc_mc_css(parameters=params,model="pbtk")



 set.seed(1234)
 # Standard HTTK Monte Carlo:
 NSAMP = 500
 calc_mc_css(chem.cas="90-43-7",model="pbtk",samples=NSAMP)
 set.seed(1234)
 calc_mc_css(chem.cas="90-43-7",
 model="pbtk",
 samples=NSAMP,
 invitro.mc.arg.list = list(
   adjusted.Funbound.plasma = TRUE,
   poormetab = TRUE, 
   fup.censored.dist = FALSE, 
   fup.lod = 0.01, 
   fup.meas.cv = 0.0, 
   clint.meas.cv = 0.0, 
   fup.pop.cv = 0.3, 
   clint.pop.cv = 0.3))
 set.seed(1234)
 # HTTK Monte Carlo with no HTTK-Pop physiological variability):
 calc_mc_css(chem.cas="90-43-7",model="pbtk",samples=NSAMP,httkpop=FALSE)
 set.seed(1234)
 # HTTK Monte Carlo with no in vitro uncertainty and variability):
 calc_mc_css(chem.cas="90-43-7",model="pbtk",samples=NSAMP,invitrouv=FALSE)
 set.seed(1234)
 # HTTK Monte Carlo with no HTTK-Pop and no in vitro uncertainty and variability):
 calc_mc_css(chem.cas="90-43-7",model="pbtk",samples=NSAMP,httkpop=FALSE,invitrouv=FALSE)
 # Should be the same as the mean result:
 calc_analytic_css(chem.cas="90-43-7",model="pbtk",output.units="mg/L")
 set.seed(1234)
 # HTTK Monte Carlo using basic Monte Carlo sampler:
 calc_mc_css(chem.cas="90-43-7",
 model="pbtk",
 samples=NSAMP,
 httkpop=FALSE,
 invitrouv=FALSE,
 vary.params=list(Pow=0.3))


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_mc\_oral\_equiv}{Calculate Monte Carlo Oral Equivalent Dose}{calc.Rul.mc.Rul.oral.Rul.equiv}
\keyword{Monte-Carlo}{calc\_mc\_oral\_equiv}
\keyword{Steady-State}{calc\_mc\_oral\_equiv}
%
\begin{Description}\relax
This functions converts a chemical plasma concetration to an oral equivalent
dose using a concentration obtained from \code{\LinkA{calc\_mc\_css}{calc.Rul.mc.Rul.css}}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_mc_oral_equiv(
  conc,
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  which.quantile = 0.95,
  species = "Human",
  input.units = "uM",
  output.units = "mgpkgpday",
  suppress.messages = FALSE,
  return.samples = FALSE,
  concentration = "plasma",
  restrictive.clearance = TRUE,
  bioactive.free.invivo = F,
  tissue = NULL,
  IVIVE = NULL,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{conc}] Bioactive in vitro concentration in units of uM.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{chem.cas}] Either the CAS number or the chemical name must be
specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{which.quantile}] Which quantile from Monte Carlo steady-state
simulation (\code{\LinkA{calc\_mc\_css}{calc.Rul.mc.Rul.css}}) is requested. Can be a vector. Note that 95th
concentration quantile is the same population as the 5th dose quantile.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{input.units}] Units of given concentration, default of uM but can also
be mg/L.

\item[\code{output.units}] Units of dose, default of 'mgpkgpday' for mg/kg BW/ day or
'umolpkgpday' for umol/ kg BW/ day.

\item[\code{suppress.messages}] Suppress text messages.

\item[\code{return.samples}] Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.

\item[\code{concentration}] Desired concentration type, 'blood','tissue', or default 'plasma'.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{bioactive.free.invivo}] If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.

\item[\code{tissue}] Desired steady state tissue conentration.

\item[\code{IVIVE}] Honda et al. (2019) identified six plausible sets of
assumptions for \emph{in vitro-in vivo} extrapolation (IVIVE) assumptions.
Argument may be set to "Honda1" through "Honda6". If used, this function
overwrites the tissue, restrictive.clearance, and plasma.binding arguments.
See Details below for more information.

\item[\code{...}] Additional parameters passed to \code{\LinkA{calc\_mc\_css}{calc.Rul.mc.Rul.css}} for httkpop and
variance of parameters.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
All arguments after httkpop only apply if httkpop is set to TRUE and species
to "Human".

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitutes human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.

Tissue concentrations are calculated for the pbtk model with oral infusion
dosing.  All tissues other than gut, liver, and lung are the product of the
steady state plasma concentration and the tissue to plasma partition
coefficient.

The six sets of plausible \emph{in vitro-in vivo} extrapolation (IVIVE)
assumptions identified by Honda et al. (2019) are: 
\Tabular{lrrrr}{
& \emph{in vivo} Conc. & Metabolic Clearance & Bioactive Chemical
Conc. & TK Statistic Used* \\{} Honda1 & Veinous (Plasma) &
Restrictive & Free & Mean Conc. \\{} Honda2 & Veinous &
Restrictive & Free & Max Conc. \\{} Honda3 & Veinous &
Non-restrictive & Total & Mean Conc. \\{} Honda4 & Veinous &
Non-restrictive & Total & Max Conc. \\{} Honda5 & Target Tissue &
Non-restrictive & Total & Mean Conc. \\{} Honda6 & Target Tissue
& Non-restrictive & Total & Max Conc. \\{} } *Assumption is
currently ignored because analytical steady-state solutions are currently
used by this function.
\end{Details}
%
\begin{Value}
Equivalent dose in specified units, default of mg/kg BW/day.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Wetmore, Barbara A., et al. "Incorporating high-throughput 
exposure predictions with dosimetry-adjusted in vitro bioactivity to inform 
chemical toxicity testing." Toxicological Sciences 148.1 (2015): 121-136.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment international 106 (2017): 105-118. 

Honda, Gregory S., et al. "Using the Concordance of In Vitro and 
In Vivo Data to Evaluate Extrapolation Assumptions." 2019. PLoS ONE 14(5): e0217564.

Rowland, Malcolm, Leslie Z. Benet, and Garry G. Graham. "Clearance concepts in 
pharmacokinetics." Journal of pharmacokinetics and biopharmaceutics 1.2 (1973): 123-136.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}



calc_mc_oral_equiv(0.1,chem.cas="34256-82-1",which.quantile=c(0.05,0.5,0.95),
       tissue='brain')


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_mc\_tk}{Conduct multiple TK simulations using Monte Carlo}{calc.Rul.mc.Rul.tk}
\keyword{Monte-Carlo}{calc\_mc\_tk}
\keyword{dynamic}{calc\_mc\_tk}
\keyword{simulation}{calc\_mc\_tk}
%
\begin{Description}\relax
This function finds the analytical steady state plasma concentration(from
calc\_analytic\_css) using a monte carlo simulation (monte\_carlo).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_mc_tk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  samples = 1000,
  which.quantile = 0.95,
  species = "Human",
  suppress.messages = FALSE,
  model = "pbtk",
  httkpop = TRUE,
  invitrouv = TRUE,
  calcrb2p = TRUE,
  censored.params = list(),
  vary.params = list(),
  return.samples = FALSE,
  tissue = NULL,
  output.units = "mg/L",
  solvemodel.arg.list = list(times = c(0, 0.25, 0.5, 0.75, 1, 1.5, 2, 2.5, 3, 4, 5)),
  invitro.mc.arg.list = list(adjusted.Funbound.plasma = TRUE, poormetab = TRUE,
    fup.censored.dist = FALSE, fup.lod = 0.01, fup.meas.cv = 0.4, clint.meas.cv = 0.3,
    fup.pop.cv = 0.3, clint.pop.cv = 0.3),
  httkpop.generate.arg.list = list(method = "direct resampling", gendernum = NULL,
    agelim_years = NULL, agelim_months = NULL, weight_category = c("Underweight",
    "Normal", "Overweight", "Obese"), gfr_category = c("Normal", "Kidney Disease",
    "Kidney Failure"), reths = c("Mexican American", "Other Hispanic",
    "Non-Hispanic White", "Non-Hispanic Black", "Other")),
  convert.httkpop.arg.list = list(),
  parameterize.arg.list = list(default.to.human = FALSE, clint.pvalue.threshold = 0.05,
    restrictive.clearance = T, regression = TRUE),
  return.all.sims = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the CAS number, parameters, or the chemical name must
be specified.

\item[\code{chem.name}] Either the chemical parameters, name, or the CAS number
must be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from parameterize\_steadystate. Not used with
httkpop model.

\item[\code{samples}] Number of samples generated in calculating quantiles.

\item[\code{which.quantile}] Which quantile from Monte Carlo simulation is
requested. Can be a vector.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").  Species must be set to "Human" to run httkpop model.

\item[\code{suppress.messages}] Whether or not to suppress output message.

\item[\code{model}] Model used in calculation: 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.  This only applies when httkpop=TRUE and species="Human",
otherwise '3compartmentss' is used.

\item[\code{httkpop}] Whether or not to use population generator and sampler from
httkpop.  This is overwrites censored.params and vary.params and is only for
human physiology.  Species must also be set to 'Human'.

\item[\code{invitrouv}] Logical to indicate whether to include in vitro parameters
in uncertainty and variability analysis

\item[\code{calcrb2p}] Logical determining whether or not to recalculate the 
chemical ratio of blood to plasma

\item[\code{censored.params}] The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sub-lists. Each sublist is named for a parameter in
"parameters" and contains two elements: "CV" (coefficient of variation) and
"LOD" (limit of detection, below which parameter values are censored. New
values are sampled with mean equal to the value in "parameters" and standard
deviation equal to the mean times the CV.  Censored values are sampled on a
uniform distribution between 0 and the limit of detection. Not used with
httkpop model.

\item[\code{vary.params}] The parameters listed in vary.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (CV) for the normal distribution. Each
entry in the list is named for a parameter in "parameters". New values are
sampled with mean equal to the value in "parameters" and standard deviation
equal to the mean times the CV. Not used with httkpop model.

\item[\code{return.samples}] Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.

\item[\code{tissue}] Desired steady state tissue conentration.

\item[\code{output.units}] Plasma concentration units, either uM or default mg/L.

\item[\code{solvemodel.arg.list}] Additional arguments ultimately passed to 
\code{\LinkA{solve\_model}{solve.Rul.model}}

\item[\code{invitro.mc.arg.list}] List of additional parameters passed to 
\code{\LinkA{invitro\_mc}{invitro.Rul.mc}}

\item[\code{httkpop.generate.arg.list}] Additional parameters passed to 
\code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.

\item[\code{convert.httkpop.arg.list}] Additional parameters passed to the 
convert\_httkpop\_* function for the model.

\item[\code{parameterize.arg.list}] Additional parameters passed to the 
parameterize\_* function for the model.

\item[\code{return.all.sims}] Logical indicating whether to return the results
of all simulations, in addition to the default toxicokinetic statistics
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
All arguments after httkpop only apply if httkpop is set to TRUE and species
to "Human".

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.

Tissue concentrations are calculated for the pbtk model with oral infusion
dosing.  All tissues other than gut, liver, and lung are the product of the
steady state plasma concentration and the tissue to plasma partition
coefficient.

The six sets of plausible \emph{in vitro-in vivo} extrpolation (IVIVE)
assumptions identified by Honda et al. (2019) are: 
\Tabular{lrrrr}{
& \emph{in vivo} Conc. & Metabolic Clearance & Bioactive Chemical
Conc. & TK Statistic Used* \\{} Honda1 & Veinous (Plasma) &
Restrictive & Free & Mean Conc. \\{} Honda2 & Veinous &
Restrictive & Free & Max Conc. \\{} Honda3 & Veinous &
Non-restrictive & Total & Mean Conc. \\{} Honda4 & Veinous &
Non-restrictive & Total & Max Conc. \\{} Honda5 & Target Tissue &
Non-restrictive & Total & Mean Conc. \\{} Honda6 & Target Tissue
& Non-restrictive & Total & Max Conc. \\{} } *Assumption is
currently ignored because analytical steady-state solutions are currently
used by this function.
\end{Details}
%
\begin{Value}
If return.all.sims == FALSE (default) a list with:
\begin{ldescription}
\item[\code{means}] The mean concentration for each model compartment as a function
of time across the Monte Carlo simulation
\item[\code{sds}] The standard deviation for each model compartment as a function
of time across the Monte Carlo simulation

\end{ldescription}
If return.all.sums == TRUE then a list is returned with:
\begin{ldescription}
\item[\code{stats}] The list of means and sds from return.all.sums=FALSE
\item[\code{sims}] The concentration vs. time results for each compartment for 
every (samples) set of parameters in the Monte Carlo simulation
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}


NSAMP <- 50
chemname="Abamectin"
times<- c(0,0.25,0.5,0.75,1,1.5,2,2.5,3,4,5)
age.ranges <- seq(6,80,by=10)
forward <- NULL
for (age.lower in age.ranges)
{
  label <- paste("Ages ",age.lower,"-",age.lower+4,sep="")
  set.seed(1234)
  forward[[label]] <- calc_mc_tk(
                        chem.name=chemname,
                        samples=NSAMP,
                        httkpop.generate.arg.list=list(
                          method="d",
                          agelim_years = c(age.lower, age.lower+9)),
                        solvemodel.arg.list = list(
                          times=times))
}


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_rblood2plasma}{Calculate the constant ratio of the blood concentration to the plasma concentration.}{calc.Rul.rblood2plasma}
\keyword{Parameter}{calc\_rblood2plasma}
%
\begin{Description}\relax
This function calculates the constant ratio of the blood concentration to
the plasma concentration.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_rblood2plasma(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hematocrit = NULL,
  Krbc2pu = NULL,
  Funbound.plasma = NULL,
  default.to.human = FALSE,
  species = "Human",
  adjusted.Funbound.plasma = TRUE,
  suppress.messages = TRUE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the CAS number or the chemical name must be
specified.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from \code{\LinkA{parameterize\_schmitt}{parameterize.Rul.schmitt}}

\item[\code{hematocrit}] Overwrites default hematocrit value in calculating
Rblood2plasma.

\item[\code{Krbc2pu}] The red blood cell to unbound plasma chemical partition
coefficient, typically from \code{\LinkA{predict\_partitioning\_schmitt}{predict.Rul.partitioning.Rul.schmitt}}

\item[\code{Funbound.plasma}] The fraction of chemical unbound (free) in the
presence of plasma protein

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{adjusted.Funbound.plasma}] Whether or not to use Funbound.plasma
adjustment.

\item[\code{suppress.messages}] Determine whether to display certain usage
feedback.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
The red blood cell (RBC) parition coefficient as predicted by the Schmitt
(2008) method is used in the calculation. The value is calculated with the
equation: 1 - hematocrit + hematocrit * Krbc2pu * Funbound.plasma, summing
the red blood cell to plasma and plasma:plasma (equal to 1) partition
coefficients multiplied by their respective fractional volumes. When
species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data (hematocrit and temperature), but substitues
human fraction unbound and tissue volumes.
\end{Details}
%
\begin{Value}
The blood to plasma chemical concentration ratio
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Schmitt W. "General approach for the calculation of tissue to
plasma partition coefficients." Toxicology In Vitro, 22, 457-467 (2008).

Pearce, Robert G., et al. "Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues." Journal of 
pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.

Ruark, Christopher D., et al. "Predicting passive and active tissue: plasma 
partition coefficients: interindividual and interspecies variability." 
Journal of pharmaceutical sciences 103.7 (2014): 2189-2198.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

calc_rblood2plasma(chem.name="Bisphenol A")
calc_rblood2plasma(chem.name="Bisphenol A",species="Rat")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_stats}{Calculate toxicokinetic summary statistics (deprecated).}{calc.Rul.stats}
\keyword{Solve}{calc\_stats}
\keyword{Statistics}{calc\_stats}
%
\begin{Description}\relax
\#' This function is included for backward compatibility. It calls
\code{\LinkA{calc\_tkstats}{calc.Rul.tkstats}} which 
calculates the area under the curve, the mean, and the peak values
for the venous blood or plasma concentration of a specified chemical or all
chemicals if none is specified for the multiple compartment model with a
given number of days, dose, and number of doses per day.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_stats(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  route = "oral",
  stats = c("AUC", "peak", "mean"),
  species = "Human",
  days = 28,
  daily.dose = 1,
  dose = NULL,
  doses.per.day = 1,
  output.units = "uM",
  concentration = "plasma",
  tissue = "plasma",
  model = "pbtk",
  default.to.human = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = T,
  suppress.messages = FALSE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Name of desired chemical.

\item[\code{chem.cas}] CAS number of desired chemical.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk function,
overrides chem.name and chem.cas.

\item[\code{route}] String specification of route of exposure for simulation:
"oral", "iv", "inhalation", ...

\item[\code{stats}] Desired values (either 'AUC', 'mean', 'peak', or a vector
containing any combination).

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{days}] Length of the simulation.

\item[\code{daily.dose}] Total daily dose, mg/kg BW.

\item[\code{dose}] Amount of a single dose at time zero, mg/kg BW.

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{output.units}] Desired units (either "mg/L", "mg", "umol", or default
"uM").

\item[\code{concentration}] Desired concentration type, 'blood' or default
'plasma'.

\item[\code{tissue}] Desired steady state tissue conentration.

\item[\code{model}] Model used in calculation, 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{suppress.messages}] Whether to suppress output message.

\item[\code{...}] Arguments passed to solve function.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Default value of 0 for doses.per.day solves for a single dose.

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{AUC}] Area under the plasma concentration curve.
\item[\code{mean.conc}] The area under the curve divided by the number of days.
\item[\code{peak.conc}] The highest concentration.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{calc\_tkstats}{Calculate toxicokinetic summary statistics.}{calc.Rul.tkstats}
\keyword{Solve}{calc\_tkstats}
\keyword{Statistics}{calc\_tkstats}
%
\begin{Description}\relax
This function calculates the area under the curve, the mean, and the peak values
for the venous blood or plasma concentration of a specified chemical or all
chemicals if none is specified for the multiple compartment model with a
given number of days, dose, and number of doses per day.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_tkstats(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  route = "oral",
  stats = c("AUC", "peak", "mean"),
  species = "Human",
  days = 28,
  daily.dose = 1,
  dose = NULL,
  doses.per.day = 1,
  output.units = "uM",
  concentration = "plasma",
  tissue = "plasma",
  model = "pbtk",
  default.to.human = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = T,
  suppress.messages = FALSE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Name of desired chemical.

\item[\code{chem.cas}] CAS number of desired chemical.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk function,
overrides chem.name and chem.cas.

\item[\code{route}] String specification of route of exposure for simulation:
"oral", "iv", "inhalation", ...

\item[\code{stats}] Desired values (either 'AUC', 'mean', 'peak', or a vector
containing any combination).

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{days}] Length of the simulation.

\item[\code{daily.dose}] Total daily dose, mg/kg BW.

\item[\code{dose}] Amount of a single dose at time zero, mg/kg BW.

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{output.units}] Desired units (either "mg/L", "mg", "umol", or default
"uM").

\item[\code{concentration}] Desired concentration type, 'blood' or default
'plasma'.

\item[\code{tissue}] Desired steady state tissue conentration.

\item[\code{model}] Model used in calculation, 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{suppress.messages}] Whether to suppress output message.

\item[\code{...}] Arguments passed to solve function.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Default value of 0 for doses.per.day solves for a single dose.

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{AUC}] Area under the plasma concentration curve.
\item[\code{mean.conc}] The area under the curve divided by the number of days.
\item[\code{peak.conc}] The highest concentration.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

calc_tkstats(chem.name='Bisphenol-A',days=100,stats='mean',model='3compartment')


calc_tkstats(chem.name='Bisphenol-A',days=100,stats=c('peak','mean'),species='Rat')

triclosan.stats <- calc_tkstats(days=10, chem.name = "triclosan")


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_total\_clearance}{Calculate the total clearance.}{calc.Rul.total.Rul.clearance}
\keyword{1compartment}{calc\_total\_clearance}
\keyword{Parameter}{calc\_total\_clearance}
%
\begin{Description}\relax
This function calculates the total clearance rate for a one compartment model
where clearance is entirely due to metablism by the liver and glomerular
filtration in the kidneys, identical to clearance of three compartment
steady state model.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_total_clearance(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  suppress.messages = FALSE,
  default.to.human = FALSE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_steadystate
function, overrides chem.name and chem.cas.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.

\item[\code{restrictive.clearance}] Protein binding is not taken into account (set
to 1) in liver clearance if FALSE.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE.

\item[\code{...}] Additional parameters passed to parameterize\_steadystate if
parameters is NULL.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{Total Clearance}] Units of L/h/kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

calc_total_clearance(chem.name="Ibuprofen") 


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calc\_vdist}{Calculate the volume of distribution for a one compartment model.}{calc.Rul.vdist}
\keyword{1compartment}{calc\_vdist}
\keyword{Parameter}{calc\_vdist}
%
\begin{Description}\relax
This function predicts partition coefficients for all tissues, then lumps them
into a single compartment.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calc_vdist(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  default.to.human = FALSE,
  species = "Human",
  suppress.messages = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the CAS number or the chemical name must be specified
when Funbound.plasma is not given in parameter list.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified when Funbound.plasma is not given in parameter list.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from parameterize\_3comp, parameterize\_pbtk or
predict\_partitioning\_schmitt.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with parition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
The effective volume of distribution is calculated by summing each tissues
volume times it's partition coefficient relative to plasma. Plasma, and the
paritioning into RBCs are also added to get the total volume of distribution
in L/KG BW. Partition coefficients are calculated using Schmitt's (2008)
method.  When species is specified as rabbit, dog, or mouse, the function
uses the appropriate physiological data(volumes and flows) but substitues
human fraction unbound, partition coefficients, and intrinsic hepatic
clearance.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{Volume of distribution}] Units of L/ kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Schmitt W. "General approach for the calculation of tissue to
plasma partition coefficients." Toxicology In Vitro, 22, 457-467 (2008).
Peyret, T., Poulin, P., Krishnan, K., "A unified algorithm for predicting
partition coefficients for PBPK modeling of drugs and environmental
chemicals." Toxicology and Applied Pharmacology, 249, 197-207 (2010).
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

calc_vdist(chem.cas="80-05-7")
calc_vdist(chem.name="Bisphenol A")
calc_vdist(chem.name="Bisphenol A",species="Rat")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{chem.invivo.PK.aggregate.data}{Parameter Estimates from Wambaugh et al. (2018)}{chem.invivo.PK.aggregate.data}
\keyword{data}{chem.invivo.PK.aggregate.data}
%
\begin{Description}\relax
This table includes 1 and 2 compartment fits of plasma concentration vs time
data aggregated from chem.invivo.PK.data, performed in Wambaugh et al. 2018.
Data includes volume of distribution (Vdist, L/kg), elimination rate (kelim,
1/h), gut absorption rate (kgutabs, 1/h), fraction absorbed (Fgutabs), and
steady state concentration (Css, mg/L).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
chem.invivo.PK.aggregate.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
data.frame
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. 2018 Toxicological Sciences, in press
\end{Source}
\inputencoding{utf8}
\HeaderA{chem.invivo.PK.data}{Published toxicokinetic time course measurements}{chem.invivo.PK.data}
\keyword{data}{chem.invivo.PK.data}
%
\begin{Description}\relax
This data set includes time and dose specific measurements of chemical
concentration in tissues taken from animals administered control doses of
the chemicals either orally or intravenously. This plasma concentration-time
data is from rat experiments reported in public sources. Toxicokinetic data
were retrieved from those studies by the Netherlands Organisation for
Applied Scientific Research (TNO) using curve stripping (TechDig v2).  This
data is provided for statistical analysis as in Wambaugh et al. 2018.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
chem.invivo.PK.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 597 rows and 13 columns.
\end{Format}
%
\begin{Author}\relax
Sieto Bosgra
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. 2018 Toxicological Sciences, in press
\end{Source}
%
\begin{References}\relax
Aanderud L, Bakke OM (1983). Pharmacokinetics of antipyrine,
paracetamol, and morphine in rat at 71 ATA. Undersea Biomed Res.
10(3):193-201. PMID: 6636344

Aasmoe L, Mathiesen M, Sager G (1999). Elimination of methoxyacetic acid and
ethoxyacetic acid in rat. Xenobiotica. 29(4):417-24. PMID: 10375010

Ako RA. Pharmacokinetics/pharmacodynamics (PK/PD) of oral diethylstilbestrol
(DES) in recurrent prostate cancer patients and of oral dissolving film
(ODF)-DES in rats. PhD dissertation, College of Pharmacy, University of
Houston, USA, 2011.

Anadon A, Martinez-Larranaga MR, Fernandez-Cruz ML, Diaz MJ, Fernandez MC,
Martinez MA (1996). Toxicokinetics of deltamethrin and its 4'-HO-metabolite
in the rat. Toxicol Appl Pharmacol. 141(1):8-16. PMID: 8917670

Binkerd PE, Rowland JM, Nau H, Hendrickx AG (1988). Evaluation of valproic
acid (VPA) developmental toxicity and pharmacokinetics in Sprague-Dawley
rats. Fundam Appl Toxicol. 11(3):485-93. PMID: 3146521

Boralli VB, Coelho EB, Cerqueira PM, Lanchote VL (2005). Stereoselective
analysis of metoprolol and its metabolites in rat plasma with application to
oxidative metabolism. J Chromatogr B Analyt Technol Biomed Life Sci.
823(2):195-202. PMID: 16029965

Chan MP, Morisawa S, Nakayama A, Kawamoto Y, Sugimoto M, Yoneda M (2005).
Toxicokinetics of 14C-endosulfan in male Sprague-Dawley rats following oral
administration of single or repeated doses. Environ Toxicol. 20(5):533-41.
PMID: 16161119

Cruz L, Castaneda-Hernandez G, Flores-Murrieta FJ, Garcia-Lopez P,
Guizar-Sahagun G (2002). Alteration of phenacetin pharmacokinetics after
experimental spinal cord injury. Proc West Pharmacol Soc. 45:4-5. PMID:
12434508

Della Paschoa OE, Mandema JW, Voskuyl RA, Danhof M (1998).
Pharmacokinetic-pharmacodynamic modeling of the anticonvulsant and
electroencephalogram effects of phenytoin in rats. J Pharmacol Exp Ther.
284(2):460-6. PMID: 9454785

Du B, Li X, Yu Q, A Y, Chen C (2010). Pharmacokinetic comparison of orally
disintegrating, beta-cyclodextrin inclusion complex and conventional tablets
of nicardipine in rats. Life Sci J. 7(2):80-4.

Farris FF, Dedrick RL, Allen PV, Smith JC (1993). Physiological model for
the pharmacokinetics of methyl mercury in the growing rat. Toxicol Appl
Pharmacol. 119(1):74-90. PMID: 8470126

Hays SM, Elswick BA, Blumenthal GM, Welsch F, Conolly RB, Gargas ML (2000).
Development of a physiologically based pharmacokinetic model of
2-methoxyethanol and 2-methoxyacetic acid disposition in pregnant rats.
Toxicol Appl Pharmacol. 163(1):67-74. PMID: 10662606

Igari Y, Sugiyama Y, Awazu S, Hanano M (1982). Comparative physiologically
based pharmacokinetics of hexobarbital, phenobarbital and thiopental in the
rat. J Pharmacokinet Biopharm. 10(1):53-75. PMID: 7069578

Ito K, Houston JB (2004). Comparison of the use of liver models for
predicting drug clearance using in vitro kinetic data from hepatic
microsomes and isolated hepatocytes. Pharm Res. 21(5):785-92. PMID: 15180335

Jia L, Wong H, Wang Y, Garza M, Weitman SD (2003). Carbendazim: disposition,
cellular permeability, metabolite identification, and pharmacokinetic
comparison with its nanoparticle. J Pharm Sci. 92(1):161-72. PMID: 12486692

Kawai R, Mathew D, Tanaka C, Rowland M (1998). Physiologically based
pharmacokinetics of cyclosporine A: extension to tissue distribution
kinetics in rats and scale-up to human. J Pharmacol Exp Ther. 287(2):457-68.
PMID: 9808668

Kim YC, Kang HE, Lee MG (2008). Pharmacokinetics of phenytoin and its
metabolite, 4'-HPPH, after intravenous and oral administration of phenytoin
to diabetic rats induced by alloxan or streptozotocin. Biopharm Drug Dispos.
29(1):51-61. PMID: 18022993

Kobayashi S, Takai K, Iga T, Hanano M (1991). Pharmacokinetic analysis of
the disposition of valproate in pregnant rats. Drug Metab Dispos.
19(5):972-6. PMID: 1686245

Kotegawa T, Laurijssens BE, Von Moltke LL, Cotreau MM, Perloff MD,
Venkatakrishnan K, Warrington JS, Granda BW, Harmatz JS, Greenblatt DJ
(2002). In vitro, pharmacokinetic, and pharmacodynamic interactions of
ketoconazole and midazolam in the rat. J Pharmacol Exp Ther. 302(3):1228-37.
PMID: 12183684

Krug AK, Kolde R, Gaspar JA, Rempel E, Balmer NV, Meganathan K, Vojnits K,
Baquie M, Waldmann T, Ensenat-Waser R, Jagtap S, Evans RM, Julien S,
Peterson H, Zagoura D, Kadereit S, Gerhard D, Sotiriadou I, Heke M,
Natarajan K, Henry M, Winkler J, Marchan R, Stoppini L, Bosgra S, Westerhout
J, Verwei M, Vilo J, Kortenkamp A, Hescheler J, Hothorn L, Bremer S, van
Thriel C, Krause KH, Hengstler JG, Rahnenfuhrer J, Leist M, Sachinidis A
(2013). Human embryonic stem cell-derived test systems for developmental
neurotoxicity: a transcriptomics approach. Arch Toxicol. 87(1):123-43. PMID:
23179753

Leon-Reyes MR, Castaneda-Hernandez G, Ortiz MI (2009). Pharmacokinetic of
diclofenac in the presence and absence of glibenclamide in the rat. J Pharm
Pharm Sci. 12(3):280-7. PMID: 20067705

Nagata M, Hidaka M, Sekiya H, Kawano Y, Yamasaki K, Okumura M, Arimori K
(2007). Effects of pomegranate juice on human cytochrome P450 2C9 and
tolbutamide pharmacokinetics in rats. Drug Metab Dispos. 35(2):302-5. PMID:
17132763

Okiyama M, Ueno K, Ohmori S, Igarashi T, Kitagawa H (1988). Drug
interactions between imipramine and benzodiazepines in rats. J Pharm Sci.
77(1):56-63. PMID: 2894451

Pelissier-Alicot AL, Schreiber-Deturmeny E, Simon N, Gantenbein M,
Bruguerolle B (2002). Time-of-day dependent pharmacodynamic and
pharmacokinetic profiles of caffeine in rats. Naunyn Schmiedebergs Arch
Pharmacol. 365(4):318-25. PMID: 11919657

Piersma AH, Bosgra S, van Duursen MB, Hermsen SA, Jonker LR, Kroese ED, van
der Linden SC, Man H, Roelofs MJ, Schulpen SH, Schwarz M, Uibel F, van
Vugt-Lussenburg BM, Westerhout J, Wolterbeek AP, van der Burg B (2013).
Evaluation of an alternative in vitro test battery for detecting
reproductive toxicants. Reprod Toxicol. 38:53-64. PMID: 23511061

Pollack GM, Li RC, Ermer JC, Shen DD (1985). Effects of route of
administration and repetitive dosing on the disposition kinetics of
di(2-ethylhexyl) phthalate and its mono-de-esterified metabolite in rats.
Toxicol Appl Pharmacol. Jun 30;79(2):246-56. PMID: 4002226

Saadeddin A, Torres-Molina F, Carcel-Trullols J, Araico A, Peris JE (2004).
Pharmacokinetics of the time-dependent elimination of all-trans-retinoic
acid in rats. AAPS J. 6(1):1-9. PMID: 18465253

Satterwhite JH, Boudinot FD (1991). Effects of age and dose on the
pharmacokinetics of ibuprofen in the rat. Drug Metab Dispos. 19(1):61-7.
PMID: 1673423

Szymura-Oleksiak J, Panas M, Chrusciel W (1983). Pharmacokinetics of
imipramine after single and multiple intravenous administration in rats. Pol
J Pharmacol Pharm. 35(2):151-7. PMID: 6622297

Tanaka C, Kawai R, Rowland M (2000). Dose-dependent pharmacokinetics of
cyclosporin A in rats: events in tissues. Drug Metab Dispos. 28(5):582-9.
PMID: 10772639

Timchalk C, Nolan RJ, Mendrala AL, Dittenber DA, Brzak KA, Mattsson JL
(2002). A Physiologically based pharmacokinetic and pharmacodynamic
(PBPK/PD) model for the organophosphate insecticide chlorpyrifos in rats and
humans. Toxicol Sci. Mar;66(1):34-53. PMID: 11861971

Tokuma Y, Sekiguchi M, Niwa T, Noguchi H (1988). Pharmacokinetics of
nilvadipine, a new dihydropyridine calcium antagonist, in mice, rats,
rabbits and dogs. Xenobiotica 18(1):21-8. PMID: 3354229

Treiber A, Schneiter R, Delahaye S, Clozel M (2004). Inhibition of organic
anion transporting polypeptide-mediated hepatic uptake is the major
determinant in the pharmacokinetic interaction between bosentan and
cyclosporin A in the rat. J Pharmacol Exp Ther. 308(3):1121-9. PMID:
14617681

Tsui BC, Feng JD, Buckley SJ, Yeung PK (1994). Pharmacokinetics and
metabolism of diltiazem in rats following a single intra-arterial or single
oral dose. Eur J Drug Metab Pharmacokinet. 19(4):369-73. PMID: 7737239

Wambaugh, John F., et al. "Toxicokinetic triage for environmental
chemicals." Toxicological Sciences (2015): 228-237.

Wang Y, Roy A, Sun L, Lau CE (1999). A double-peak phenomenon in the
pharmacokinetics of alprazolam after oral administration. Drug Metab Dispos.
27(8):855-9. PMID: 10421610

Wang X, Lee WY, Or PM, Yeung JH (2010). Pharmacokinetic interaction studies
of tanshinones with tolbutamide, a model CYP2C11 probe substrate, using
liver microsomes, primary hepatocytes and in vivo in the rat. Phytomedicine.
17(3-4):203-11. PMID: 19679455

Yang SH, Lee MG (2008). Dose-independent pharmacokinetics of ondansetron in
rats: contribution of hepatic and intestinal first-pass effects to low
bioavailability. Biopharm Drug Dispos. 29(7):414-26. PMID: 18697186

Yeung PK, Alcos A, Tang J (2009). Pharmacokinetics and Hemodynamic Effects
of Diltiazem in Rats Following Single vs Multiple Doses In Vivo. Open Drug
Metab J. 3:56-62.
\end{References}
\inputencoding{utf8}
\HeaderA{chem.invivo.PK.summary.data}{Summary of published toxicokinetic time course experiments}{chem.invivo.PK.summary.data}
\keyword{data}{chem.invivo.PK.summary.data}
%
\begin{Description}\relax
This data set summarizes the time course data in the chem.invivo.PK.data
table. Maximum concentration (Cmax), time integrated plasma concentration
for the duration of treatment (AUC.treatment) and extrapolated to zero
concentration (AUC.infinity) as well as half-life are calculated. Summary
values are given for each study and dosage. These data can be used to
evaluate toxicokinetic model predictions.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
chem.invivo.PK.summary.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 100 rows and 25 columns.
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. 2018 Toxicological Sciences, in press
\end{Source}
%
\begin{References}\relax
Aanderud L, Bakke OM (1983). Pharmacokinetics of antipyrine,
paracetamol, and morphine in rat at 71 ATA. Undersea Biomed Res.
10(3):193-201. PMID: 6636344

Aasmoe L, Mathiesen M, Sager G (1999). Elimination of methoxyacetic acid and
ethoxyacetic acid in rat. Xenobiotica. 29(4):417-24. PMID: 10375010

Ako RA. Pharmacokinetics/pharmacodynamics (PK/PD) of oral diethylstilbestrol
(DES) in recurrent prostate cancer patients and of oral dissolving film
(ODF)-DES in rats. PhD dissertation, College of Pharmacy, University of
Houston, USA, 2011.

Anadon A, Martinez-Larranaga MR, Fernandez-Cruz ML, Diaz MJ, Fernandez MC,
Martinez MA (1996). Toxicokinetics of deltamethrin and its 4'-HO-metabolite
in the rat. Toxicol Appl Pharmacol. 141(1):8-16. PMID: 8917670

Binkerd PE, Rowland JM, Nau H, Hendrickx AG (1988). Evaluation of valproic
acid (VPA) developmental toxicity and pharmacokinetics in Sprague-Dawley
rats. Fundam Appl Toxicol. 11(3):485-93. PMID: 3146521

Boralli VB, Coelho EB, Cerqueira PM, Lanchote VL (2005). Stereoselective
analysis of metoprolol and its metabolites in rat plasma with application to
oxidative metabolism. J Chromatogr B Analyt Technol Biomed Life Sci.
823(2):195-202. PMID: 16029965

Chan MP, Morisawa S, Nakayama A, Kawamoto Y, Sugimoto M, Yoneda M (2005).
Toxicokinetics of 14C-endosulfan in male Sprague-Dawley rats following oral
administration of single or repeated doses. Environ Toxicol. 20(5):533-41.
PMID: 16161119

Cruz L, Castaneda-Hernandez G, Flores-Murrieta FJ, Garcia-Lopez P,
Guizar-Sahagun G (2002). Alteration of phenacetin pharmacokinetics after
experimental spinal cord injury. Proc West Pharmacol Soc. 45:4-5. PMID:
12434508

Della Paschoa OE, Mandema JW, Voskuyl RA, Danhof M (1998).
Pharmacokinetic-pharmacodynamic modeling of the anticonvulsant and
electroencephalogram effects of phenytoin in rats. J Pharmacol Exp Ther.
284(2):460-6. PMID: 9454785

Du B, Li X, Yu Q, A Y, Chen C (2010). Pharmacokinetic comparison of orally
disintegrating, beta-cyclodextrin inclusion complex and conventional tablets
of nicardipine in rats. Life Sci J. 7(2):80-4.

Farris FF, Dedrick RL, Allen PV, Smith JC (1993). Physiological model for
the pharmacokinetics of methyl mercury in the growing rat. Toxicol Appl
Pharmacol. 119(1):74-90. PMID: 8470126

Hays SM, Elswick BA, Blumenthal GM, Welsch F, Conolly RB, Gargas ML (2000).
Development of a physiologically based pharmacokinetic model of
2-methoxyethanol and 2-methoxyacetic acid disposition in pregnant rats.
Toxicol Appl Pharmacol. 163(1):67-74. PMID: 10662606

Igari Y, Sugiyama Y, Awazu S, Hanano M (1982). Comparative physiologically
based pharmacokinetics of hexobarbital, phenobarbital and thiopental in the
rat. J Pharmacokinet Biopharm. 10(1):53-75. PMID: 7069578

Ito K, Houston JB (2004). Comparison of the use of liver models for
predicting drug clearance using in vitro kinetic data from hepatic
microsomes and isolated hepatocytes. Pharm Res. 21(5):785-92. PMID: 15180335

Jia L, Wong H, Wang Y, Garza M, Weitman SD (2003). Carbendazim: disposition,
cellular permeability, metabolite identification, and pharmacokinetic
comparison with its nanoparticle. J Pharm Sci. 92(1):161-72. PMID: 12486692

Kawai R, Mathew D, Tanaka C, Rowland M (1998). Physiologically based
pharmacokinetics of cyclosporine A: extension to tissue distribution
kinetics in rats and scale-up to human. J Pharmacol Exp Ther. 287(2):457-68.
PMID: 9808668

Kim YC, Kang HE, Lee MG (2008). Pharmacokinetics of phenytoin and its
metabolite, 4'-HPPH, after intravenous and oral administration of phenytoin
to diabetic rats induced by alloxan or streptozotocin. Biopharm Drug Dispos.
29(1):51-61. PMID: 18022993

Kobayashi S, Takai K, Iga T, Hanano M (1991). Pharmacokinetic analysis of
the disposition of valproate in pregnant rats. Drug Metab Dispos.
19(5):972-6. PMID: 1686245

Kotegawa T, Laurijssens BE, Von Moltke LL, Cotreau MM, Perloff MD,
Venkatakrishnan K, Warrington JS, Granda BW, Harmatz JS, Greenblatt DJ
(2002). In vitro, pharmacokinetic, and pharmacodynamic interactions of
ketoconazole and midazolam in the rat. J Pharmacol Exp Ther. 302(3):1228-37.
PMID: 12183684

Krug AK, Kolde R, Gaspar JA, Rempel E, Balmer NV, Meganathan K, Vojnits K,
Baquie M, Waldmann T, Ensenat-Waser R, Jagtap S, Evans RM, Julien S,
Peterson H, Zagoura D, Kadereit S, Gerhard D, Sotiriadou I, Heke M,
Natarajan K, Henry M, Winkler J, Marchan R, Stoppini L, Bosgra S, Westerhout
J, Verwei M, Vilo J, Kortenkamp A, Hescheler J, Hothorn L, Bremer S, van
Thriel C, Krause KH, Hengstler JG, Rahnenfuhrer J, Leist M, Sachinidis A
(2013). Human embryonic stem cell-derived test systems for developmental
neurotoxicity: a transcriptomics approach. Arch Toxicol. 87(1):123-43. PMID:
23179753

Leon-Reyes MR, Castaneda-Hernandez G, Ortiz MI (2009). Pharmacokinetic of
diclofenac in the presence and absence of glibenclamide in the rat. J Pharm
Pharm Sci. 12(3):280-7. PMID: 20067705

Nagata M, Hidaka M, Sekiya H, Kawano Y, Yamasaki K, Okumura M, Arimori K
(2007). Effects of pomegranate juice on human cytochrome P450 2C9 and
tolbutamide pharmacokinetics in rats. Drug Metab Dispos. 35(2):302-5. PMID:
17132763

Okiyama M, Ueno K, Ohmori S, Igarashi T, Kitagawa H (1988). Drug
interactions between imipramine and benzodiazepines in rats. J Pharm Sci.
77(1):56-63. PMID: 2894451

Pelissier-Alicot AL, Schreiber-Deturmeny E, Simon N, Gantenbein M,
Bruguerolle B (2002). Time-of-day dependent pharmacodynamic and
pharmacokinetic profiles of caffeine in rats. Naunyn Schmiedebergs Arch
Pharmacol. 365(4):318-25. PMID: 11919657

Piersma AH, Bosgra S, van Duursen MB, Hermsen SA, Jonker LR, Kroese ED, van
der Linden SC, Man H, Roelofs MJ, Schulpen SH, Schwarz M, Uibel F, van
Vugt-Lussenburg BM, Westerhout J, Wolterbeek AP, van der Burg B (2013).
Evaluation of an alternative in vitro test battery for detecting
reproductive toxicants. Reprod Toxicol. 38:53-64. PMID: 23511061

Pollack GM, Li RC, Ermer JC, Shen DD (1985). Effects of route of
administration and repetitive dosing on the disposition kinetics of
di(2-ethylhexyl) phthalate and its mono-de-esterified metabolite in rats.
Toxicol Appl Pharmacol. Jun 30;79(2):246-56. PMID: 4002226

Saadeddin A, Torres-Molina F, Carcel-Trullols J, Araico A, Peris JE (2004).
Pharmacokinetics of the time-dependent elimination of all-trans-retinoic
acid in rats. AAPS J. 6(1):1-9. PMID: 18465253

Satterwhite JH, Boudinot FD (1991). Effects of age and dose on the
pharmacokinetics of ibuprofen in the rat. Drug Metab Dispos. 19(1):61-7.
PMID: 1673423

Szymura-Oleksiak J, Panas M, Chrusciel W (1983). Pharmacokinetics of
imipramine after single and multiple intravenous administration in rats. Pol
J Pharmacol Pharm. 35(2):151-7. PMID: 6622297

Tanaka C, Kawai R, Rowland M (2000). Dose-dependent pharmacokinetics of
cyclosporin A in rats: events in tissues. Drug Metab Dispos. 28(5):582-9.
PMID: 10772639

Timchalk C, Nolan RJ, Mendrala AL, Dittenber DA, Brzak KA, Mattsson JL
(2002). A Physiologically based pharmacokinetic and pharmacodynamic
(PBPK/PD) model for the organophosphate insecticide chlorpyrifos in rats and
humans. Toxicol Sci. Mar;66(1):34-53. PMID: 11861971

Tokuma Y, Sekiguchi M, Niwa T, Noguchi H (1988). Pharmacokinetics of
nilvadipine, a new dihydropyridine calcium antagonist, in mice, rats,
rabbits and dogs. Xenobiotica 18(1):21-8. PMID: 3354229

Treiber A, Schneiter R, Delahaye S, Clozel M (2004). Inhibition of organic
anion transporting polypeptide-mediated hepatic uptake is the major
determinant in the pharmacokinetic interaction between bosentan and
cyclosporin A in the rat. J Pharmacol Exp Ther. 308(3):1121-9. PMID:
14617681

Tsui BC, Feng JD, Buckley SJ, Yeung PK (1994). Pharmacokinetics and
metabolism of diltiazem in rats following a single intra-arterial or single
oral dose. Eur J Drug Metab Pharmacokinet. 19(4):369-73. PMID: 7737239

Wambaugh, John F., et al. "Toxicokinetic triage for environmental
chemicals." Toxicological Sciences (2015): 228-237.

Wang Y, Roy A, Sun L, Lau CE (1999). A double-peak phenomenon in the
pharmacokinetics of alprazolam after oral administration. Drug Metab Dispos.
27(8):855-9. PMID: 10421610

Wang X, Lee WY, Or PM, Yeung JH (2010). Pharmacokinetic interaction studies
of tanshinones with tolbutamide, a model CYP2C11 probe substrate, using
liver microsomes, primary hepatocytes and in vivo in the rat. Phytomedicine.
17(3-4):203-11. PMID: 19679455

Yang SH, Lee MG (2008). Dose-independent pharmacokinetics of ondansetron in
rats: contribution of hepatic and intestinal first-pass effects to low
bioavailability. Biopharm Drug Dispos. 29(7):414-26. PMID: 18697186

Yeung PK, Alcos A, Tang J (2009). Pharmacokinetics and Hemodynamic Effects
of Diltiazem in Rats Following Single vs Multiple Doses In Vivo. Open Drug
Metab J. 3:56-62.
\end{References}
\inputencoding{utf8}
\HeaderA{chem.lists}{Chemical membership in different research projects}{chem.lists}
\keyword{data}{chem.lists}
%
\begin{Description}\relax
A static list of lists identifying chemical membership in different research
projects. While it is our intent to keep these lists up-to-date, the
information here is only for convenience and should not be considered to be
definitive.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
chem.lists
\end{verbatim}
\end{Usage}
%
\begin{Format}
A list containing ten lists.
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Bucher, J. R. (2008). Guest Editorial: NTP: New Initiatives, New
Alignment. Environ Health Perspect 116(1).

Judson, R. S., Houck, K. A., Kavlock, R. J., Knudsen, T. B., Martin, M. T.,
Mortensen, H. M., Reif, D. M., Rotroff, D. M., Shah, I., Richard, A. M. and
Dix, D. J. (2010). In Vitro Screening of Environmental Chemicals for
Targeted Testing Prioritization: The ToxCast Project. Environmental Health
Perspectives 118(4), 485-492.

Wambaugh, J. F., Wang, A., Dionisio, K. L., Frame, A., Egeghy, P., Judson,
R. and Setzer, R. W. (2014). High Throughput Heuristics for Prioritizing
Human Exposure to Environmental Chemicals. Environmental Science \&
Technology, 10.1021/es503583j.

CDC (2014). National Health and Nutrition Examination Survey. Available at:
https://www.cdc.gov/nchs/nhanes.htm.
\end{References}
\inputencoding{utf8}
\HeaderA{chem.physical\_and\_invitro.data}{Physico-chemical properties and in vitro measurements for toxicokinetics}{chem.physical.Rul.and.Rul.invitro.data}
\keyword{data}{chem.physical\_and\_invitro.data}
%
\begin{Description}\relax
This data set contains the necessary information to make basic,
high-throughput toxicokinetic (HTTK) predictions for compounds, including
Funbound.plasma, molecular weight (g/mol), logP, logMA (membrane affinity),
intrinsic clearance(uL/min/10\textasciicircum{}6 cells), and pKa. These data have been
compiled from multiple sources, and can be used to parameterize a variety of
toxicokinetic models. See variable EPA.ref for information on the reference EPA.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
chem.physical_and_invitro.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 9411 rows and 54 columns.

\Tabular{lll}{
\strong{Column Name} & \strong{Description} & \strong{Units} \\{}
Compound & The preferred name of the chemical compound & none \\{}                      
CAS& The preferred Chemical Abstracts Service Registry Number & none \\{}                     
CAS.Checksum & A logical indicating whether the CAS number is valid & none \\{}                   
DTXSID & DSSTox Structure ID 
(\url{http://comptox.epa.gov/dashboard}) & none \\{}                  
Formula & The proportions of atoms within the chemical compound  & none \\{}                   
SMILES.desalt & The simplified molecular-input line-entry system
structure & none \\{}                 
All.Compound.Names & All names of the chemical as they occured in the
data & none \\{}              
logHenry & The log10 Henry's law constant & 
log10(atmosphers*m\textasciicircum{}3/mole) \\{}                 
logHenry.Reference & Reference for Henry's law constant & \\{}           
logP & The log10 octanol:water partition coefficient (PC)& log10 unitless ratio \\{}                 
logP.Reference & Reference for logPow & \\{}               
logPwa & The log10 water:air PC & log10 unitless ratio \\{}                 
logPwa.Reference & Reference for logPwa & \\{}             
logMA & The log10 phospholipid:water PC or
"Membrane affinity" & unitless ratio \\{}                
logMA.Reference & Reference for membrane affinity & \\{}   \#'  logWSol & The log10 water solubility & log10(mole/L) \\{}                  
logWSol.Reference & Reference for logWsol & \\{}              
MP & The chemical compound melting point & degrees Celsius \\{}                  
MP.Reference & Reference for melting point & \\{}                   
MW & The chemical compound molecular weight & g/mol \\{}                
MW.Reference & Reference for molecular weight & \\{}                 
pKa\_Accept & The hydrogen acceptor equilibria concentrations 
& logarithm \\{}              
pKa\_Accept.Reference & Reference for pKa\_Accept & \\{}           
pKa\_Donor & The hydrogen acceptor equilibria concentrations 
& logarithm \\{}               
pKa\_Donor.Reference & Reference for pKa\_Donor & \\{}             
All.Species & All species for which data were available & none \\{}                
DTXSID.Reference & Reference for DTXSID & \\{}               
Formula.Reference & Reference for chemical formulat & \\{}             
[SPECIES].Clint & (Primary hepatocyte suspension) 
intrinsic hepatic clearance & uL/min/10\textasciicircum{}6 hepatocytes \\{}                   
[SPECIES].Clint.pValue & Probability that there is no clearance observed. & none \\{}           
[SPECIES].Clint.pValue.Ref & Reference for Clint pValue &  \\{}   
[SPECIES].Clint.Reference & Reference for Clint &  \\{}         
[SPECIES].Fgutabs & Fraction of chemical absorbed from the
gut & unitless fraction \\{}           
[SPECIES].Fgutabs.Reference & Reference for Fgutabs & \\{}        
[SPECIES].Funbound.plasma & Chemical fraction unbound in presence of 
plasma proteins & unitless fraction \\{}         
[SPECIES].Funbound.plasma.Ref& Reference for Funbound.plasma & \\{} 
[SPECIES].Rblood2plasma & Chemical concentration blood to plasma ratio & unitless ratio \\{}         
[SPECIES].Rblood2plasma.Ref & Reference for Rblood2plasma &  \\{}  
SMILES.desalt.Reference"& Reference for SMILES structure &  \\{}          
Chemical.Class & All classes to which the chemical has been assigned & \\{}
}
\end{Format}
%
\begin{Details}\relax
In some cases the rapid equilbrium dailysis method (Waters et al., 2008)
fails to yield detectable concentrations for the free fraction of chemical. 
In those cases we assume the compound is highly bound (that is, Fup approaches
zero). For some calculations (for example, steady-state plasma concentration)
there is precendent (Rotroff et al., 2010) for using half the average limit 
of detection, that is 0.005. We do not recomend using other models where 
quantities like partition coefficients must be predicted using Fup. We also
do not recomend including the value 0.005 in training sets for Fup predictive
models. 

\strong{Note} that in some cases the \strong{Funbound.plasma} and the 
\strong{intrinsic clearance} are
\emph{provided as a series of numbers separated by commas}. These values are the 
result of Bayesian analysis and characterize a distribution: the first value
is the median of the distribution, while the second and third values are the 
lower and upper 95th percentile (that is qunatile 2.5 and 97.5) respectively.
For intrinsic clearance a fourth value indicating a p-value for a decrease is
provided. Typically 4000 samples were used for the Bayesian analusis, such
that a p-value of "0" is equivale to "<0.00025". See Wambaugh et al. (2019)
for more details.

Any one chemical compound \emph{may have multiple ionization equilibria} 
(see Strope et al., 2018) may both for donating or accepting a proton (and
therefore changing charge state). If there are multiple equlibria of the same
type (donor/accept])the are concatonated by commas.

All species-specific information is initially from experimental measurements.
The functions \code{\LinkA{load\_sipes2017}{load.Rul.sipes2017}}, \code{\LinkA{load\_pradeep2020}{load.Rul.pradeep2020}}, 
and \code{\LinkA{load\_dawson2021}{load.Rul.dawson2021}} may be used to add in silico, structure-based
predictions for many thousands of additional compounds to this table.
\end{Details}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh, John F., et al. "Toxicokinetic triage for environmental
chemicals." Toxicological Sciences (2015): 228-237.
\end{Source}
%
\begin{References}\relax
CompTox Chemicals Dashboard (\url{http://comptox.epa.gov/dashboard})

EPI Suite, https://www.epa.gov/opptintr/exposure/pubs/episuite.htm

Hilal, S., Karickhoff, S. and Carreira, L. (1995). A rigorous test for
SPARC's chemical reactivity models: Estimation of more than 4300 ionization
pKas. Quantitative Structure-Activity Relationships 14(4), 348-355.

Honda, G. S., Pearce, R. G., Pham, L. L., Setzer, R. W., Wetmore, B. A., 
Sipes, N. S., ... \& Wambaugh, J. F. (2019). Using the concordance of in 
vitro and in vivo data to evaluate extrapolation assumptions. PloS one, 
14(5), e0217564.

Ito, K. and Houston, J. B. (2004). Comparison of the use of liver models for
predicting drug clearance using in vitro kinetic data from hepatic
microsomes and isolated hepatocytes. Pharm Res 21(5), 785-92.

Jones, O. A., Voulvoulis, N. and Lester, J. N. (2002). Aquatic environmental
assessment of the top 25 English prescription pharmaceuticals. Water
research 36(20), 5013-22.

Lau, Y. Y., Sapidou, E., Cui, X., White, R. E. and Cheng, K. C. (2002).
Development of a novel in vitro model to predict hepatic clearance using
fresh, cryopreserved, and sandwich-cultured hepatocytes. Drug Metabolism and
Disposition 30(12), 1446-54.

Linakis, M. W., Sayre, R. R., Pearce, R. G., Sfeir, M. A., Sipes, N. S., 
Pangburn, H. A., ... \& Wambaugh, J. F. (2020). Development and evaluation of 
a high throughput inhalation model for organic chemicals. Journal of 
Exposure Science \& Environmental Epidemiology, 1-12.

Lombardo, F., Berellini, G., \& Obach, R. S. (2018). Trend analysis of a 
database of intravenous pharmacokinetic parameters in humans for 1352 drug 
compounds. Drug Metabolism and Disposition, 46(11), 1466-1477.

McGinnity, D. F., Soars, M. G., Urbanowicz, R. A. and Riley, R. J. (2004).
Evaluation of fresh and cryopreserved hepatocytes as in vitro drug
metabolism tools for the prediction of metabolic clearance. Drug Metabolism
and Disposition 32(11), 1247-53, 10.1124/dmd.104.000026.

Naritomi, Y., Terashita, S., Kagayama, A. and Sugiyama, Y. (2003). Utility
of Hepatocytes in Predicting Drug Metabolism: Comparison of Hepatic
Intrinsic Clearance in Rats and Humans in Vivo and in Vitro. Drug Metabolism
and Disposition 31(5), 580-588, 10.1124/dmd.31.5.580.

Obach, R. S. (1999). Prediction of human clearance of twenty-nine drugs from
hepatic microsomal intrinsic clearance data: An examination of in vitro
half-life approach and nonspecific binding to microsomes. Drug Metabolism
and Disposition 27(11), 1350-9.

Paini, Alicia; Cole, Thomas; Meinero, Maria; Carpi, Donatella; Deceuninck, 
Pierre; Macko, Peter; Palosaari, Taina; Sund, Jukka; Worth, Andrew; Whelan, 
Maurice (2020):  EURL ECVAM in vitro hepatocyte clearance and blood plasma 
protein binding dataset for 77 chemicals. European Commission, Joint Research 
Centre (JRC) [Dataset] PID: https://data.europa.eu/89h/a2ff867f-db80-4acf-8e5c-e45502713bee

Paixao, P., Gouveia, L. F., \& Morais, J. A. (2012). Prediction of the human
oral bioavailability by using in vitro and in silico drug related parameters
in a physiologically based absorption model. International journal of
pharmaceutics, 429(1), 84-98.

Pirovano, Alessandra, et al. "QSARs for estimating intrinsic hepatic
clearance of organic chemicals in humans." Environmental toxicology and
pharmacology 42 (2016): 190-197.

Schmitt, W. (2008). General approach for the calculation of tissue to plasma
partition coefficients. Toxicology in vitro : an international journal
published in association with BIBRA 22(2), 457-67,
10.1016/j.tiv.2007.09.010.

Shibata, Y., Takahashi, H., Chiba, M. and Ishii, Y. (2002). Prediction of
Hepatic Clearance and Availability by Cryopreserved Human Hepatocytes: An
Application of Serum Incubation Method. Drug Metabolism and Disposition
30(8), 892-896, 10.1124/dmd.30.8.892.

Tonnelier, A., Coecke, S. and Zaldivar, J.-M. (2012). Screening of chemicals
for human bioaccumulative potential with a physiologically based
toxicokinetic model. Archives of Toxicology 86(3), 393-403,
10.1007/s00204-011-0768-0.

Uchimura, Takahide, et al. "Prediction of human blood-to-plasma drug
concentration ratio." Biopharmaceutics \& drug disposition 31.5-6 (2010):
286-297.

Wambaugh, J. F., Wetmore, B. A., Ring, C. L., Nicolas, C. I., Pearce, R. G., 
Honda, G. S., ... \& Badrinarayanan, A. (2019). Assessing Toxicokinetic 
Uncertainty and Variability in Risk Prioritization. Toxicological Sciences, 
172(2), 235-251.

Wetmore, B. A., Wambaugh, J. F., Ferguson, S. S., Sochaski, M. A., Rotroff,
D. M., Freeman, K., Clewell, H. J., 3rd, Dix, D. J., Andersen, M. E., Houck,
K. A., Allen, B., Judson, R. S., Singh, R., Kavlock, R. J., Richard, A. M.
and Thomas, R. S. (2012). Integration of dosimetry, exposure, and
high-throughput screening data in chemical toxicity assessment.
Toxicological sciences : an official journal of the Society of Toxicology
125(1), 157-74, 10.1093/toxsci/kfr254.

Wetmore, B. A., Wambaugh, J. F., Ferguson, S. S., Li, L., Clewell, H. J.,
Judson, R. S., Freeman, K., Bao, W., Sochaski, M. A., Chu, T.-M., Black, M.
B., Healy, E., Allen, B., Andersen, M. E., Wolfinger, R. D. and Thomas, R.
S. (2013). Relative Impact of Incorporating Pharmacokinetics on Predicting
In Vivo Hazard and Mode of Action from High-Throughput In Vitro Toxicity
Assays. Toxicological Sciences 132(2), 327-346, 10.1093/toxsci/kft012.

Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
"Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing" Toxicological
Sciences, kfv171.
\end{References}
\inputencoding{utf8}
\HeaderA{ckd\_epi\_eq}{CKD-EPI equation for GFR.}{ckd.Rul.epi.Rul.eq}
\keyword{httk-pop}{ckd\_epi\_eq}
%
\begin{Description}\relax
Predict GFR from serum creatinine, gender, and age.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
ckd_epi_eq(scr, gender, reth, age_years, ckd_epi_race_coeff = FALSE)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{scr}] Vector of serum creatinine values in mg/dL.

\item[\code{gender}] Vector of genders (either 'Male' or 'Female').

\item[\code{reth}] Vector of races/ethnicities. Not used unless ckd\_epi\_race\_coeff is TRUE.

\item[\code{age\_years}] Vector of ages in years.

\item[\code{ckd\_epi\_race\_coeff}] Whether to use the "race coefficient" in the CKD-EPI equation. Default is FALSE.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
From Levey AS, Stevens LA, Schmid CH, Zhang YL, Castro AF, Feldman HI, et al. A new
equation to estimate glomerular filtration rate. Ann Intern Med 2009;
150(9):604-612. doi:10.7326/0003-4819-150-9-200905050-00006
\end{Details}
%
\begin{Value}
Vector of GFR values in mL/min/1.73m\textasciicircum{}2.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{concentration\_data\_Linakis2020}{Concentration data involved in Linakis 2020 vignette analysis.}{concentration.Rul.data.Rul.Linakis2020}
\keyword{data}{concentration\_data\_Linakis2020}
%
\begin{Description}\relax
Concentration data involved in Linakis 2020 vignette analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
concentration_data_Linakis2020
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing x rows and y columns.
\end{Format}
%
\begin{Author}\relax
Matt Linakis
\end{Author}
%
\begin{Source}\relax
Matt Linakis
\end{Source}
%
\begin{References}\relax
DSStox database (https:// www.epa.gov/ncct/dsstox
\end{References}
\inputencoding{utf8}
\HeaderA{convert\_httkpop\_1comp}{Converts HTTK-Pop physiology into parameters relevant to the one compartment model}{convert.Rul.httkpop.Rul.1comp}
\keyword{1compartment}{convert\_httkpop\_1comp}
\keyword{httk-pop}{convert\_httkpop\_1comp}
%
\begin{Description}\relax
Converts HTTK-Pop physiology into parameters relevant to the one
compartment model
\end{Description}
%
\begin{Usage}
\begin{verbatim}
convert_httkpop_1comp(parameters.dt, httkpop.dt, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{parameters.dt}] Data table returned by \code{\LinkA{create\_mc\_samples}{create.Rul.mc.Rul.samples}}

\item[\code{httkpop.dt}] Data table returned by \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}

\item[\code{...}] Additional arguments passed to \code{\LinkA{propagate\_invitrouv\_1comp}{propagate.Rul.invitrouv.Rul.1comp}}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table whose columns are the parameters of the HTTK model
specified in \code{model}.
\end{Value}
%
\begin{Author}\relax
Caroline Ring, John Wambaugh, and Greg Honda
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{convert\_units}{convert\_units}{convert.Rul.units}
%
\begin{Description}\relax
This function is designed to accept input units, output units, and the 
molecular weight (MW) of a substance of interest to then use a table lookup
to return a scaling factor that can be readily applied for the intended
conversion. It can also take chemical identifiers in the place of a 
specified molecular weight value to retrieve that value for its own use.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
convert_units(
  input.units = NULL,
  output.units = NULL,
  MW = NULL,
  vol = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  temp = 25,
  state = "liquid"
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{input.units}] Assigned input units of interest

\item[\code{output.units}] Desired output units

\item[\code{MW}] Molecular weight of substance of interest in g/mole

\item[\code{vol}] Volume for the target tissue of interest in liters (L).
NOTE: Volume should not be in units of per BW, i.e. "kg".

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID 
(\url{http://comptox.epa.gov/dashboard}) the chemical must be identified by
either CAS, name, or DTXSIDs

\item[\code{parameters}] A set of model parameters, especially a set that
includes MW (molecular weight) for our conversions

\item[\code{temp}] Temperature for conversions (default = 25 degreees C)

\item[\code{state}] Chemical state (gas or default liquid)
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
If input or output units not contained in the table are queried,
it gives a corresponding error message. It gives a warning message about the
handling of 'ppmv,' as the function is only set up to convert between ppmv 
and mass-based units (like 
mg/\eqn{m^3}{} or umol/L) 
in the context of ideal gases.

convert\_units is not directly configured to accept and convert units based
on BW, like mg/kg. For this purpose, see \code{\LinkA{scale\_dosing}{scale.Rul.dosing}}.

The function supports a limited set of most relevant units across
toxicological models, currently including umol, uM, mg, mg/L, 
mg/\eqn{m^3}{} or umol/L), and
in the context of gases assumed to be ideal, ppmv. 

\emph{Andersen and Clewell's Rules of PBPK Modeling:}
\begin{itemize}

\item{} 1Check Your Units
\item{} 2\strong{Check Your Units}
\item{} 3Check Mass Balance

\end{itemize}

\end{Details}
%
\begin{Author}\relax
Mark Sfeir, John Wambaugh, and Sarah E. Davidson
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

# MW BPA is 228.29 g/mol
# 1 mg/L -> 1/228.29*1000 = 4.38 uM
convert_units("mg/L","uM",chem.cas="80-05-7")

# MW Diclofenac is 296.148 g/mol
# 1 uM -> 296.148/1000 =  0.296
convert_units("uM","mg/L",chem.name="diclofenac")

convert_units("uM","ppmv",chem.name="styrene")

# Compare with https://www3.epa.gov/ceampubl/learn2model/part-two/onsite/ia_unit_conversion.html
# 1 ug/L Toluene -> 0.263 ppmv
convert_units("ug/L","ppmv",chem.name="toluene")
# 1 pppmv Toluene, 0.0038 mg/L
convert_units("ppmv","mg/L",chem.name="toluene")

MW_pyrene <- get_physchem_param(param='MW', chem.name='pyrene')
conversion_factor <- convert_units(input.units='mg/L', output.units ='uM',
  MW=MW_pyrene)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{create\_mc\_samples}{Create a data table of draws of parameter values for Monte Carlo}{create.Rul.mc.Rul.samples}
\keyword{Monte-Carlo}{create\_mc\_samples}
%
\begin{Description}\relax
This function creates a data table of draws of parameter values for use with 
Monte Carlo methods
\end{Description}
%
\begin{Usage}
\begin{verbatim}
create_mc_samples(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  samples = 1000,
  species = "Human",
  suppress.messages = FALSE,
  model = "3compartmentss",
  httkpop = TRUE,
  invitrouv = TRUE,
  calcrb2p = TRUE,
  censored.params = list(),
  vary.params = list(),
  return.samples = FALSE,
  tissue = NULL,
  httkpop.dt = NULL,
  invitro.mc.arg.list = list(adjusted.Funbound.plasma = TRUE, poormetab = TRUE,
    fup.censored.dist = FALSE, fup.lod = 0.01, fup.meas.cv = 0.4, clint.meas.cv = 0.3,
    fup.pop.cv = 0.3, clint.pop.cv = 0.3),
  httkpop.generate.arg.list = list(method = "direct resampling", gendernum = NULL,
    agelim_years = NULL, agelim_months = NULL, weight_category = c("Underweight",
    "Normal", "Overweight", "Obese"), gfr_category = c("Normal", "Kidney Disease",
    "Kidney Failure"), reths = c("Mexican American", "Other Hispanic",
    "Non-Hispanic White", "Non-Hispanic Black", "Other")),
  convert.httkpop.arg.list = list(),
  propagate.invitrouv.arg.list = list(),
  parameterize.arg.list = list(restrictive.clearance = T, default.to.human = FALSE,
    clint.pvalue.threshold = 0.05, regression = TRUE)
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs

\item[\code{parameters}] Parameters from the appropriate parameterization function
for the model indicated by argument model

\item[\code{samples}] Number of samples generated in calculating quantiles.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human"). Species must be set to "Human" to run httkpop model.

\item[\code{suppress.messages}] Whether or not to suppress output message.

\item[\code{model}] Model used in calculation: 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.  This only applies when httkpop=TRUE and species="Human",
otherwise '3compartmentss' is used.

\item[\code{httkpop}] Whether or not to use the Ring et al. (2017) "httkpop"
population generator. Species must be 'Human'.

\item[\code{invitrouv}] Logical to indicate whether to include in vitro parameters
such as intrinsic hepatic clearance rate and fraction unbound in plasma
in uncertainty and variability analysis

\item[\code{calcrb2p}] Logical determining whether or not to recalculate the 
chemical ratio of blood to plasma

\item[\code{censored.params}] The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sub-lists. Each sublist is named for a parameter in
"parameters" and contains two elements: "CV" (coefficient of variation) and
"LOD" (limit of detection, below which parameter values are censored. New
values are sampled with mean equal to the value in "parameters" and standard
deviation equal to the mean times the CV.  Censored values are sampled on a
uniform distribution between 0 and the limit of detection. Not used with
httkpop model.

\item[\code{vary.params}] The parameters listed in vary.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (CV) for the normal distribution. Each
entry in the list is named for a parameter in "parameters". New values are
sampled with mean equal to the value in "parameters" and standard deviation
equal to the mean times the CV. Not used with httkpop model.

\item[\code{return.samples}] Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.

\item[\code{tissue}] Desired steady state tissue conentration.

\item[\code{httkpop.dt}] A data table generated by \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.
This defaults to NULL, in which case \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}} is 
called to generate this table.

\item[\code{invitro.mc.arg.list}] Additional parameters passed to 
\code{\LinkA{invitro\_mc}{invitro.Rul.mc}}.

\item[\code{httkpop.generate.arg.list}] Additional parameters passed to 
\code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.

\item[\code{convert.httkpop.arg.list}] Additional parameters passed to the 
convert\_httkpop\_* function for the model.

\item[\code{propagate.invitrouv.arg.list}] Additional parameters passed to model's
associated in vitro uncertainty and variability propagation function

\item[\code{parameterize.arg.list}] Additional parameters passed to the 
parameterize\_* function for the model.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data table where each column corresponds to parameters needed for the 
specified model and each row represents a different Monte Carlo sample of
parameter values.
\end{Value}
%
\begin{Author}\relax
Caroline Ring, Robert Pearce, and John Wambaugh
\end{Author}
%
\begin{References}\relax
Wambaugh, John F., et al. "Toxicokinetic triage for 
environmental chemicals." Toxicological Sciences 147.1 (2015): 55-67.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment international 106 (2017): 105-118.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}


sample_set = create_mc_samples(chem.name = 'bisphenol a')


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{dawson2021}{Dawson et al. 2021 data}{dawson2021}
\aliasA{Dawson2021}{dawson2021}{Dawson2021}
\keyword{data}{dawson2021}
%
\begin{Description}\relax
This table includes QSAR (Random Forest) model predicted values for unbound
fraction plasma protein (fup) and intrinsic hepatic clearance (clint) for a
subset of chemicals in the Tox21 library
(see \url{https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21}).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
dawson2021
\end{verbatim}
\end{Usage}
%
\begin{Format}
data.frame
\end{Format}
%
\begin{Details}\relax
Predictions were made with a set of Random Forest QSAR models,
as reported in Dawson et al. (2021).
\end{Details}
%
\begin{Author}\relax
Daniel E. Dawson
\end{Author}
%
\begin{Source}\relax
Dawson et al. 2021 Random Forest QSAR Model
\end{Source}
%
\begin{References}\relax
Dawson, Daniel E. et al. "Designing QSARs for parameters
of high-throughput toxicokinetic models using open-source descriptors."
Environmental Science \& Technology\_\_\_\_. (2021):\_\_\_\_\_\_.
\end{References}
\inputencoding{utf8}
\HeaderA{EPA.ref}{Reference for EPA Physico-Chemical Data}{EPA.ref}
\keyword{datasets}{EPA.ref}
%
\begin{Description}\relax
The physico-chemical data in the chem.phys\_and\_invitro.data table are
obtained from EPA's Comptox Chemicals dashboard. This variable indicates
the date the Dashboard was accessed.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
EPA.ref
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class \code{character} of length 1.
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
\url{https://comptox.epa.gov/dashboard}
\end{Source}
\inputencoding{utf8}
\HeaderA{estimate\_gfr}{Predict GFR.}{estimate.Rul.gfr}
\keyword{httk-pop}{estimate\_gfr}
%
\begin{Description}\relax
Predict GFR using CKD-EPI equation (for adults) or BSA-based equation (for children).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
estimate_gfr(gfrtmp.dt, gfr_resid_var = TRUE, ckd_epi_race_coeff = FALSE)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{gfrtmp.dt}] A data.table with columns \code{gender}, \code{reth}, 
\code{age\_years}, \code{age\_months}, \code{BSA\_adj}, \code{serum\_creat}.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Add residual variability based on reported residuals for each equation.
\end{Details}
%
\begin{Value}
The same data.table with a \code{gfr\_est} column added, containing 
estimated GFR values.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{estimate\_gfr\_ped}{Predict GFR in children.}{estimate.Rul.gfr.Rul.ped}
\keyword{httk-pop}{estimate\_gfr\_ped}
%
\begin{Description}\relax
BSA-based equation from Johnson et al. 2006, Clin Pharmacokinet 45(9) 931-56. 
Used in Wetmore et al. 2014.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
estimate_gfr_ped(BSA)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{BSA}] Vector of body surface areas in m\textasciicircum{}2.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Vector of GFRs in mL/min/1.73m\textasciicircum{}2.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{estimate\_hematocrit}{Predict hematocrit using smoothing spline.}{estimate.Rul.hematocrit}
\keyword{httk-pop}{estimate\_hematocrit}
%
\begin{Description}\relax
Using precalculated smoothing splines on NHANES log hematocrit vs. age in 
months (and KDE residuals) by gender and race/ethnicity, generate hematocrit 
values for individuals specified by age, gender, and race/ethnicity.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
estimate_hematocrit(hcttmp_dt)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{hcttmp\_dt}] A data.table with columns \code{age\_years},
\code{age\_months}, \code{gender}, \code{reth}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The same data.table with a \code{hematocrit} column added.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{export\_pbtk\_jarnac}{Export model to jarnac.}{export.Rul.pbtk.Rul.jarnac}
\keyword{Export}{export\_pbtk\_jarnac}
%
\begin{Description}\relax
This function exports the multiple compartment PBTK model to a jarnac file.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
export_pbtk_jarnac(
  chem.cas = NULL,
  chem.name = NULL,
  species = "Human",
  initial.amounts = list(Agutlumen = 0),
  filename = "default.jan",
  digits = 4
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the chemical name or CAS number must be specified.

\item[\code{chem.name}] Either the chemical name or CAS number must be specified.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", or default
"Human").

\item[\code{initial.amounts}] Must specify initial amounts in units of choice.

\item[\code{filename}] The name of the jarnac file containing the model.

\item[\code{digits}] Desired number of decimal places to round the parameters.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Compartments to enter into the initial.amounts list includes Agutlumen,
Aart, Aven, Alung, Agut, Aliver, Akidney, and Arest.

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
Text containing a Jarnac language version of the PBTK model.
\end{Value}
%
\begin{Author}\relax
Robert Pearce
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}


export_pbtk_jarnac(chem.name='Nicotine',initial.amounts=list(Agutlumen=1),filename='PBTKmodel.jan')


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{export\_pbtk\_sbml}{Export model to sbml.}{export.Rul.pbtk.Rul.sbml}
\keyword{Export}{export\_pbtk\_sbml}
%
\begin{Description}\relax
This function exports the multiple compartment PBTK model to an sbml file.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
export_pbtk_sbml(
  chem.cas = NULL,
  chem.name = NULL,
  species = "Human",
  initial.amounts = list(Agutlumen = 0),
  filename = "default.xml",
  digits = 4
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the chemical name or CAS number must be specified.

\item[\code{chem.name}] Either the chemical name or CAS number must be specified.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", or default
"Human").

\item[\code{initial.amounts}] Must specify initial amounts in units of choice.

\item[\code{filename}] The name of the jarnac file containing the model.

\item[\code{digits}] Desired number of decimal places to round the parameters.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Compartments to enter into the initial.amounts list includes Agutlumen,
Aart, Aven, Alung, Agut, Aliver, Akidney, and Arest.

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
Text describing the PBTK model in SBML.
\end{Value}
%
\begin{Author}\relax
Robert Pearce
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}


export_pbtk_sbml(chem.name='Nicotine',initial.amounts=list(Agutlumen=1),filename='PBTKmodel.xml')


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Frank2018invivo}{Literature In Vivo Data on Doses Causing Neurological Effects}{Frank2018invivo}
\keyword{data}{Frank2018invivo}
%
\begin{Description}\relax
Studies were selected from Table 1 in Mundy et al., 2015, as
the studies in that publication were cited as examples of
compounds with evidence for developmental neurotoxicity. There
were sufficient in vitro toxicokinetic data available for this
package for only 6 of the 42 chemicals.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Frank2018invivo
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 14 rows and 16 columns.
\end{Format}
%
\begin{Author}\relax
Timothy J. Shafer
\end{Author}
%
\begin{References}\relax
Frank, Christopher L., et al. "Defining toxicological tipping points
in neuronal network development." Toxicology and Applied
Pharmacology 354 (2018): 81-93.

Mundy, William R., et al. "Expanding the test set: Chemicals with
potential to disrupt mammalian brain development." Neurotoxicology
and Teratology 52 (2015): 25-35.
\end{References}
\inputencoding{utf8}
\HeaderA{gen\_age\_height\_weight}{Generate ages, heights, and weights for a virtual population using the  virtual-individuals method.}{gen.Rul.age.Rul.height.Rul.weight}
\keyword{httk-pop}{gen\_age\_height\_weight}
%
\begin{Description}\relax
Generate ages, heights, and weights for a virtual population using the 
virtual-individuals method.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
gen_age_height_weight(
  nsamp = NULL,
  gendernum = NULL,
  reths,
  weight_category,
  agelim_years,
  agelim_months
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{nsamp}] The desired number of individuals in the virtual population. 
\code{nsamp} need not be provided if \code{gendernum} is provided.

\item[\code{gendernum}] Optional: A named list giving the numbers of male and female 
individuals to include in the population, e.g. \code{list(Male=100, 
Female=100)}. Default is NULL, meaning both males and females are included, 
in their proportions in the NHANES data. If both \code{nsamp} and 
\code{gendernum} are provided, they must agree (i.e., \code{nsamp} must be
the sum of \code{gendernum}).

\item[\code{reths}] Optional: a character vector giving the races/ethnicities to 
include in the population. Default is \code{c('Mexican American','Other 
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')}, to include 
all races and ethnicities in their proportions in the NHANES data. 
User-supplied vector must contain one or more of these strings.

\item[\code{weight\_category}] Optional: The weight categories to include in the 
population. Default is \code{c('Underweight', 'Normal', 'Overweight', 
'Obese')}. User-supplied vector must contain one or more of these strings.

\item[\code{agelim\_years}] Optional: A two-element numeric vector giving the minimum 
and maximum ages (in years) to include in the population. Default is 
c(0,79). If \code{agelim\_years} is provided and \code{agelim\_months} is not,
\code{agelim\_years} will override the default value of \code{agelim\_months}.

\item[\code{agelim\_months}] Optional: A two-element numeric vector giving the minimum
and maximum ages (in months) to include in the population. Default is c(0, 
959), equivalent to the default \code{agelim\_years}. If \code{agelim\_months}
is provided and \code{agelim\_years} is not, agelim\_months will override the 
default values of \code{agelim\_years}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table containing variables \begin{description}
 
\item[\code{gender}] Gender of each virtual individual
\item[\code{reth}] Race/ethnicity of each virtual individual
\item[\code{age\_months}] Age in months of each virtual individual
\item[\code{age\_years}] Age in years of each virtual individual
\item[\code{weight}] Body weight in kg of each virtual individual
\item[\code{height}] Height in cm of each virtual individual
\end{description}

\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118

importFrom survey svymean
\end{References}
\inputencoding{utf8}
\HeaderA{gen\_height\_weight}{Generate heights and weights for a virtual population.}{gen.Rul.height.Rul.weight}
\keyword{httk-pop}{gen\_height\_weight}
%
\begin{Description}\relax
Generate heights and weights for a virtual population.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
gen_height_weight(hbw_dt)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{hbw\_dt}] A data.table describing the virtual population by race,
gender, and age (in years and months). Must have variables \code{gender},
\code{reth}, \code{age}, and \code{age.years}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The same data.table with two new variables added: \code{weight} and
\code{height}. Respectively, these give individual body weights in kg, and
individual heights in cm.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{gen\_serum\_creatinine}{Predict GFR.}{gen.Rul.serum.Rul.creatinine}
\keyword{httk-pop}{gen\_serum\_creatinine}
%
\begin{Description}\relax
Predict serum creatinine using smoothing splines and kernel density estimates of residual variability
\end{Description}
%
\begin{Usage}
\begin{verbatim}
gen_serum_creatinine(serumcreat.dt)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{serumcreat.dt}] A data.table with columns \code{gender}, \code{reth}, 
\code{age\_years}, \code{age\_months}, \code{BSA\_adj}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The same data.table with a \code{serum\_creat} column added, containing 
spline-interpolated serum creatinine values.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{get\_cheminfo}{Retrieve chemical information from HTTK package}{get.Rul.cheminfo}
\keyword{Retrieval}{get\_cheminfo}
%
\begin{Description}\relax
This function provides the information specified in "info=" (can be single entry
or vector) for all chemicals for which a toxicokinetic model can be
parameterized for a given species. Since different models have different 
requirements and not all chemicals have complete data, this function will 
return different number of chemicals depending on the model specififed.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_cheminfo(
  info = "CAS",
  species = "Human",
  fup.lod.default = 0.005,
  model = "3compartmentss",
  default.to.human = FALSE,
  median.only = FALSE,
  fup.ci.cutoff = TRUE,
  clint.pvalue.threshold = 0.05,
  suppress.messages = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{info}] A single character vector (or collection of character vectors)
from "Compound", "CAS", "DTXSID, "logP", "pKa\_Donor"," pKa\_Accept", "MW", "Clint",
"Clint.pValue", "Funbound.plasma","Structure\_Formula", or "Substance\_Type". info="all"
gives all information for the model and species.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{fup.lod.default}] Default value used for fraction of unbound plasma for
chemicals where measured value was below the limit of detection. Default
value is 0.0005.

\item[\code{model}] Model used in calculation, 'pbtk' for the multiple compartment
model, '1compartment' for the one compartment model, '3compartment' for
three compartment model, '3compartmentss' for the three compartment model
without partition coefficients, or 'schmitt' for chemicals with logP and
fraction unbound (used in predict\_partitioning\_schmitt).

\item[\code{default.to.human}] Substitutes missing values with human values if
true.

\item[\code{median.only}] Use median values only for fup and clint.  Default is FALSE.

\item[\code{fup.ci.cutoff}] Cutoff for the level of uncertainty in fup estimates.
This value should be between (0,1). Default is `NULL` specifying no filtering.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.

\item[\code{suppress.messages}] Whether or not the output messages are suppressed.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
When default.to.human is set to TRUE, and the species-specific data,
Funbound.plasma and Clint, are missing from 
\code{\LinkA{chem.physical\_and\_invitro.data}{chem.physical.Rul.and.Rul.invitro.data}}, human values are given instead.

In some cases the rapid equilbrium dailysis method (Waters et al., 2008)
fails to yield detectable concentrations for the free fraction of chemical. 
In those cases we assume the compound is highly bound (that is, Fup approaches
zero). For some calculations (for example, steady-state plasma concentration)
there is precendent (Rotroff et al., 2010) for using half the average limit 
of detection, that is, 0.005 (this value is configurable via the argument
fup.lod.default). We do not recomend using other models where 
quantities like partition coefficients must be predicted using Fup. We also
do not recomend including the value 0.005 in training sets for Fup predictive
models.

\strong{Note} that in some cases the \strong{Funbound.plasma} and the 
\strong{intrinsic clearance} are
\emph{provided as a series of numbers separated by commas}. These values are the 
result of Bayesian analysis and characterize a distribution: the first value
is the median of the distribution, while the second and third values are the 
lower and upper 95th percentile (that is qunatile 2.5 and 97.5) respectively.
For intrinsic clearance a fourth value indicating a p-value for a decrease is
provided. Typically 4000 samples were used for the Bayesian analusis, such
that a p-value of "0" is equivale to "<0.00025". See Wambaugh et al. (2019)
for more details. If argument meadian.only == TRUE then only the median is
reported for parameters with Bayesian analysis distributions. If the 95
credible interval is larger than fup.ci.cutoff (defaults
to NULL) then the Fup is treated as too uncertain and the value NA is given.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{vector/data.table}] Table (if info has multiple entries) or 
vector containing a column for each valid entry 
specified in the argument "info" and a row for each chemical with sufficient
data for the model specified by argument "model":

\Tabular{lll}{
\strong{Column} & \strong{Description} & \strong{units} \\{}
Compound & The preferred name of the chemical compound & none \\{} 
CAS & The preferred Chemical Abstracts Service Registry Number & none \\{}  
DTXSID & DSSTox Structure ID 
(\url{http://comptox.epa.gov/dashboard}) & none \\{} 
logP & The log10 octanol:water partition coefficient& log10 unitless ratio \\{} 
MW & The chemical compound molecular weight & g/mol \\{} 
pKa\_Accept & The hydrogen acceptor equilibria concentrations 
& logarithm \\{}   
pKa\_Donor & The hydrogen donor equilibria concentrations 
& logarithm \\{}   
[SPECIES].Clint & (Primary hepatocyte suspension) 
intrinsic hepatic clearance & uL/min/10\textasciicircum{}6 hepatocytes \\{}    
[SPECIES].Clint.pValue & Probability that there is no clearance observed. & none \\{}  
[SPECIES].Funbound.plasma & Chemical fraction unbound in presence of 
plasma proteins & unitless fraction \\{} 
[SPECIES].Rblood2plasma & Chemical concentration blood to plasma ratio & unitless ratio \\{}  
}

\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh, Robert Pearce, and Sarah E. Davidson
\end{Author}
%
\begin{References}\relax
Rotroff, Daniel M., et al. "Incorporating human dosimetry and exposure into 
high-throughput in vitro toxicity screening." Toxicological Sciences 117.2 
(2010): 348-358.

Waters, Nigel J., et al. "Validation of a rapid equilibrium dialysis approach 
for the measurement of plasma protein binding." Journal of pharmaceutical 
sciences 97.10 (2008): 4586-4595.

Wambaugh, John F., et al. "Assessing toxicokinetic uncertainty and 
variability in risk prioritization." Toxicological Sciences 172.2 (2019): 
235-251.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}


# List all CAS numbers for which the 3compartmentss model can be run in humans: 
get_cheminfo()

get_cheminfo(info=c('compound','funbound.plasma','logP'),model='pbtk') 
# See all the data for humans:
get_cheminfo(info="all")

TPO.cas <- c("741-58-2", "333-41-5", "51707-55-2", "30560-19-1", "5598-13-0", 
"35575-96-3", "142459-58-3", "1634-78-2", "161326-34-7", "133-07-3", "533-74-4", 
"101-05-3", "330-54-1", "6153-64-6", "15299-99-7", "87-90-1", "42509-80-8", 
"10265-92-6", "122-14-5", "12427-38-2", "83-79-4", "55-38-9", "2310-17-0", 
"5234-68-4", "330-55-2", "3337-71-1", "6923-22-4", "23564-05-8", "101-02-0", 
"140-56-7", "120-71-8", "120-12-7", "123-31-9", "91-53-2", "131807-57-3", 
"68157-60-8", "5598-15-2", "115-32-2", "298-00-0", "60-51-5", "23031-36-9", 
"137-26-8", "96-45-7", "16672-87-0", "709-98-8", "149877-41-8", "145701-21-9", 
"7786-34-7", "54593-83-8", "23422-53-9", "56-38-2", "41198-08-7", "50-65-7", 
"28434-00-6", "56-72-4", "62-73-7", "6317-18-6", "96182-53-5", "87-86-5", 
"101-54-2", "121-69-7", "532-27-4", "91-59-8", "105-67-9", "90-04-0", 
"134-20-3", "599-64-4", "148-24-3", "2416-94-6", "121-79-9", "527-60-6", 
"99-97-8", "131-55-5", "105-87-3", "136-77-6", "1401-55-4", "1948-33-0", 
"121-00-6", "92-84-2", "140-66-9", "99-71-8", "150-13-0", "80-46-6", "120-95-6",
"128-39-2", "2687-25-4", "732-11-6", "5392-40-5", "80-05-7", "135158-54-2", 
"29232-93-7", "6734-80-1", "98-54-4", "97-53-0", "96-76-4", "118-71-8", 
"2451-62-9", "150-68-5", "732-26-3", "99-59-2", "59-30-3", "3811-73-2", 
"101-61-1", "4180-23-8", "101-80-4", "86-50-0", "2687-96-9", "108-46-3", 
"95-54-5", "101-77-9", "95-80-7", "420-04-2", "60-54-8", "375-95-1", "120-80-9",
"149-30-4", "135-19-3", "88-58-4", "84-16-2", "6381-77-7", "1478-61-1", 
"96-70-8", "128-04-1", "25956-17-6", "92-52-4", "1987-50-4", "563-12-2", 
"298-02-2", "79902-63-9", "27955-94-8")
httk.TPO.rat.table <- subset(get_cheminfo(info="all",species="rat"),
 CAS %in% TPO.cas)
 
httk.TPO.human.table <- subset(get_cheminfo(info="all",species="human"),
 CAS %in% TPO.cas)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_chem\_id}{Retrieve chemical identity from HTTK package}{get.Rul.chem.Rul.id}
\keyword{cheminformatics}{get\_chem\_id}
%
\begin{Description}\relax
Given one of chem.name, chem.cas (Chemical Abstract Service Registry Number),
or DTXSID (DSStox Substance Identifier \url{https://comptox.epa.gov/dashboard}) this
function checks if the chemical is available and, if so, returns all three
pieces of information.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_chem_id(chem.cas = NULL, chem.name = NULL, dtxsid = NULL)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] CAS regstry number

\item[\code{chem.name}] Chemical name

\item[\code{dtxsid}] DSSTox Substance identifier
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A list containing the following chemical identifiers:
\begin{ldescription}
\item[\code{chem.cas}] CAS registry number
\item[\code{chem.name}] Name
\item[\code{dtxsid}] DTXSID
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
\inputencoding{utf8}
\HeaderA{get\_gfr\_category}{Categorize kidney function by GFR.}{get.Rul.gfr.Rul.category}
\keyword{httk-pop}{get\_gfr\_category}
%
\begin{Description}\relax
For adults: 
In general GFR > 60 is considered normal 
15 < GFR < 60 is considered kidney disease 
GFR < 15 is considered kidney failure
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_gfr_category(age_years, age_months, gfr_est)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{age\_years}] Vector of ages in years.

\item[\code{age\_months}] Vector of ages in months.

\item[\code{gfr\_est}] Vector of estimated GFR values in mL/min/1.73m\textasciicircum{}2.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
These values can also be used for children 2 years old and greater (see PEDIATRICS IN
REVIEW Vol. 29 No. 10 October 1, 2008 pp. 335-341 (doi:
10.1542/pir.29-10-335))
\end{Details}
%
\begin{Value}
Vector of GFR categories: 'Normal', 'Kidney Disease', 'Kidney
Failure'.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{get\_invitroPK\_param}{Retrieve data from chem.physical\_and\_invitro.data table}{get.Rul.invitroPK.Rul.param}
%
\begin{Description}\relax
or fraction unbound in plasma) from the main HTTK data. This function looks
for species-specific values.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_invitroPK_param(
  param,
  species,
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{param}] The in vitro pharmacokinetic parameter needed.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The value of the parameter, if found
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
\inputencoding{utf8}
\HeaderA{get\_lit\_cheminfo}{Get literature Chemical Information.}{get.Rul.lit.Rul.cheminfo}
\keyword{Literature}{get\_lit\_cheminfo}
\keyword{Retrieval}{get\_lit\_cheminfo}
%
\begin{Description}\relax
This function provides the information specified in "info=" for all
chemicals with data from the Wetmore et al. (2012) and (2013) publications
and other literature.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_lit_cheminfo(info = "CAS", species = "Human")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{info}] A single character vector (or collection of character vectors)
from
"Compound", "CAS", "MW", "Raw.Experimental.Percentage.Unbound",
"Entered.Experimental.Percentage.Unbound", "Fub", "source\_PPB",
"Renal\_Clearance", "Met\_Stab", "Met\_Stab\_entered",
"r2", "p.val", "Concentration..uM.", "Css\_lower\_5th\_perc.mg.L.", 
"Css\_median\_perc.mg.L.", "Css\_upper\_95th\_perc.mg.L.",
"Css\_lower\_5th\_perc.uM.","Css\_median\_perc.uM.","Css\_upper\_95th\_perc.uM.",
and "Species".

\item[\code{species}] Species desired (either "Rat" or default "Human").
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{info}] Table/vector containing values specified in "info" for
valid chemicals.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., "Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment," Toxicological Sciences 125
157-174 (2012)

Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., "The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays"
Toxicological Sciences, 132:327-346 (2013).

Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
"Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing" Toxicological
Sciences, kfv171.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

get_lit_cheminfo()
get_lit_cheminfo(info=c('CAS','MW'))

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_lit\_css}{Get literature Css}{get.Rul.lit.Rul.css}
\keyword{Literature}{get\_lit\_css}
\keyword{Monte-Carlo}{get\_lit\_css}
%
\begin{Description}\relax
This function retrives a steady-state plasma concentration as a result of
infusion dosing from the Wetmore et al. (2012) and (2013) publications and
other literature.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_lit_css(
  chem.cas = NULL,
  chem.name = NULL,
  daily.dose = 1,
  which.quantile = 0.95,
  species = "Human",
  clearance.assay.conc = NULL,
  output.units = "mg/L",
  suppress.messages = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the cas number or the chemical name must be
specified.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{daily.dose}] Total daily dose infused in units of mg/kg BW/day.
Defaults to 1 mg/kg/day.

\item[\code{which.quantile}] Which quantile from the SimCYP Monte Carlo simulation
is requested. Can be a vector.

\item[\code{species}] Species desired (either "Rat" or default "Human").

\item[\code{clearance.assay.conc}] Concentration of chemical used in measureing
intrinsic clearance data, 1 or 10 uM.

\item[\code{output.units}] Returned units for function, defaults to mg/L but can
also be uM (specify units = "uM").

\item[\code{suppress.messages}] Whether or not the output message is suppressed.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A numeric vector with the literature steady-state plasma 
concentration (1 mg/kg/day) for the requested quantiles
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., "Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment," Toxicological Sciences 125
157-174 (2012)

Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., "The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays"
Toxicological Sciences, 132:327-346 (2013).

Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
"Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing" Toxicological
Sciences, kfv171.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
get_lit_css(chem.cas="34256-82-1")

get_lit_css(chem.cas="34256-82-1",species="Rat",which.quantile=0.5)

get_lit_css(chem.cas="80-05-7", daily.dose = 1,which.quantile = 0.5, output.units = "uM")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_lit\_oral\_equiv}{Get Literature Oral Equivalent Dose}{get.Rul.lit.Rul.oral.Rul.equiv}
\keyword{Literature}{get\_lit\_oral\_equiv}
\keyword{Monte-Carlo}{get\_lit\_oral\_equiv}
%
\begin{Description}\relax
This function converts a chemical plasma concetration to an oral equivalent
dose using the values from the Wetmore et al. (2012) and (2013) publications
and other literature.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_lit_oral_equiv(
  conc,
  chem.name = NULL,
  chem.cas = NULL,
  suppress.messages = FALSE,
  which.quantile = 0.95,
  species = "Human",
  input.units = "uM",
  output.units = "mg",
  clearance.assay.conc = NULL,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{conc}] Bioactive in vitro concentration in units of specified
input.units, default of uM.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{chem.cas}] Either the CAS number or the chemical name must be
specified.

\item[\code{suppress.messages}] Suppress output messages.

\item[\code{which.quantile}] Which quantile from the SimCYP Monte Carlo simulation
is requested. Can be a vector.  Papers include 0.05, 0.5, and 0.95 for
humans and 0.5 for rats.

\item[\code{species}] Species desired (either "Rat" or default "Human").

\item[\code{input.units}] Units of given concentration, default of uM but can also
be mg/L.

\item[\code{output.units}] Units of dose, default of 'mg' for mg/kg BW/ day or
'mol' for mol/ kg BW/ day.

\item[\code{clearance.assay.conc}] Concentration of chemical used in measureing
intrinsic clearance data, 1 or 10 uM.

\item[\code{...}] Additional parameters passed to get\_lit\_css.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Equivalent dose in specified units, default of mg/kg BW/day.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., "Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment," Toxicological Sciences 125
157-174 (2012)

Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., "The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays"
Toxicological Sciences, 132:327-346 (2013).

Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
"Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing" Toxicological
Sciences, kfv171.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

table <- NULL
for(this.cas in sample(get_lit_cheminfo(),50)) table <- rbind(table,cbind(
as.data.frame(this.cas),as.data.frame(get_lit_oral_equiv(conc=1,chem.cas=this.cas))))




get_lit_oral_equiv(0.1,chem.cas="34256-82-1")

get_lit_oral_equiv(0.1,chem.cas="34256-82-1",which.quantile=c(0.05,0.5,0.95))

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_physchem\_param}{Get physico-chemical parameters from chem.physical\_and\_invitro.data}{get.Rul.physchem.Rul.param}
%
\begin{Description}\relax
This function retrieves physico-chemical properties ("param") for the chemical specified 
by chem.name or chem.cas from the vLiver tables.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_physchem_param(param, chem.name = NULL, chem.cas = NULL, dtxsid = NULL)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{param}] The desired parameters, a vector or single value.

\item[\code{chem.name}] The chemical names that you want parameters for, a vector or single value

\item[\code{chem.cas}] The chemical CAS numbers that you want parameters for, a vector or single value

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The parameters, either a single value, a named list for a single chemical, or a list of lists
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

get_physchem_param(param = 'logP', chem.cas = '80-05-7')
get_physchem_param(param = c('logP','MW'), chem.cas = c('80-05-7','81-81-2'))


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_rblood2plasma}{Get ratio of the blood concentration to the plasma concentration.}{get.Rul.rblood2plasma}
\keyword{Parameter}{get\_rblood2plasma}
%
\begin{Description}\relax
This function attempts to retrieve a measured species- and chemical-specific 
blood:plasma concentration ratio.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_rblood2plasma(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{chem.cas}] Either the CAS number or the chemical name must be
specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
A value of NA is returned when the requested value is unavailable.  Values
are retrieved from chem.physical\_and\_invitro.data. 
details than the description above \textasciitilde{}\textasciitilde{}
\end{Details}
%
\begin{Value}
A numeric value for the steady-state ratio of chemical concentration in blood
to plasma
\end{Value}
%
\begin{Author}\relax
Robert Pearce
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}

get_rblood2plasma(chem.name="Bisphenol A")
get_rblood2plasma(chem.name="Bisphenol A",species="Rat")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_weight\_class}{Given vectors of age, BMI, recumbent length, weight, and gender, categorizes weight classes using CDC and WHO categories.}{get.Rul.weight.Rul.class}
\keyword{httk-pop}{get\_weight\_class}
%
\begin{Description}\relax
Given vectors of age, BMI, recumbent length, weight, and gender,
categorizes weight classes using CDC and WHO categories.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_weight_class(age_years, age_months, bmi, recumlen, weight, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{age\_years}] A vector of
ages in years.

\item[\code{age\_months}] A vector of ages in months.

\item[\code{bmi}] A vector of BMIs.

\item[\code{recumlen}] A vector of heights or recumbent lengths in cm.

\item[\code{weight}] A vector of body weights in kg.

\item[\code{gender}] A vector of genders (as 'Male' or 'Female').
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A character vector of weight classes. Each element will be one of
'Underweight', 'Normal', 'Overweight', or 'Obese'.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{hematocrit\_infants}{Predict hematocrit in infants under 1 year old.}{hematocrit.Rul.infants}
\keyword{httk-pop}{hematocrit\_infants}
%
\begin{Description}\relax
For infants under 1 year, hematocrit was not measured in NHANES. Assume a
log-normal distribution where plus/minus 1 standard deviation of the
underlying normal distribution is given by the reference range. Draw
hematocrit values from these distributions by age.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
hematocrit_infants(age_months)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{age\_months}] Vector of ages in months; all must be <= 12.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax

\Tabular{cc}{ 
Age & Reference range\\{} 
<1 month & 31-49\\{} 
1-6 months & 29-42\\{} 
7-12 months & 33-38 
}
\end{Details}
%
\begin{Value}
Vector of hematocrit percentages corresponding to the input vector
of ages.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{honda.ivive}{Return the assumptions used in Honda et al. 2019}{honda.ivive}
\keyword{Solve}{honda.ivive}
%
\begin{Description}\relax
This function returns four of the better performing sets of assumptions evaluated in Honda et al. 2019 
(https://doi.org/10.1371/journal.pone.0217564).These include four different combinations of hepatic clearance assumption, in vivo bioactivity assumption, 
and relevant tissue assumption. Generally, this function is not called directly by the user, but instead
called by setting the IVIVE option in calc\_mc\_oral\_equiv, calc\_mc\_css, and calc\_analytic functions. Currently, these IVIVE option 
is not implemented the solve\_1comp etc. functions.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
honda.ivive(method = "Honda1", tissue = "liver")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{method}] This is set to one of "Honda1", "Honda2", "Honda3", or "Honda4".

\item[\code{tissue}] This is only relevant to "Honda4" and indicates the relevant tissue compartment.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
"Honda1" - tissue = NULL, restrictive.clearance = TRUE, bioactive.free.invivo = TRUE
This assumption assumes restrictive hepatic clearance, and treats the free concentration in plasma as 
the bioactive concentration in vivo. This option must be used in combination with the concentration in vitro 
predicted by armitage\_eval(), otherwise the result will be the same as "Honda2". This option corresponds to the result 
in Figure 8 panel c) restrictive, mean free plasma conc., Armitage in Honda et al. 2019.
"Honda2" - tissue = NULL, restrictive.clearance = TRUE, bioactive.free.invivo = TRUE
This assumption assumes restrictive hepatic clearance, and treats the free concentration in plasma as 
the bioactive concentration in vivo. This option corresponds to the result 
in Figure 8 panel b) restrictive, mean free plasma conc. in Honda et al. 2019.
"Honda3" - tissue = NULL, restrictive.clearance = TRUE, bioactive.free.invivo = TRUE
This assumption assumes restrictive hepatic clearance, and treats the free concentration in plasma as 
the bioactive concentration in vivo. This option corresponds to the result 
in Figure 8 panel a) restrictive, mean total plasma conc. in Honda et al. 2019.
"Honda4" - tissue = tissue, restrictive.clearance = FALSE, bioactive.free.invivo = TRUE
This assumption assumes restrictive hepatic clearance, and treats the free concentration in plasma as 
the bioactive concentration in vivo. The input tissue should be relevant to the in vitro assay endpoint used as input or that
the result is being compared to. This option corresponds to the result 
in Figure 8 panel d) nonrestrictive, mean tissue conc. in Honda et al. 2019.
\end{Details}
%
\begin{Value}
A list of tissue, bioactive.free.invivo, and restrictive.clearance assumptions.
\end{Value}
%
\begin{Author}\relax
Greg Honda and John Wambaugh
\end{Author}
%
\begin{References}\relax
Honda, Gregory S., et al. "Using the Concordance of In Vitro and 
In Vivo Data to Evaluate Extrapolation Assumptions." 2019. PLoS ONE 14(5): e0217564.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
honda.ivive(method = "Honda1", tissue = NULL)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{howgate}{Howgate 2006}{howgate}
\keyword{data}{howgate}
%
\begin{Description}\relax
This data set is only used in Vignette 5.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
howgate
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table containing 24 rows and 11 columns.
\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Howgate, E. M., et al. "Prediction of in vivo drug clearance from in vitro
data. I: impact of inter-individual variability." Xenobiotica 36.6 (2006):
473-497.
\end{References}
\inputencoding{utf8}
\HeaderA{httkpop}{httkpop: Virtual population generator for HTTK.}{httkpop}
\aliasA{httkpop-package}{httkpop}{httkpop.Rdash.package}
\keyword{httk-pop}{httkpop}
%
\begin{Description}\relax
The httkpop package generates virtual population physiologies for use in 
population TK.
\end{Description}
%
\begin{Details}\relax
To simulate inter-individual variability in the TK model, a MC approach
is used: the model parameters are sampled from known or assumed
distributions, and the model is evaluated for each sampled set of
parameters. To simulate variability across subpopulations, the MC approach
needs to capture the parameter correlation structure. For example,
kidney function changes with age (Levey et al., 2009), thus the
distribution of GFR is likely different in 6-year-olds than in 65-yearolds.
To directly measure the parameter correlation structure, all parameters
need to be measured in each individual in a representative
sample population. Such direct measurements are extremely limited.
However, the correlation structure of the physiological parameters can
be inferred from their known individual correlations with demographic
and anthropometric quantities for which direct population measurements
do exist. These quantities are sex, race/ethnicity, age, height, and
weight (Howgate et al., 2006; Jamei et al., 2009a; Johnson et al., 2006;
McNally et al., 2014; Price et al., 2003). Direct measurements of these
quantities in a large, representative sample of the U.S. population are
publicly available from NHANES. NHANES also includes laboratory
measurements, including both serum creatinine, which can be used to
estimate GFR (Levey et al., 2009), and hematocrit. For conciseness, sex,
race/ethnicity, age, height, weight, serum creatinine, and hematocrit
will be called the NHANES quantities.

HTTK-Pop's correlated MC approach begins by sampling from the
joint distribution of the NHANES quantities to simulate a population.
Then, for each individual in the simulated population, HTTKePop
predicts the physiological parameters from the NHANES
quantities using regression equations from the literature (Barter et al.,
2007; Baxter-Jones et al., 2011; Bosgra et al., 2012; Koo et al., 2000;
Levey et al., 2009; Looker et al., 2013; McNally et al., 2014; Ogiu et al.,
1997; Price et al., 2003; Schwartz and Work, 2009; Webber and Barr 2012). 
Correlations among the physiological parameters are induced by
their mutual dependence on the correlated NHANES quantities. Finally,
residual variability is added to the predicted physiological parameters
using estimates of residual marginal variance (i.e., variance not explained
by the regressions on the NHANES quantities) (McNally et al.,
2014).

Data were combined from the three most recent publicly-available
NHANES cycles: 2007-2008, 2009-2010, and 2011-2012. For each
cycle, some NHANES quantities - height, weight, serum creatinine,
and hematocrit - were measured only in a subset of respondents. Only
these subsets were included in HTTKePop. The pooled subsets from the
three cycles contained 29,353 unique respondents. Some respondents
were excluded from analysis: those with age recorded as 80 years (because
all NHANES respondents 80 years and older were marked as
"80"); those with missing height, weight or hematocrit data; and those
aged 12 years or older with missing serum creatinine data. These criteria
excluded 4807 respondents, leaving 24,546 unique respondents. Each
NHANES respondent was assigned a cycle-specific sample weight,
which can be interpreted as the number of individuals in the total U.S.
population represented by each NHANES respondent in each cycle
(Johnson et al., 2013). Because data from three cycles were combined,
the sample weights were rescaled (divided by the number of cycles
being combined, as recommended in NHANES data analysis documentation)
(Johnson et al., 2013). To handle the complex NHANES
sampling structure, the R survey package was used to analyze the
NHANES data (Lumley, 2004). 

To allow generation of virtual populations specified by weight class,
we coded a categorical variable for each NHANES respondent. The
categories Underweight, Normal, Overweight, or Obese were assigned
based on weight, age, and height/length (Grummer-Strawn et al., 2010;
Kuczmarski et al., 2002; Ogden et al., 2014; WHO, 2006, 2010). 
We implemented two population simulation methods within HTTK-Pop:
the direct-resampling method and the virtual-individuals method.
The direct-resampling method simulated a population by sampling
NHANES respondents with replacement, with probabilities proportional
to the sample weights. Each individual in the resulting simulated population
was an NHANES respondent, identified by a unique NHANES
sequence number. By contrast, the second method generates "virtual
individuals" - sets of NHANES quantities that obey the approximate
joint distribution of the NHANES quantities (calculated using weighted
smoothing functions and kernel density estimators), but do not necessarily correspond to
any particular NHANES respondent. The direct-resampling method removed
the possibility of generating unrealistic combinations of the
NHANES quantities; the virtual-individuals method allowed the use of
interpolation to simulate subpopulations represented by only a small
number of NHANES respondents. 

For either method, HTTK-Pop
takes optional specifications about the population to be simulated
and then samples from the appropriate conditional joint
distribution of the NHANES quantities.

Once HTTK-Pop has simulated a population characterized by the
NHANES quantities, the physiological parameters of the TK model
are predicted from the NHANES quantities using regression
equations from the literature. Liver mass was predicted for individuals
over age 18 using allometric scaling with height from Reference Man
(Valentin, 2002), and for individuals under 18 using regression relationships
with height and weight published by Ogiu et al. (1997).
Residual marginal variability was added for each individual as in
PopGen (McNally et al., 2014). Similarly, hepatic portal vein blood
flows (in L/h) are predicted as fixed fractions of a cardiac output allometrically
scaled with height from Reference Man (Valentin, 2002),
and residual marginal variability is added for each individual (McNally
et al., 2014). Glomerular filtration rate (GFR) (in L/h/1.73 m2 body
surface area) is predicted from age, race, sex, and serum creatinine
using the CKD-EPI equation, for individuals over age 18 (Levey et al.,
2009). For individuals under age 18, GFR is estimated from body surface
area (BSA) (Johnson et al., 2006); BSA is predicted using Mosteller's
formula (Verbraecken et al., 2006) for adults and Haycock's
formula (Haycock et al., 1978) for children. Hepatocellularity (in millions
of cells per gram of liver tissue) is predicted from age using an
equation developed by Barter et al. (2007). Hematocrit is estimated
from NHANES data for individuals 1 year and older. For individuals
younger than 1 year, for whom NHANES did not measure hematocrit
directly, hematocrit was predicted from age in months, using published
reference ranges (Lubin, 1987).

In addition to the HTTK physiological parameters, the HTTK models
include chemical-specific parameters representing the fraction of chemical
unbound in plasma (Fup) and intrinsic clearance (CLint). Because
these parameters represent interactions of the chemical with the body,
their values will vary between individuals. To simulate this variability,
Fub and CLint were included in MC simulations, by sampling from estimated
or assumed distributions for the parameters defining them.

Variability in hematocrit was simulated either using NHANES data
(for individuals ages 1 and older) or using age-based reference ranges
(for individuals under age 1). Fup was treated as a random variable
obeying a distribution censored below the average limit of quantification
(LOQ) of the in vitro assay. Specifically, Fup was assumed to obey a
normal distribution truncated below at 0 and above at 1, centered at the
Fup value measured in vitro, with a 30
the average LOQ (0.01), Fup was instead drawn from a uniform distribution
between 0 and 0.01. Fup was assumed to be independent of all
other parameters. This censored normal distribution was chosen to
match that used in Wambaugh et al. (2015).

Variability in hepatocellularity (106 cells/g liver) and Mliver (kg)
were simulated. The remaining source of variability in CLint,h is variability
in CLint, which was simulated using a Gaussian mixture distribution
to represent the population proportions of poor metabolizers
(PMs) and non-PMs of each substance. The true prevalence of PMs is
isozyme-specific (Ma et al., 2002; Yasuda et al., 2008); however, isozyme-
specific metabolism data were not available for the majority of
chemicals considered. We therefore made a simplifying assumption that
5
slower than average.
With 95
a normal distribution truncated below at zero, centered at the value
measured in vitro, with a 30
CLint was drawn from a PM distribution: a truncated normal
distribution centered on one-tenth of the in vitro value with 30
Both CLint itself and the probability of being a PM were assumed to be
independent of all other parameters. The truncated normal nonePM
distribution was chosen because it has been used (with 100
in previous work (Rotroff et al., 2010; Wambaugh et al., 2015;
Wetmore et al., 2014; Wetmore et al., 2015; Wetmore et al., 2012); the
PM distribution was chosen to comport with the nonePM distribution.
\end{Details}
%
\begin{Section}{Main function to generate a population}

If you just want to generate
a table of (chemical-independent) population physiology parameters, use 
\code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.
\end{Section}
%
\begin{Section}{Using HTTK-Pop with HTTK}

To generate a population and then run an 
HTTK model for that population, the workflow is as follows: \begin{enumerate}
 
\item{} Generate a population using \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}. \item{} For
a given HTTK chemical and general model, convert the population data to 
corresponding sets of HTTK model parameters using 
\code{\LinkA{httkpop\_mc}{httkpop.Rul.mc}}.
\end{enumerate}

\end{Section}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118     

Levey, A.S., Stevens, L.A., Schmid, C.H., Zhang, Y.L., Castro, A.F., Feldman, H.I., et al.,
2009. A new equation to estimate glomerular filtration rate. Ann. Intern. Med. 150,
604-612.

Howgate, E., Rowland-Yeo, K., Proctor, N., Tucker, G., Rostami-Hodjegan, A., 2006.
Prediction of in vivo drug clearance from in vitro data. I: impact of inter-individual
variability. Xenobiotica 36, 473-497.

Jamei, M., Dickinson, G.L., Rostami-Hodjegan, A., 2009a. A framework for assessing
inter-individual variability in pharmacokinetics using virtual human populations and
integrating general knowledge of physical chemistry, biology, anatomy, physiology
and genetics: a tale of 'bottom-up' vs 'top-down' recognition of covariates. Drug Metab.
Pharmacokinet. 24, 53-75.

Johnson, T.N., Rostami-Hodjegan, A., Tucker, G.T., 2006. Prediction of the clearance of
eleven drugs and associated variability in neonates, infants and children. Clin.
Pharmacokinet. 45, 931-956.

McNally, K., Cotton, R., Hogg, A., Loizou, G., 2014. PopGen: a virtual human population
generator. Toxicology 315, 70-85.

Price, P.S., Conolly, R.B., Chaisson, C.F., Gross, E.A., Young, J.S., Mathis, E.T., et al.,
2003. Modeling interindividual variation in physiological factors used in PBPK
models of humans. Crit. Rev. Toxicol. 33, 469-503.

Barter, Z.E., Bayliss, M.K., Beaune, P.H., Boobis, A.R., Carlile, D.J., Edwards, R.J., et al.,
2007. Scaling factors for the extrapolation of in vivo metabolic drug clearance from
in vitro data: reaching a consensus on values of human micro-somal protein and
hepatocellularity per gram of liver. Curr. Drug Metab. 8, 33-45.

Baxter-Jones, A.D., Faulkner, R.A., Forwood, M.R., Mirwald, R.L., Bailey, D.A., 2011.
Bone mineral accrual from 8 to 30 years of age: an estimation of peak bone mass. J.
Bone Miner. Res. 26, 1729-1739.

Bosgra, S., van Eijkeren, J., Bos, P., Zeilmaker, M., Slob, W., 2012. An improved model to
predict physiologically based model parameters and their inter-individual variability
from anthropometry. Crit. Rev. Toxicol. 42, 751-767.

Koo, W.W., Walters, J.C., Hockman, E.M., 2000. Body composition in human infants at
birth and postnatally. J. Nutr. 130, 2188-2194.

Looker, A., Borrud, L., Hughes, J., Fan, B., Shepherd, J., Sherman, M., 2013. Total body
bone area, bone mineral content, and bone mineral density for individuals aged 8
years and over: United States, 1999-2006. In: Vital and health statistics Series 11,
Data from the National Health Survey, pp. 1-78.

Ogiu, N., Nakamura, Y., Ijiri, I., Hiraiwa, K., Ogiu, T., 1997. A statistical analysis of the
internal organ weights of normal Japanese people. Health Phys. 72, 368-383.

Schwartz, G.J., Work, D.F., 2009. Measurement and estimation of GFR in children and
adolescents. Clin. J. Am. Soc. Nephrol. 4, 1832-1843.

Webber, C.E., Barr, R.D., 2012. Age-and gender-dependent values of skeletal muscle mass
in healthy children and adolescents. J. Cachex. Sarcopenia Muscle 3, 25-29.

Johnson, C.L., Paulose-Ram, R., Ogden, C.L., Carroll, M.D., Kruszon-Moran, D.,
Dohrmann, S.M., et al., 2013. National health and nutrition examination survey:
analytic guidelines, 1999-2010. Vital and health statistics Series 2. Data Eval.
Methods Res. 1-24.

Lumley, T., 2004. Analysis of complex survey samples. J. Stat. Softw. 9, 1-19.

Grummer-Strawn, L.M., Reinold, C.M., Krebs, N.F., Control, C.f.D.; Prevention, 2010. Use
of World Health Organization and CDC Growth Charts for Children Aged 0-59
Months in the United States. Department of Health and Human Services, Centers for
Disease Control and Prevention.

Kuczmarski, R.J., Ogden, C.L., Guo, S.S., Grummer-Strawn, L.M., Flegal, K.M., Mei, Z.,
et al., 2002. 2000 CDC growth charts for the United States: methods and development.
Vital Health Stat. Series 11, Data from the national health survey 246, 1-190.

Ogden, C.L., Carroll, M.D., Kit, B.K., Flegal, K.M., 2014. Prevalence of childhood and
adult obesity in the United States, 2011-2012. JAMA 311, 806-814.

WHO, 2006. In: WHO D.o.N.f.H.a.D. (Ed.), WHO Child Growth Standards: Length/Heightfor-
Age, Weight-for-Age, Weight-for-Length, Weight-for-Height and Body Mass Indexfor-
Age: Methods and Development.

WHO, 2010. In: (WHO) W.H.O. (Ed.), WHO Anthro for Personal Computers Manual:
Software for Assessing Growth and Development of the World's Children, Version
3.2.2, 2011. WHO, Geneva.

Valentin, J., 2002. Basic anatomical and physiological data for use in radiological protection:
reference values: ICRP publication 89. Ann. ICRP 32, 1-277.

Johnson, T.N., Rostami-Hodjegan, A., Tucker, G.T., 2006. Prediction of the clearance of
eleven drugs and associated variability in neonates, infants and children. Clin.
Pharmacokinet. 45, 931-956.

Verbraecken, J., Van de Heyning, P., De Backer, W., Van Gaal, L., 2006. Body surface area
in normal-weight, overweight, and obese adults. A comparison study. Metabolism 55,
515-524

Haycock, G.B., Schwartz, G.J., Wisotsky, D.H., 1978. Geometric method for measuring
body surface area: a height-weight formula validated in infants, children, and adults.
J. Pediatr. 93, 62-66.

Lubin, B., 1987. Reference values in infancy and childhood. In: Nathan, D., Oski, F. (Eds.),
Hematology of Infancy and Childhood.

Wambaugh, J.F., Wetmore, B.A., Pearce, R., Strope, C., Goldsmith, R., Sluka, J.P., et al.,
2015. Toxicokinetic triage for environmental chemicals. Toxicol. Sci. 147, 55-67

Ma, M.K., Woo, M.H., Mcleod, H.L., 2002. Genetic basis of drug metabolism. Am. J.
Health Syst. Pharm. 59, 2061-2069.

Yasuda, S.U., Zhang, L., Huang, S.M., 2008. The role of ethnicity in variability in response
to drugs: focus on clinical pharmacology studies. Clin. Pharmacol. Ther. 84, 417-423.

Rotroff, D.M., Wetmore, B.A., Dix, D.J., Ferguson, S.S., Clewell, H.J., Houck, K.A., et al.,
2010. Incorporating human dosimetry and exposure into high-throughput in vitro
toxicity screening. Toxicol. Sci. 117, 348-358.

Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A., Rotroff, D.M., Freeman,
K., et al., 2012. Integration of dosimetry, exposure, and high-throughput screening
data in chemical toxicity assessment. Toxicol. Sci. 125, 157-174.

Wetmore, B.A., Allen, B., Clewell 3rd, H.J., Parker, T., Wambaugh, J.F., Almond, L.M.,
et al., 2014. Incorporating population variability and susceptible subpopulations into
dosimetry for high-throughput toxicity testing. Toxicol. Sci. 142, 210-224.

Wetmore, B.A., Wambaugh, J.F., Allen, B., Ferguson, S.S., Sochaski, M.A., Setzer, R.W.,
et al., 2015. Incorporating high-throughput exposure predictions with Dosimetryadjusted
in vitro bioactivity to inform chemical toxicity testing. Toxicol. Sci. 148,
121-136.
\end{References}
\inputencoding{utf8}
\HeaderA{httkpop\_biotophys\_default}{Convert HTTK-Pop-generated parameters to HTTK physiological parameters}{httkpop.Rul.biotophys.Rul.default}
\keyword{httk-pop}{httkpop\_biotophys\_default}
\keyword{monte-carlo}{httkpop\_biotophys\_default}
%
\begin{Description}\relax
Convert HTTK-Pop-generated parameters to HTTK physiological parameters
\end{Description}
%
\begin{Usage}
\begin{verbatim}
httkpop_biotophys_default(indiv_dt)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{indiv\_dt}] The data.table object returned by \code{httkpop\_generate()}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table with the physiological parameters expected by any HTTK 
model, including body weight (BW), hematocrit, tissue volumes per kg body
weight, tissue flows as fraction of CO, CO per (kg BW)\textasciicircum{}3/4, GFR per (kg
BW)\textasciicircum{}3/4, portal vein flow per (kg BW)\textasciicircum{}3/4, and liver density.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{httkpop\_direct\_resample}{Generate a virtual population by directly resampling the NHANES data.}{httkpop.Rul.direct.Rul.resample}
\keyword{httk-pop}{httkpop\_direct\_resample}
\keyword{monte-carlo}{httkpop\_direct\_resample}
%
\begin{Description}\relax
Generate a virtual population by directly resampling the NHANES data.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
httkpop_direct_resample(
  nsamp = NULL,
  gendernum = NULL,
  agelim_years = c(0, 79),
  agelim_months = c(0, 959),
  weight_category = c("Underweight", "Normal", "Overweight", "Obese"),
  gfr_category = c("Normal", "Kidney Disease", "Kidney Failure"),
  reths = c("Mexican American", "Other Hispanic", "Non-Hispanic White",
    "Non-Hispanic Black", "Other"),
  gfr_resid_var = TRUE,
  ckd_epi_race_coeff = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{nsamp}] The desired number of individuals in the virtual population.
\code{nsamp} need not be provided if \code{gendernum} is provided.

\item[\code{gendernum}] Optional: A named list giving the numbers of male and
female individuals to include in the population, e.g. \code{list(Male=100,
Female=100)}. Default is NULL, meaning both males and females are included,
in their proportions in the NHANES data. If both \code{nsamp} and
\code{gendernum} are provided, they must agree (i.e., \code{nsamp} must be
the sum of \code{gendernum}).

\item[\code{agelim\_years}] Optional: A two-element numeric vector giving the
minimum and maximum ages (in years) to include in the population. Default is
c(0,79). If \code{agelim\_years} is provided and \code{agelim\_months} is not,
\code{agelim\_years} will override the default value of \code{agelim\_months}.

\item[\code{agelim\_months}] Optional: A two-element numeric vector giving the
minimum and maximum ages (in months) to include in the population. Default
is c(0, 959), equivalent to the default \code{agelim\_years}. If
\code{agelim\_months} is provided and \code{agelim\_years} is not,
agelim\_months will override the default values of \code{agelim\_years}.

\item[\code{weight\_category}] Optional: The weight categories to include in the
population. Default is \code{c('Underweight', 'Normal', 'Overweight',
'Obese')}. User-supplied vector must contain one or more of these strings.

\item[\code{gfr\_category}] The kidney function categories to include in the
population. Default is \code{c('Normal','Kidney Disease', 'Kidney Failure')}
to include all kidney function levels.

\item[\code{reths}] Optional: a character vector giving the races/ethnicities to
include in the population. Default is \code{c('Mexican American','Other
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')}, to include
all races and ethnicities in their proportions in the NHANES data.
User-supplied vector must contain one or more of these strings.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table where each row represents an individual, and each
column represents a demographic, anthropometric, or physiological parameter.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{httkpop\_direct\_resample\_inner}{Inner loop function called by \code{httkpop\_direct\_resample}.}{httkpop.Rul.direct.Rul.resample.Rul.inner}
\keyword{httk-pop}{httkpop\_direct\_resample\_inner}
\keyword{monte-carlo}{httkpop\_direct\_resample\_inner}
%
\begin{Description}\relax
Inner loop function called by \code{httkpop\_direct\_resample}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
httkpop_direct_resample_inner(
  nsamp,
  gendernum,
  agelim_months,
  agelim_years,
  reths,
  weight_category,
  gfr_resid_var,
  ckd_epi_race_coeff
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{nsamp}] The desired number of individuals in the virtual population. 
\code{nsamp} need not be provided if \code{gendernum} is provided.

\item[\code{gendernum}] Optional: A named list giving the numbers of male and female 
individuals to include in the population, e.g. \code{list(Male=100, 
Female=100)}. Default is NULL, meaning both males and females are included, 
in their proportions in the NHANES data. If both \code{nsamp} and 
\code{gendernum} are provided, they must agree (i.e., \code{nsamp} must be
the sum of \code{gendernum}).

\item[\code{agelim\_months}] Optional: A two-element numeric vector giving the minimum
and maximum ages (in months) to include in the population. Default is c(0, 
959), equivalent to the default \code{agelim\_years}. If \code{agelim\_months}
is provided and \code{agelim\_years} is not, agelim\_months will override the 
default values of \code{agelim\_years}.

\item[\code{agelim\_years}] Optional: A two-element numeric vector giving the minimum 
and maximum ages (in years) to include in the population. Default is 
c(0,79). If \code{agelim\_years} is provided and \code{agelim\_months} is not,
\code{agelim\_years} will override the default value of \code{agelim\_months}.

\item[\code{reths}] Optional: a character vector giving the races/ethnicities to 
include in the population. Default is \code{c('Mexican American','Other 
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')}, to include 
all races and ethnicities in their proportions in the NHANES data. 
User-supplied vector must contain one or more of these strings.

\item[\code{weight\_category}] Optional: The weight categories to include in the 
population. Default is \code{c('Underweight', 'Normal', 'Overweight', 
'Obese')}. User-supplied vector must contain one or more of these strings.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table where each row represents an individual, and
each column represents a demographic, anthropometric, or physiological
parameter.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{httkpop\_generate}{Generate a virtual population}{httkpop.Rul.generate}
\keyword{httk-pop}{httkpop\_generate}
\keyword{monte-carlo}{httkpop\_generate}
%
\begin{Description}\relax
Generate a virtual population
\end{Description}
%
\begin{Usage}
\begin{verbatim}
httkpop_generate(
  method = "direct resampling",
  nsamp = NULL,
  gendernum = NULL,
  agelim_years = NULL,
  agelim_months = NULL,
  weight_category = c("Underweight", "Normal", "Overweight", "Obese"),
  gfr_category = c("Normal", "Kidney Disease", "Kidney Failure"),
  reths = c("Mexican American", "Other Hispanic", "Non-Hispanic White",
    "Non-Hispanic Black", "Other"),
  gfr_resid_var = TRUE,
  ckd_epi_race_coeff = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{method}] The population-generation method to use. Either "virtual
individuals" or "direct resampling." Short names may be used: "d" or "dr"
for "direct resampling", and "v" or "vi" for "virtual individuals".

\item[\code{nsamp}] The desired number of individuals in the virtual population.
\code{nsamp} need not be provided if \code{gendernum} is provided.

\item[\code{gendernum}] Optional: A named list giving the numbers of male and
female individuals to include in the population, e.g. \code{list(Male=100,
Female=100)}. Default is NULL, meaning both males and females are included,
in their proportions in the NHANES data. If both \code{nsamp} and
\code{gendernum} are provided, they must agree (i.e., \code{nsamp} must be
the sum of \code{gendernum}).

\item[\code{agelim\_years}] Optional: A two-element numeric vector giving the
minimum and maximum ages (in years) to include in the population. Default is
c(0,79). If only a single value is provided, both minimum and maximum ages
will be set to that value; e.g. \code{agelim\_years=3} is equivalent to
\code{agelim\_years=c(3,3)}. If \code{agelim\_years} is provided and
\code{agelim\_months} is not, \code{agelim\_years} will override the default
value of \code{agelim\_months}.

\item[\code{agelim\_months}] Optional: A two-element numeric vector giving the
minimum and maximum ages (in months) to include in the population. Default
is c(0, 959), equivalent to the default \code{agelim\_years}. If only a
single value is provided, both minimum and maximum ages will be set to that
value; e.g. \code{agelim\_months=36} is equivalent to
\code{agelim\_months=c(36,36)}. If \code{agelim\_months} is provided and
\code{agelim\_years} is not, \code{agelim\_months} will override the default
values of \code{agelim\_years}.

\item[\code{weight\_category}] Optional: The weight categories to include in the
population. Default is \code{c('Underweight', 'Normal', 'Overweight',
'Obese')}. User-supplied vector must contain one or more of these strings.

\item[\code{gfr\_category}] The kidney function categories to include in the
population. Default is \code{c('Normal','Kidney Disease', 'Kidney Failure')}
to include all kidney function levels.

\item[\code{reths}] Optional: a character vector giving the races/ethnicities to
include in the population. Default is \code{c('Mexican American','Other
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')}, to include
all races and ethnicities in their proportions in the NHANES data.
User-supplied vector must contain one or more of these strings.

\item[\code{gfr\_resid\_var}] TRUE to add residual variability to GFR predicted from serum creatinine; FALSE to not add residual variability

\item[\code{ckd\_epi\_race\_coeff}] TRUE to use the CKD-EPI equation as originally published (with a coefficient changing predicted GFR for individuals identified as "Non-Hispanic Black"); FALSE to set this coefficient to 1.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table where each row represents an individual, and each
column represents a demographic, anthropometric, or physiological parameter.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
%
\begin{Examples}
\begin{ExampleCode}


#Simply generate a virtual population of 100 individuals,
 #using the direct-resampling method
 set.seed(42)
httkpop_generate(method='direct resampling', nsamp=100)
#Generate a population using the virtual-individuals method,
#including 80 females and 20 males,
#including only ages 20-65,
#including only Mexican American and 
 #Non-Hispanic Black individuals,
 #including only non-obese individuals
httkpop_generate(method = 'virtual individuals',
gendernum=list(Female=80, 
Male=20),
agelim_years=c(20,65), 
reths=c('Mexican American', 
'Non-Hispanic Black'),
weight_category=c('Underweight',
'Normal',
'Overweight'))




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{httkpop\_mc}{Converts the HTTK-Pop population data table to a table of the parameters needed by HTTK, for a specific chemical.}{httkpop.Rul.mc}
\keyword{httk-pop}{httkpop\_mc}
\keyword{monte-carlo}{httkpop\_mc}
%
\begin{Description}\relax
Takes the data table generated by \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}, and converts it
to the corresponding table of HTTK model parameters for a specified chemical
and HTTK model.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
httkpop_mc(model, samples = 1000, httkpop.dt = NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{model}] One of the HTTK models: "1compartment", "3compartmentss",
"3compartment", or "pbtk".

\item[\code{samples}] The number of Monte Carlo samples to use (can often think of these
as separate individuals)

\item[\code{httkpop.dt}] A data table generated by \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.
This defaults to NULL, in which case \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}} is 
called to generate this table.

\item[\code{...}] Additional arugments passed on to \code{\LinkA{httkpop\_generate}{httkpop.Rul.generate}}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table with a row for each individual in the sample and a column for
each parater in the model.
\end{Value}
%
\begin{Author}\relax
Caroline Ring and John Wambaugh
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118

Rowland, Malcolm, Leslie Z. Benet, and Garry G. Graham. "Clearance concepts
in pharmacokinetics." Journal of Pharmacokinetics and Biopharmaceutics 1.2
(1973): 123-136.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

set.seed(42)
indiv_examp <- httkpop_generate(method="d", nsamp=100)
httk_param <- httkpop_mc(httkpop.dt=indiv_examp, 
model="1compartment")

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{httkpop\_virtual\_indiv}{Generate a virtual population by the virtual individuals method.}{httkpop.Rul.virtual.Rul.indiv}
\keyword{httk-pop}{httkpop\_virtual\_indiv}
\keyword{monte-carlo}{httkpop\_virtual\_indiv}
%
\begin{Description}\relax
Generate a virtual population by the virtual individuals method.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
httkpop_virtual_indiv(
  nsamp = NULL,
  gendernum = NULL,
  agelim_years = c(0, 79),
  agelim_months = c(0, 959),
  weight_category = c("Underweight", "Normal", "Overweight", "Obese"),
  gfr_category = c("Normal", "Kidney Disease", "Kidney Failure"),
  reths = c("Mexican American", "Other Hispanic", "Non-Hispanic White",
    "Non-Hispanic Black", "Other"),
  gfr_resid_var = TRUE,
  ckd_epi_race_coeff = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{nsamp}] The desired number of individuals in the virtual population. 
\code{nsamp} need not be provided if \code{gendernum} is provided.

\item[\code{gendernum}] Optional: A named list giving the numbers of male and female 
individuals to include in the population, e.g. \code{list(Male=100, 
Female=100)}. Default is NULL, meaning both males and females are included, 
in their proportions in the NHANES data. If both \code{nsamp} and 
\code{gendernum} are provided, they must agree (i.e., \code{nsamp} must be
the sum of \code{gendernum}).

\item[\code{agelim\_years}] Optional: A two-element numeric vector giving the minimum 
and maximum ages (in years) to include in the population. Default is 
c(0,79). If \code{agelim\_years} is provided and \code{agelim\_months} is not,
\code{agelim\_years} will override the default value of \code{agelim\_months}.

\item[\code{agelim\_months}] Optional: A two-element numeric vector giving the minimum
and maximum ages (in months) to include in the population. Default is c(0, 
959), equivalent to the default \code{agelim\_years}. If \code{agelim\_months}
is provided and \code{agelim\_years} is not, agelim\_months will override the 
default values of \code{agelim\_years}.

\item[\code{weight\_category}] Optional: The weight categories to include in the 
population. Default is \code{c('Underweight', 'Normal', 'Overweight', 
'Obese')}. User-supplied vector must contain one or more of these strings.

\item[\code{gfr\_category}] The kidney function categories to include in the 
population. Default is \code{c('Normal','Kidney Disease', 'Kidney Failure')}
to include all kidney function levels.

\item[\code{reths}] Optional: a character vector giving the races/ethnicities to 
include in the population. Default is \code{c('Mexican American','Other 
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')}, to include 
all races and ethnicities in their proportions in the NHANES data. 
User-supplied vector must contain one or more of these strings.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table where each row represents an individual, and
each column represents a demographic, anthropometric, or physiological
parameter.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{in.list}{Convenience Boolean (yes/no) functions to identify chemical membership in several key lists.}{in.list}
\aliasA{is.expocast}{in.list}{is.expocast}
\aliasA{is.nhanes}{in.list}{is.nhanes}
\methaliasA{is.nhanes.blood.analyte}{in.list}{is.nhanes.blood.analyte}
\methaliasA{is.nhanes.blood.parent}{in.list}{is.nhanes.blood.parent}
\methaliasA{is.nhanes.serum.analyte}{in.list}{is.nhanes.serum.analyte}
\methaliasA{is.nhanes.serum.parent}{in.list}{is.nhanes.serum.parent}
\methaliasA{is.nhanes.urine.analyte}{in.list}{is.nhanes.urine.analyte}
\methaliasA{is.nhanes.urine.parent}{in.list}{is.nhanes.urine.parent}
\aliasA{is.pharma}{in.list}{is.pharma}
\aliasA{is.tox21}{in.list}{is.tox21}
\aliasA{is.toxcast}{in.list}{is.toxcast}
%
\begin{Description}\relax
These functions allow easy identification of whether or not a chemical CAS
is included in various research projects. While it is our intent to keep
these lists up-to-date, the information here is only for convenience and
should not be considered to be definitive.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
in.list(chem.cas = NULL, which.list = "ToxCast")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] The Chemical Abstracts Service Resgistry Number (CAS-RN)
corresponding to the chemical of interest.

\item[\code{which.list}] A character string that can take the following values:
"ToxCast", "Tox21", "ExpoCast", "NHANES", ""NHANES.serum.parent",
"NHANES.serum.analyte","NHANES.blood.parent","NHANES.blood.analyte",
"NHANES.urine.parent","NHANES.urine.analyte"
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Tox21: Toxicology in the 21st Century (Tox21) is a U.S. federal High
Throughput Screening (HTS) collaboration among EPA, NIH, including National
Center for Advancing Translational Sciences and the National Toxicology
Program at the National Institute of Environmental Health Sciences, and the
Food and Drug Administration.  (Bucher et al., 2008)

ToxCast: The Toxicity Forecaster (ToxCast) is a HTS screening project led by
the U.S. EPA to perform additional testing of a subset of Tox21 chemicals.
(Judson et al. 2010)

ExpoCast: ExpoCast (Exposure Forecaster) is an U.S. EPA research project to
generate tenetative exposure estimates (e.g., mg/kg BW/day) for thousands of
chemicals that have little other information using models and informatics.
(Wambaugh et al. 2014)

NHANES: The U.S. Centers for Disease Control (CDC) National Health and
Nutrition Examination Survery (NHANES) is an on-going survey to characterize
the health and biometrics (e.g., weight, height) of the U.S. population. One
set of measurments includes the quantification of xenobiotic chemicals in
various samples (blood, serum, urine) of the thousands of surveyed
individuals. (CDC, 2014)
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{logical}] A Boolean (1/0) value that is TRUE if the chemical is
in the list.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Bucher, J. R. (2008). Guest Editorial: NTP: New Initiatives, New
Alignment. Environ Health Perspect 116(1).

Judson, R. S., Houck, K. A., Kavlock, R. J., Knudsen, T. B., Martin, M. T.,
Mortensen, H. M., Reif, D. M., Rotroff, D. M., Shah, I., Richard, A. M. and
Dix, D. J. (2010). In Vitro Screening of Environmental Chemicals for
Targeted Testing Prioritization: The ToxCast Project. Environmental Health
Perspectives 118(4), 485-492.

Wambaugh, J. F., Wang, A., Dionisio, K. L., Frame, A., Egeghy, P., Judson,
R. and Setzer, R. W. (2014). High Throughput Heuristics for Prioritizing
Human Exposure to Environmental Chemicals. Environmental Science \&
Technology, 10.1021/es503583j.

CDC (2014). National Health and Nutrition Examination Survey. Available at:
https://www.cdc.gov/nchs/nhanes.htm.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{is.httk}{is.httk}} for determining inclusion in httk project
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}


httk.table <- get_cheminfo(info=c("CAS","Compound"))
httk.table[,"Rat"] <- ""
httk.table[,"NHANES"] <- ""
httk.table[,"Tox21"] <- ""
httk.table[,"ToxCast"] <- ""
httk.table[,"ExpoCast"] <- ""
httk.table[,"PBTK"] <- ""
# To make this example run quickly, this loop is only over the first five 
# chemicals. To build a table with all available chemicals use:
# for (this.cas in httk.table$CAS)
for (this.cas in httk.table$CAS[1:5])
{
  this.index <- httk.table$CAS==this.cas
  if (is.nhanes(this.cas)) httk.table[this.index,"NHANES"] <- "Y"
  if (is.tox21(this.cas)) httk.table[this.index,"Tox21"] <- "Y"
  if (is.toxcast(this.cas)) httk.table[this.index,"ToxCast"] <- "Y"
  if (is.expocast(this.cas)) httk.table[this.index,"ExpoCast"] <- "Y"
  if (is.httk(this.cas,model="PBTK")) httk.table[this.index,"PBTK"] <- "Y"
  if (is.httk(this.cas,species="rat")) httk.table[this.index,"Rat"] <- "Y"
}


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{invitro\_mc}{Draw in vitro TK parameters including uncertainty and variability.}{invitro.Rul.mc}
\keyword{in-vitro}{invitro\_mc}
\keyword{monte-carlo}{invitro\_mc}
%
\begin{Description}\relax
Given a CAS in the HTTK data set, a virtual population from HTTK-Pop, some
user specifications on the assumed distributions of Funbound.plasma and
Clint, draw "individual" values of Funbound.plasma and Clint from those
distributions.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
invitro_mc(
  parameters.dt = NULL,
  samples,
  fup.meas.cv = 0.4,
  clint.meas.cv = 0.3,
  fup.pop.cv = 0.3,
  clint.pop.cv = 0.3,
  poormetab = TRUE,
  fup.lod = 0.01,
  fup.censored.dist = FALSE,
  adjusted.Funbound.plasma = TRUE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{parameters.dt}] A data table of physiological parameters

\item[\code{samples}] The number of samples to draw.

\item[\code{fup.meas.cv}] Coefficient of variation of distribution of measured
\code{Funbound.plasma} values.

\item[\code{clint.meas.cv}] Coefficient of variation of distribution of measured 
\code{Clint} values.

\item[\code{fup.pop.cv}] Coefficient of variation of distribution of population
\code{Funbound.plasma} values.

\item[\code{clint.pop.cv}] Coefficient of variation of distribution of population
\code{Clint} values.

\item[\code{poormetab}] Logical. Whether to include poor metabolizers in the Clint
distribution or not.

\item[\code{fup.lod}] The average limit of detection for \code{Funbound.plasma}, below
which distribution will be censored if fup.censored.dist is TRUE. Default 0.01.

\item[\code{fup.censored.dist}] Logical. Whether to draw \code{Funbound.plasma} from a
censored distribution or not.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{parameters}] A list of chemical-specific model parameters containing at
least Funbound.plasma, Clint, and Fhep.assay.correction.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table with three columns: \code{Funbound.plasma} and
\code{Clint}, containing the sampled values, and
\code{Fhep.assay.correction}, containing the value for fraction unbound in
hepatocyte assay.
\end{Value}
%
\begin{Author}\relax
Caroline Ring and John Wambaugh
\end{Author}
%
\begin{References}\relax
Wambaugh, John F., et al. "Assessing Toxicokinetic Uncertainty and 
Variability in Risk Prioritization." Toxicological Sciences (2019).
\end{References}
\inputencoding{utf8}
\HeaderA{is.httk}{Convenience Boolean (yes/no) function to identify chemical membership and treatment within the httk project.}{is.httk}
%
\begin{Description}\relax
Allows easy identification of whether or not a chemical CAS is included in
various aspects of the httk research project (by model type and species of
interest). While it is our intent to keep these lists up-to-date, the 
information here is only for convenience and should not be considered
definitive.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is.httk(chem.cas, species = "Human", model = "3compartmentss")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] The Chemical Abstracts Service Resgistry Number (CAS-RN)
corresponding to the chemical of interest.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{model}] Model used in calculation, 'pbtk' for the multiple compartment
model, '1compartment' for the one compartment model, '3compartment' for
three compartment model, '3compartmentss' for the three compartment model
without partition coefficients, or 'schmitt' for chemicals with logP and
fraction unbound (used in predict\_partitioning\_schmitt).
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Tox21: Toxicology in the 21st Century (Tox21) is a U.S. federal High
Throughput Screening (HTS) collaboration among EPA, NIH, including National
Center for Advancing Translational Sciences and the National Toxicology
Program at the National Institute of Environmental Health Sciences, and the
Food and Drug Administration.  (Bucher et al., 2008)

ToxCast: The Toxicity Forecaster (ToxCast) is a HTS screening project led by
the U.S. EPA to perform additional testing of a subset of Tox21 chemicals.
(Judson et al. 2010)

ExpoCast: ExpoCast (Exposure Forecaster) is an U.S. EPA research project to
generate tenetative exposure estimates (e.g., mg/kg BW/day) for thousands of
chemicals that have little other information using models and informatics.
(Wambaugh et al. 2014)

NHANES: The U.S. Centers for Disease Control (CDC) National Health and
Nutrition Examination Survery (NHANES) is an on-going survey to characterize
the health and biometrics (e.g., weight, height) of the U.S. population. One
set of measurments includes the quantification of xenobiotic chemicals in
various samples (blood, serum, urine) of the thousands of surveyed
individuals. (CDC, 2014)
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{logical}] A Boolean (1/0) value that is TRUE if the chemical
is included in the httk project with a given modeling scheme (PBTK) and 
a given species
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Bucher, J. R. (2008). Guest Editorial: NTP: New Initiatives, New
Alignment. Environ Health Perspect 116(1).

Judson, R. S., Houck, K. A., Kavlock, R. J., Knudsen, T. B., Martin, M. T.,
Mortensen, H. M., Reif, D. M., Rotroff, D. M., Shah, I., Richard, A. M. and
Dix, D. J. (2010). In Vitro Screening of Environmental Chemicals for
Targeted Testing Prioritization: The ToxCast Project. Environmental Health
Perspectives 118(4), 485-492.

Wambaugh, J. F., Wang, A., Dionisio, K. L., Frame, A., Egeghy, P., Judson,
R. and Setzer, R. W. (2014). High Throughput Heuristics for Prioritizing
Human Exposure to Environmental Chemicals. Environmental Science \&
Technology, 10.1021/es503583j.

CDC (2014). National Health and Nutrition Examination Survey. Available at:
https://www.cdc.gov/nchs/nhanes.htm.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{in.list}{in.list}} for determining chemical membership in 
several other key lists
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}


httk.table <- get_cheminfo(info=c("CAS","Compound"))
httk.table[,"Rat"] <- ""
httk.table[,"NHANES"] <- ""
httk.table[,"Tox21"] <- ""
httk.table[,"ToxCast"] <- ""
httk.table[,"ExpoCast"] <- ""
httk.table[,"PBTK"] <- ""
# To make this example run quickly, this loop is only over the first five 
# chemicals. To build a table with all available chemicals use:
# for (this.cas in httk.table$CAS)
for (this.cas in httk.table$CAS[1:5])
{
  this.index <- httk.table$CAS==this.cas
  if (is.nhanes(this.cas)) httk.table[this.index,"NHANES"] <- "Y"
  if (is.tox21(this.cas)) httk.table[this.index,"Tox21"] <- "Y"
  if (is.toxcast(this.cas)) httk.table[this.index,"ToxCast"] <- "Y"
  if (is.expocast(this.cas)) httk.table[this.index,"ExpoCast"] <- "Y"
  if (is.httk(this.cas,model="PBTK")) httk.table[this.index,"PBTK"] <- "Y"
  if (is.httk(this.cas,species="rat")) httk.table[this.index,"Rat"] <- "Y"
}


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{is\_in\_inclusive}{Checks whether a value, or all values in a vector, is within inclusive limits}{is.Rul.in.Rul.inclusive}
\keyword{httk-pop}{is\_in\_inclusive}
%
\begin{Description}\relax
Checks whether a value, or all values in a vector, is within inclusive
limits
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is_in_inclusive(x, lims)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{x}] A numeric value, or vector of values.

\item[\code{lims}] A two-element vector of (min, max) values for the inclusive
limits. If \code{x} is a vector, \code{lims} may also be a two-column matrix
with \code{nrow=length(x)} where the first column is lower limits and the
second column is upper limits. If \code{x} is a vector and \code{lims} is a
two-element vector, then each element of \code{x} will be checked against
the same limits. If \code{x} is a vector and \code{lims} is a matrix, then
each element of \code{x} will be checked against the limits given by the
corresponding row of \code{lims}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A logical vector the same length as \code{x}, indicating whether
each element of \code{x} is within the inclusive limits given by
\code{lims}.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{johnson}{Johnson 2006}{johnson}
\keyword{data}{johnson}
%
\begin{Description}\relax
This data set is only used in Vignette 5.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
johnson
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table containing 60 rows and 11 columns.
\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Johnson, Trevor N., Amin Rostami-Hodjegan, and Geoffrey T. Tucker.
"Prediction of the clearance of eleven drugs and associated variability in
neonates, infants and children." Clinical pharmacokinetics 45.9 (2006):
931-956.
\end{References}
\inputencoding{utf8}
\HeaderA{kidney\_mass\_children}{Predict kidney mass for children}{kidney.Rul.mass.Rul.children}
\keyword{httk-pop}{kidney\_mass\_children}
%
\begin{Description}\relax
For individuals under age 18, predict kidney mass from weight, height, and
gender. using equations from Ogiu et al. 1997
\end{Description}
%
\begin{Usage}
\begin{verbatim}
kidney_mass_children(weight, height, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{weight}] Vector of weights in kg.

\item[\code{height}] Vector of heights in cm.

\item[\code{gender}] Vector of genders (either 'Male' or 'Female').
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of kidney masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ogiu, Nobuko, et al. "A statistical analysis of the internal 
organ weights of normal Japanese people." Health physics 72.3 (1997): 368-383.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{liver\_mass\_children}{Predict liver mass for children}{liver.Rul.mass.Rul.children}
\keyword{httk-pop}{liver\_mass\_children}
%
\begin{Description}\relax
For individuals under 18, predict the liver mass from height, weight, and
gender, using equations from Ogiu et al. 1997
\end{Description}
%
\begin{Usage}
\begin{verbatim}
liver_mass_children(height, weight, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{height}] Vector of heights in cm.

\item[\code{weight}] Vector of weights in kg.

\item[\code{gender}] Vector of genders (either 'Male' or 'Female').
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of liver masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ogiu, Nobuko, et al. "A statistical analysis of the internal 
organ weights of normal Japanese people." Health physics 72.3 (1997): 368-383.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{load\_dawson2021}{Load data from Dawson et al. 2021.}{load.Rul.dawson2021}
%
\begin{Description}\relax
This function returns an updated version of chem.physical\_and\_invitro.data
that includes data predicted with Random Forest QSAR models developed and
presented in Dawson et al. 2021, included in dawson2021.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
load_dawson2021(overwrite = FALSE, exclude_oad = TRUE, target.env = .GlobalEnv)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{overwrite}] Only matters if load.image=FALSE. If overwrite=TRUE then
existing data in chem.physical\_and\_invitro.data will be replaced by any
data/predictions in Dawson et al. (2021) that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.

\item[\code{exclude\_oad}] Include the chemicals only within the applicability domain.
If exlude\_oad=TRUE (DEFAULT) chemicals outside the applicability domain do not
have their predicted values loaded.

\item[\code{target.env}] The environment where the new
chem.physical\_and\_invitro.data is loaded. Defaults to global environment.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{data.frame}] An updated version of
chem.physical\_and\_invitro.data.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Sarah E. Davidson
\end{Author}
%
\begin{References}\relax
\bsl{}insertRefdawson2021qsarhttk
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

## Not run: 
chem.physical_and_invitro.data <- load_dawson2021()
chem.physical_and_invitro.data <- load_dawson2021(overwrite=TRUE) 

## End(Not run)                        

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{load\_sipes2017}{Load data from Sipes et al 2017.}{load.Rul.sipes2017}
%
\begin{Description}\relax
This function returns an updated version of chem.physical\_and\_invitro.data
that includes data predicted with Simulations Plus' ADMET predictor that was
used in Sipes et al. 2017, included in admet.data.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
load_sipes2017(overwrite = FALSE, target.env = .GlobalEnv)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{overwrite}] Only matters if load.image=FALSE. If overwrite=TRUE then
existing data in chem.physical\_and\_invitro.data will be replaced by any
data/predictions in Sipes et al. (2017) that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.

\item[\code{target.env}] The environment where the new
chem.physical\_and\_invitro.data is loaded. Defaults to global environment.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{data.frame}] An updated version of
chem.physical\_and\_invitro.data.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
%
\begin{References}\relax
Sipes, Nisha S., et al. "An intuitive approach for predicting
potential human health risk with the Tox21 10k library." Environmental
Science \& Technology 51.18 (2017): 10786-10796.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}


num.chems <- length(get_cheminfo())
load_sipes2017()

#We should have the ADMet Predicted chemicals from Sipes et al. (2017),
#this one is a good test since the logP is nearly 10
calc_css(chem.cas="26040-51-7")

#Let's see how many chemicals we have now with the Sipes (2017) data loaded:
length(get_cheminfo())

#Now let us reset
reset_httk()

# We should be back to our original number:
num.chems == length(get_cheminfo())
                        

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{lump\_tissues}{Lump tissue parameters}{lump.Rul.tissues}
\keyword{Parameter}{lump\_tissues}
%
\begin{Description}\relax
This function takes the parameters from predict\_partitioning\_schmitt and 
lumps the partition coefficients along with the volumes and flows based on 
the given tissue list. It is useful in Monte Carlo simulation of individual
partition coefficients when calculating the rest of body partition
coefficient.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
lump_tissues(
  Ktissue2pu.in,
  parameters = NULL,
  tissuelist = NULL,
  species = "Human",
  tissue.vols = NULL,
  tissue.flows = NULL,
  model = "pbtk",
  suppress.messages = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{Ktissue2pu.in}] List of partition coefficients from
predict\_partitioning\_schmitt.

\item[\code{parameters}] A list of physiological parameters including flows and
volumes for tissues in \code{tissuelist}

\item[\code{tissuelist}] Manually specifies compartment names and tissues, which
override the standard compartment names and tissues that are usually
specified in a model's associated modelinfo file. Remaining tissues in the
model's associated \code{alltissues} listing are lumped in the rest of the body.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{tissue.vols}] A list of volumes for tissues in \code{tissuelist}

\item[\code{tissue.flows}] A list of flows for tissues in \code{tissuelist}

\item[\code{model}] Specify which model (and therefore which tissues) are being 
considered

\item[\code{suppress.messages}] Whether or not the output message is suppressed.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
This function returns the flows, volumes, and partition coefficients for the
lumped tissues specified in tissue list Ktissue2plasma -- tissue to free
plasma concentration partition coefficients for every tissue specified by 
Schmitt (2008) (the tissue.data table) tissuelist -- a list of character 
vectors, the name of each entry in the list is its own compartment.
The tissues in the alltissues vector are the Schmitt (2008) tissues that are
to be considered in the lumping process. The tissuelist can also be manually
specified for alternate lumping schemes: for example,
tissuelist<-list(Rapid=c("Brain","Kidney")) specifies the flow.col and
vol.col in the tissuedata.table.
\end{Details}
%
\begin{Value}
\begin{ldescription}
\item[\code{Krbc2pu}] Ratio of concentration of chemical in red blood cells
to unbound concentration in plasma.\item[\code{Krest2pu}] Ratio of concentration
of chemical in rest of body tissue to unbound concentration in plasma.
\item[\code{Vrestc}]  Volume of the rest of the body per kg body weight, L/kg BW.
\item[\code{Vliverc}]  Volume of the liver per kg body weight, L/kg BW.
\item[\code{Qtotal.liverf}] Fraction of cardiac output flowing to the gut and
liver, i.e. out of the liver.\item[\code{Qgutf}] Fraction of cardiac output
flowing to the gut.\item[\code{Qkidneyf}] Fraction of cardiac output flowing to
the kidneys.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Evaluation and calibration of 
high-throughput predictions of chemical distribution to tissues." Journal of
pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

pcs <- predict_partitioning_schmitt(chem.name='bisphenola')
tissuelist <- list(liver=c("liver"),kidney=c("kidney"),lung=c("lung"),gut=c("gut")
,muscle.bone=c('muscle','bone'))
lump_tissues(pcs,tissuelist=tissuelist)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{lung\_mass\_children}{Predict lung mass for children}{lung.Rul.mass.Rul.children}
\keyword{httk-pop}{lung\_mass\_children}
%
\begin{Description}\relax
For individuals under 18, predict the liver mass from height, weight, and
gender, using equations from Ogiu et al. 1997
\end{Description}
%
\begin{Usage}
\begin{verbatim}
lung_mass_children(height, weight, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{height}] Vector of heights in cm.

\item[\code{weight}] Vector of weights in kg.

\item[\code{gender}] Vector of genders (either 'Male' or 'Female').
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of lung masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ogiu, Nobuko, et al. "A statistical analysis of the internal 
organ weights of normal Japanese people." Health physics 72.3 (1997): 368-383.

Price, Paul S., et al. "Modeling interindividual variation in physiological 
factors used in PBPK models of humans." Critical reviews in toxicology 33.5 
(2003): 469-503.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{mcnally\_dt}{Reference tissue masses and flows from tables in McNally et al. 2014.}{mcnally.Rul.dt}
\keyword{data}{mcnally\_dt}
\keyword{httk-pop}{mcnally\_dt}
%
\begin{Description}\relax
Reference tissue masses, flows, and marginal distributions from McNally et
al. 2014.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
mcnally_dt
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table with variables: \begin{description}
\item[\code{tissue}] Body
tissue\item[\code{gender}] Gender: Male or Female
\item[\code{mass\_ref}] Reference mass in kg, from Reference Man
\item[\code{mass\_cv}] Coefficient of variation for mass
\item[\code{mass\_dist}] Distribution for mass: Normal or Log-normal
\item[\code{flow\_ref}] Reference flow in L/h, from Reference Man
\item[\code{flow\_cv}] Coefficient of variation for flow (all normally
distributed)\item[\code{height\_ref}] Reference heights (by gender)
\item[\code{CO\_ref}] Reference cardiac output by gender
\item[\code{flow\_frac}] Fraction of CO flowing to each tissue:
\code{flow\_ref}/\code{CO\_ref}
\end{description}

\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{Source}\relax
McNally K, Cotton R, Hogg A, Loizou G. "PopGen: A virtual human
population generator." Toxicology 315, 70-85, 2004.
\end{Source}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{metabolism\_data\_Linakis2020}{Metabolism data involved in Linakis 2020 vignette analysis.}{metabolism.Rul.data.Rul.Linakis2020}
\keyword{data}{metabolism\_data\_Linakis2020}
%
\begin{Description}\relax
Metabolism data involved in Linakis 2020 vignette analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
metabolism_data_Linakis2020
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing x rows and y columns.
\end{Format}
%
\begin{Author}\relax
Matt Linakis
\end{Author}
%
\begin{Source}\relax
Matt Linakis
\end{Source}
%
\begin{References}\relax
DSStox database (https:// www.epa.gov/ncct/dsstox
\end{References}
\inputencoding{utf8}
\HeaderA{monte\_carlo}{Monte Carlo for pharmacokinetic models}{monte.Rul.carlo}
\keyword{Monte-Carlo}{monte\_carlo}
%
\begin{Description}\relax
This function performs Monte Carlo to assess uncertainty and variability for
toxicokinetic models.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
monte_carlo(
  parameters,
  cv.params = NULL,
  censored.params = NULL,
  samples = 1000
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{parameters}] These parameters that are also listed in either
cv.params or censored.params are sampled using Monte Carlo.

\item[\code{cv.params}] The parameters listed in cv.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (cv) for the normal distribution. Each
entry in the list is named for a parameter in "parameters". New values are
sampled with mean equal to the value in "parameters" and standard deviation
equal to the mean times the cv.

\item[\code{censored.params}] The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sub-lists. Each sublist is named for a parameter in "params"
and contains two elements: "cv" (coefficient of variation) and "LOD" (limit
of detection), below which parameter values are censored. New values are
sampled with mean equal to the value in "params" and standard deviation
equal to the mean times the cv. Censored values are sampled on a uniform
distribution between 0 and the limit of detection.

\item[\code{samples}] This argument is the number of samples to be generated for
calculating quantiles.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table with a row for each individual in the sample and a column for
each parater in the model.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput 
toxicokinetics." Journal of statistical software 79.4 (2017): 1.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

#Example based on Pearce et al. (2017):

# Set up means:
params <- parameterize_pbtk(chem.name="zoxamide")
# Nothing changes:
monte_carlo(params)

vary.params <- NULL
for (this.param in names(params)[!(names(params) %in%
  c("Funbound.plasma", "pKa_Donor", "pKa_Accept" )) &
  !is.na(as.numeric(params))]) vary.params[this.param] <- 0.2
# Most everything varies with CV of 0.2:
monte_carlo(
  parameters=params, 
  cv.params = vary.params)

censored.params <- list(Funbound.plasma = list(cv = 0.2, lod = 0.01))
# Fup is censored below 0.01:
monte_carlo(
  parameters=params, 
  cv.params = vary.params,
  censored.params = censored.params)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{nhanes\_mec\_svy}{Pre-processed NHANES data.}{nhanes.Rul.mec.Rul.svy}
\keyword{data}{nhanes\_mec\_svy}
\keyword{httk-pop}{nhanes\_mec\_svy}
%
\begin{Description}\relax
NHANES data on demographics, anthropometrics, and some laboratory measures,
cleaned and combined into a single data set.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
nhanes_mec_svy
\end{verbatim}
\end{Usage}
%
\begin{Format}
A survey.design2 object, including masked cluster and strata.
Variables are available as a data.table by \code{nhanes\_mec\_svy\$variables}.
Variables are as described in NHANES Demographics and Examination
documentation, with the exception of: \begin{description}

\item[\code{wtmec6yr}] 6-year sample weights for combining 3 cycles,
computed by dividing 2-year sample weights by 3.
\item[\code{bmxhtlenavg}] Average of height and recumbent length if both
were measured; if only one was measured, takes value of the one that was
measured.\item[\code{logbmxwt}] Natural log of measured body weight.
\item[\code{logbmxhtlenavg}] Natural log of \code{bmxhtlenavg}.
\item[\code{weight\_class}] One of Underweight, Normal, Overweight, or Obese.
Assigned using methods in \code{get\_weight\_class}.
\end{description}

\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{Source}\relax
\url{https://wwwn.cdc.gov/nchs/nhanes/Default.aspx}
\end{Source}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{Obach2008}{Published Pharmacokinetic Parameters from Obach et al. 2008}{Obach2008}
\keyword{data}{Obach2008}
%
\begin{Description}\relax
This data set is used in Vignette 4 for steady state concentration.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Obach2008
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 670 rows and 8 columns.
\end{Format}
%
\begin{References}\relax
Obach, R. Scott, Franco Lombardo, and Nigel J. Waters. "Trend
analysis of a database of intravenous pharmacokinetic parameters in humans
for 670 drug compounds." Drug Metabolism and Disposition 36.7 (2008):
1385-1405.
\end{References}
\inputencoding{utf8}
\HeaderA{onlyp}{NHANES Exposure Data}{onlyp}
\keyword{data}{onlyp}
%
\begin{Description}\relax
This data set is only used in Vignette 6.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
onlyp
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table containing 1060 rows and 5 columns.
\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Wambaugh, John F., et al. "High throughput heuristics for prioritizing human
exposure to environmental chemicals." Environmental science \& technology
48.21 (2014): 12760-12767.
\end{References}
\inputencoding{utf8}
\HeaderA{pancreas\_mass\_children}{Predict pancreas mass for children}{pancreas.Rul.mass.Rul.children}
\keyword{httk-pop}{pancreas\_mass\_children}
%
\begin{Description}\relax
For individuals under 18, predict the pancreas mass from height, weight, and
gender, using equations from Ogiu et al.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pancreas_mass_children(height, weight, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{height}] Vector of heights in cm.

\item[\code{weight}] Vector of weights in kg.

\item[\code{gender}] Vector of genders (either 'Male' or 'Female').
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of pancreas masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ogiu, Nobuko, et al. "A statistical analysis of the internal 
organ weights of normal Japanese people." Health physics 72.3 (1997): 368-383.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{parameterize\_1comp}{Parameterize\_1comp}{parameterize.Rul.1comp}
\keyword{1compartment}{parameterize\_1comp}
\keyword{Parameter}{parameterize\_1comp}
%
\begin{Description}\relax
This function initializes the parameters needed in the function solve\_1comp.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
parameterize_1comp(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  well.stirred.correction = TRUE,
  suppress.messages = FALSE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing rat values with human values if
true.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with volume of distribution calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients in volume of distribution calculation.

\item[\code{restrictive.clearance}] In calculating elimination rate and hepatic
bioavailability, protein binding is not taken into account (set to 1) in
liver clearance if FALSE.

\item[\code{well.stirred.correction}] Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.

\item[\code{suppress.messages}] Whether or not to suppress messages.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-value greater than the threshold are
set to zero.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{Vdist}] Volume of distribution, units of L/kg BW.
\item[\code{Fgutabs}] Fraction of the oral dose absorbed, i.e. the fraction of the
dose that enters the gutlumen.
\item[\code{Fhep.assay.correction}] The fraction of chemical unbound in hepatocyte 
assay using the method of Kilford et al. (2008)
\item[\code{kelim}] Elimination rate, units of 1/h.
\item[\code{hematocrit}] Percent volume of red blood cells in the blood.
\item[\code{kgutabs}] Rate chemical is absorbed, 1/h.
\item[\code{million.cells.per.gliver}] Millions cells per gram of liver tissue.
\item[\code{MW}] Molecular Weight, g/mol.
\item[\code{Rblood2plasma}] The ratio of the concentration of the chemical in the 
blood to the concentration in the plasma. Not used in calculations but 
included for the conversion of plasma outputs.
\item[\code{hepatic.bioavailability}] Fraction of dose remaining after
first pass clearance, calculated from the corrected well-stirred model.
\item[\code{BW}] Body Weight, kg.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput 
toxicokinetics." Journal of statistical software 79.4 (2017): 1.

Kilford, P. J., Gertz, M., Houston, J. B. and Galetin, A.
(2008). Hepatocellular binding of drugs: correction for unbound fraction in
hepatocyte incubations using microsomal binding or drug lipophilicity data.
Drug Metabolism and Disposition 36(7), 1194-7, 10.1124/dmd.108.020834.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

 parameters <- parameterize_1comp(chem.name='Bisphenol-A',species='Rat')
 parameters <- parameterize_1comp(chem.cas='80-05-7',
                                  restrictive.clearance=FALSE,
                                  species='rabbit',
                                  default.to.human=TRUE)
 out <- solve_1comp(parameters=parameters)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{parameterize\_3comp}{Parameterize\_3comp}{parameterize.Rul.3comp}
\keyword{3compartment}{parameterize\_3comp}
\keyword{Parameter}{parameterize\_3comp}
%
\begin{Description}\relax
This function initializes the parameters needed in the function solve\_3comp.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
parameterize_3comp(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = F,
  force.human.clint.fup = F,
  clint.pvalue.threshold = 0.05,
  adjusted.Funbound.plasma = T,
  regression = T,
  suppress.messages = F,
  restrictive.clearance = T,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
-- the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true.

\item[\code{force.human.clint.fup}] Forces use of human values for hepatic
intrinsic clearance and fraction of unbound plasma if true.

\item[\code{clint.pvalue.threshold}] Hepatic clearances with clearance assays
having p-values greater than the threshold are set to zero.

\item[\code{adjusted.Funbound.plasma}] Returns adjusted Funbound.plasma when set to
TRUE along with parition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{restrictive.clearance}] In calculating hepatic.bioavailability, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{BW}] Body Weight, kg.
\item[\code{Clmetabolismc}] Hepatic Clearance, L/h/kg BW.
\item[\code{Fgutabs}] Fraction of the oral dose absorbed, i.e. the 
fraction of the dose that enters the gutlumen.
\item[\code{Funbound.plasma}] Fraction of plasma that is not bound.
\item[\code{Fhep.assay.correction}] The fraction of chemical unbound in hepatocyte 
assay using the method of Kilford et al. (2008)
\item[\code{hematocrit}] Percent volume of red blood cells in the blood.
\item[\code{Kgut2pu}] Ratio of concentration of chemical in gut tissue to unbound
concentration in plasma.
\item[\code{Kliver2pu}] Ratio of concentration of
chemical in liver tissue to unbound concentration in plasma.
\item[\code{Krbc2pu}] Ratio of concentration of chemical in red blood cells to
unbound concentration in plasma.
\item[\code{Krest2pu}] Ratio of concentration of
chemical in rest of body tissue to unbound concentration in plasma.
\item[\code{million.cells.per.gliver}] Millions cells per gram of liver tissue.
\item[\code{MW}] Molecular Weight, g/mol.
\item[\code{Qcardiacc}] Cardiac Output, L/h/kg
BW\textasciicircum{}3/4.\item[\code{Qgfrc}] Glomerular Filtration Rate, L/h/kg BW\textasciicircum{}3/4, volume of
fluid filtered from kidney and excreted.
\item[\code{Qgutf}] Fraction of cardiac output flowing to the gut.
\item[\code{Qliverf}] Fraction of cardiac output flowing to the liver.
\item[\code{Rblood2plasma}] The ratio of the concentration
of the chemical in the blood to the concentration in the plasma.
\item[\code{Vgutc}] Volume of the gut per kg body weight, L/kg BW.
\item[\code{Vliverc}] Volume of the liver per kg body weight, L/kg BW.
\item[\code{Vrestc}]  Volume of the rest of the body per kg body weight, L/kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput 
toxicokinetics." Journal of statistical software 79.4 (2017): 1.

Kilford, P. J., Gertz, M., Houston, J. B. and Galetin, A.
(2008). Hepatocellular binding of drugs: correction for unbound fraction in
hepatocyte incubations using microsomal binding or drug lipophilicity data.
Drug Metabolism and Disposition 36(7), 1194-7, 10.1124/dmd.108.020834.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

 parameters <- parameterize_3comp(chem.name='Bisphenol-A',species='Rat')
 parameters <- parameterize_3comp(chem.cas='80-05-7',
                                  species='rabbit',default.to.human=TRUE)
 out <- solve_3comp(parameters=parameters,plots=TRUE)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{parameterize\_gas\_pbtk}{Parameterize\_gas\_pbtk}{parameterize.Rul.gas.Rul.pbtk}
\keyword{Parameter}{parameterize\_gas\_pbtk}
%
\begin{Description}\relax
This function initializes the parameters needed in the function solve\_gas\_pbtk
\end{Description}
%
\begin{Usage}
\begin{verbatim}
parameterize_gas_pbtk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  tissuelist = list(liver = c("liver"), kidney = c("kidney"), lung = c("lung"), gut =
    c("gut")),
  force.human.clint.fup = F,
  clint.pvalue.threshold = 0.05,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  vmax = 0,
  km = 1,
  exercise = F,
  fR = 12,
  VT = 0.75,
  VD = 0.15,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Either the chemical name or the CAS number must be
specified.

\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{tissuelist}] Specifies compartment names and tissues groupings.
Remaining tissues in tissue.data are lumped in the rest of the body.
However, solve\_pbtk only works with the default parameters.

\item[\code{force.human.clint.fup}] Forces use of human values for hepatic
intrinsic clearance and fraction of unbound plasma if true.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.

\item[\code{adjusted.Funbound.plasma}] Returns adjusted Funbound.plasma when set to
TRUE along with parition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{vmax}] Michaelis-Menten vmax value in reactions/min

\item[\code{km}] Michaelis-Menten concentration of half-maximal reaction velocity
in desired output concentration units.

\item[\code{exercise}] Logical indicator of whether to simulate an exercise-induced
heightened respiration rate

\item[\code{fR}] Respiratory frequency (breaths/minute), used especially to adjust
breathing rate in the case of exercise. This parameter, along with VT and VD
(below) gives another option for calculating Qalv (Alveolar ventilation) 
in case pulmonary ventilation rate is not known

\item[\code{VT}] Tidal volume (L), to be modulated especially as part of simulating
the state of exercise

\item[\code{VD}] Anatomical dead space (L), to be modulated especially as part of
simulating the state of exercise

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{...}] Other parameters
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{BW}] Body Weight, kg.
\item[\code{Clint}] Hepatic intrinsic clearance, uL/min/10\textasciicircum{}6 cells
\item[\code{Clint.dist}] Distribution of hepatic intrinsic clearance values
(median, lower 95th, upper 95th, p value)
\item[\code{Clmetabolismc}] Hepatic Clearance, L/h/kg BW.
\item[\code{Fgutabs}] Fraction of the oral dose absorbed, i.e. the fraction of the
dose that enters the gut lumen.
\item[\code{Fhep.assay.correction}] The fraction of chemical unbound in hepatocyte
assay using the method of Kilford et al. (2008)
\item[\code{Funbound.plasma}] Fraction of chemical unbound to plasma.
\item[\code{Funbound.plasma.adjustment}] Fraction unbound to plasma adjusted as
described in Pearce et al. 2017
\item[\code{Funbound.plasma.dist}] Distribution of fraction unbound to plasma
(median, lower 95th, upper 95th)
\item[\code{hematocrit}] Percent volume of red blood cells in the blood.
\item[\code{Kblood2air}] Ratio of concentration of chemical in blood to air
\item[\code{Kgut2pu}] Ratio of concentration of chemical in gut tissue to unbound
concentration in plasma.
\item[\code{kgutabs}] Rate that chemical enters the gut from gutlumen, 1/h.
\item[\code{Kkidney2pu}] Ratio of concentration of chemical in kidney tissue to
unbound concentration in plasma.
\item[\code{Kliver2pu}] Ratio of concentration of chemical in liver tissue to
unbound concentration in plasma.
\item[\code{Klung2pu}] Ratio of concentration of chemical in lung tissue
to unbound concentration in plasma.
\item[\code{km}] Michaelis-Menten concentration of half-maximal activity
\item[\code{Kmuc2air}] Mucus to air partition coefficient
\item[\code{Krbc2pu}] Ratio of concentration of chemical in red blood cells to
unbound concentration in plasma.
\item[\code{Krest2pu}] Ratio of concentration of chemical in rest of body tissue to
unbound concentration in plasma.
\item[\code{kUrtc}] Unscaled upper respiratory tract uptake parameter (L/h/kg\textasciicircum{}0.75)
\item[\code{liver.density}] Density of liver in g/mL
\item[\code{MA}] phospholipid:water distribution coefficient, membrane affinity
\item[\code{million.cells.per.gliver}] Millions cells per gram of liver tissue.
\item[\code{MW}] Molecular Weight, g/mol.
\item[\code{pKa\_Accept}] compound H association equilibrium constant(s)
\item[\code{pKa\_Donor}] compound H dissociation equilibirum constant(s)
\item[\code{Pow}] octanol:water partition coefficient (not log transformed)
\item[\code{Qalvc}] Unscaled alveolar ventilation rate (L/h/kg\textasciicircum{}0.75)
\item[\code{Qcardiacc}] Cardiac Output, L/h/kg BW\textasciicircum{}3/4.
\item[\code{Qgfrc}] Glomerular Filtration Rate, L/h/kg BW\textasciicircum{}0.75, volume of fluid
filtered from kidney and excreted.
\item[\code{Qgutf}] Fraction of cardiac output flowing to the gut.
\item[\code{Qkidneyf}] Fraction of cardiac output flowing to the kidneys.
\item[\code{Qliverf}] Fraction of cardiac output flowing to the liver.
\item[\code{Qlungf}] Fraction of cardiac output flowing to lung tissue.
\item[\code{Qrestf}] Fraction of blood flow to rest of body
\item[\code{Rblood2plasma}] The ratio of the concentration of the chemical in the
blood to the concentration in the plasma from available\_rblood2plasma.
\item[\code{Vartc}] Volume of the arteries per kg body weight, L/kg BW.
\item[\code{Vgutc}] Volume of the gut per kg body weight, L/kg BW.
\item[\code{Vkidneyc}] Volume of the kidneys per kg body weight, L/kg BW.
\item[\code{Vliverc}] Volume of the liver per kg body weight, L/kg BW.
\item[\code{Vlungc}] Volume of the lungs per kg body weight, L/kg BW.
\item[\code{vmax}] Michaelis-Menten maximum reaction velocity (1/min)
\item[\code{Vmucc}] Unscaled mucosal volume (L/kg BW\textasciicircum{}0.75
\item[\code{Vrestc}]  Volume of the rest of the body per kg body weight, L/kg BW.
\item[\code{Vvenc}] Volume of the veins per kg body weight, L/kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Matt Linakis, Robert Pearce, John Wambaugh
\end{Author}
%
\begin{References}\relax
Linakis, Matthew W., et al. "Development and Evaluation of a High Throughput 
Inhalation Model for Organic Chemicals", submitted

Kilford, P. J., Gertz, M., Houston, J. B. and Galetin, A.
(2008). Hepatocellular binding of drugs: correction for unbound fraction in
hepatocyte incubations using microsomal binding or drug lipophilicity data.
Drug Metabolism and Disposition 36(7), 1194-7, 10.1124/dmd.108.020834.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
parameters <- parameterize_gas_pbtk(chem.cas='129-00-0')

parameters <- parameterize_gas_pbtk(chem.name='pyrene',species='Rat')

parameterize_gas_pbtk(chem.cas = '56-23-5')

parameters <- parameterize_gas_pbtk(chem.name='Carbon tetrachloride',species='Rat')

# Change the tissue lumping:
compartments <- list(liver=c("liver"),fast=c("heart","brain","muscle","kidney"),
                      lung=c("lung"),gut=c("gut"),slow=c("bone"))
parameterize_gas_pbtk(chem.name="Bisphenol a",species="Rat",default.to.human=TRUE,
                   tissuelist=compartments) 

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{parameterize\_pbtk}{Parameterize\_PBTK}{parameterize.Rul.pbtk}
\keyword{Parameter}{parameterize\_pbtk}
\keyword{pbtk}{parameterize\_pbtk}
%
\begin{Description}\relax
This function initializes the parameters needed in the functions solve\_pbtk,
calc\_css, and others using the multiple compartment model.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
parameterize_pbtk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  tissuelist = list(liver = c("liver"), kidney = c("kidney"), lung = c("lung"), gut =
    c("gut")),
  force.human.clint.fup = F,
  clint.pvalue.threshold = 0.05,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  suppress.messages = FALSE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{tissuelist}] Specifies compartment names and tissues groupings.
Remaining tissues in tissue.data are lumped in the rest of the body.
However, solve\_pbtk only works with the default parameters.

\item[\code{force.human.clint.fup}] Forces use of human values for hepatic
intrinsic clearance and fraction of unbound plasma if true.

\item[\code{clint.pvalue.threshold}] Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.

\item[\code{adjusted.Funbound.plasma}] Returns adjusted Funbound.plasma when set to
TRUE along with parition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{restrictive.clearance}] In calculating hepatic.bioavailability, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{BW}] Body Weight, kg.
\item[\code{Clmetabolismc}] Hepatic Clearance, L/h/kg BW.
\item[\code{Fgutabs}] Fraction of the oral dose absorbed, i.e. the fraction of
the dose that enters the gutlumen.
\item[\code{Funbound.plasma}] Fraction of plasma that is not bound.
\item[\code{Fhep.assay.correction}] The fraction of chemical unbound in hepatocyte 
assay using the method of Kilford et al. (2008)
\item[\code{hematocrit}] Percent volume of red blood cells in the blood.
\item[\code{Kgut2pu}] Ratio of concentration of chemical in gut tissue to unbound
concentration in plasma.
\item[\code{kgutabs}] Rate that chemical enters the gut from gutlumen, 1/h.
\item[\code{Kkidney2pu}] Ratio of concentration of chemical in
kidney tissue to unbound concentration in plasma.
\item[\code{Kliver2pu}] Ratio of concentration of chemical in liver tissue to 
unbound concentration in plasma.
\item[\code{Klung2pu}] Ratio of concentration of chemical in lung tissue
to unbound concentration in plasma.
\item[\code{Krbc2pu}] Ratio of concentration
of chemical in red blood cells to unbound concentration in plasma.
\item[\code{Krest2pu}] Ratio of concentration of chemical in rest of body tissue to
unbound concentration in plasma.
\item[\code{million.cells.per.gliver}] Millions cells per gram of liver tissue.
\item[\code{MW}] Molecular Weight, g/mol.
\item[\code{Qcardiacc}] Cardiac Output, L/h/kg BW\textasciicircum{}3/4.
\item[\code{Qgfrc}] Glomerular Filtration Rate, L/h/kg BW\textasciicircum{}3/4, volume of fluid 
filtered from kidney and excreted.
\item[\code{Qgutf}] Fraction of cardiac output flowing to the gut.
\item[\code{Qkidneyf}] Fraction of cardiac output flowing to the kidneys.
\item[\code{Qliverf}] Fraction of cardiac output flowing to the liver.
\item[\code{Rblood2plasma}] The ratio of the concentration of the chemical in the
blood to the concentration in the plasma from available\_rblood2plasma.
\item[\code{Vartc}] Volume of the arteries per kg body weight, L/kg BW.
\item[\code{Vgutc}] Volume of the gut per kg body weight, L/kg BW.
\item[\code{Vkidneyc}] Volume of the kidneys per kg body weight, L/kg BW.
\item[\code{Vliverc}] Volume of the liver per kg body weight, L/kg BW.
\item[\code{Vlungc}] Volume of the lungs per kg body weight, L/kg BW.
\item[\code{Vrestc}]  Volume of the rest of the body per kg body weight, L/kg BW.
\item[\code{Vvenc}] Volume of the veins per kg body weight, L/kg BW.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput 
toxicokinetics." Journal of statistical software 79.4 (2017): 1.

Kilford, P. J., Gertz, M., Houston, J. B. and Galetin, A.
(2008). Hepatocellular binding of drugs: correction for unbound fraction in
hepatocyte incubations using microsomal binding or drug lipophilicity data.
Drug Metabolism and Disposition 36(7), 1194-7, 10.1124/dmd.108.020834.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

 parameters <- parameterize_pbtk(chem.cas='80-05-7')

 parameters <- parameterize_pbtk(chem.name='Bisphenol-A',species='Rat')

 # Change the tissue lumping (note, these model parameters will not work with our current solver):
 compartments <- list(liver=c("liver"),fast=c("heart","brain","muscle","kidney"),
                      lung=c("lung"),gut=c("gut"),slow=c("bone"))
 parameterize_pbtk(chem.name="Bisphenol a",species="Rat",default.to.human=TRUE,
                   tissuelist=compartments) 

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{parameterize\_schmitt}{Get the Parameters for Schmitt's Tissue Partition Coefficient Method}{parameterize.Rul.schmitt}
\keyword{Parameter}{parameterize\_schmitt}
\keyword{schmitt}{parameterize\_schmitt}
%
\begin{Description}\relax
This function provides the necessary parameters to run
predict\_partitioning\_schmitt, excluding the data in tissue.data.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
parameterize_schmitt(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  default.to.human = FALSE,
  force.human.fup = FALSE,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemcial and physiological description parameters needed
to run the Schmitt et al. (2008) model

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{default.to.human}] Substitutes missing fraction of unbound plasma with
human values if true.

\item[\code{force.human.fup}] Returns human fraction of unbound plasma in
calculation for rats if true.
When species is specified as rabbit, dog, or mouse, the human unbound
fraction is substituted.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{Funbound.plasma}] corrected unbound fraction in plasma
\item[\code{unadjusted.Funbound.plasma}] measured unbound fraction in plasma (0.005
if below limit of detection)\item[\code{Pow}] octonol:water partition coefficient
(not log transformed)\item[\code{pKa\_Donor}] compound H dissociation equilibirum
constant(s)\item[\code{pKa\_Accept}] compound H association equilibirum
constant(s)\item[\code{MA}] phospholipid:water distribution coefficient, membrane
affinity\item[\code{Fprotein.plasma}] protein fraction in plasma
\item[\code{plasma.pH}] pH of the plasma
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
Robert Pearce and John Wambaugh
\end{Author}
%
\begin{References}\relax
Schmitt, Walter. "General approach for the calculation of 
tissue to plasma partition coefficients." Toxicology in Vitro 22.2 (2008): 
457-467.

Schmitt, Walter. "Corrigendum to: General approach for the calculation of 
tissue to plasma partition coefficients" Toxicology in Vitro 22.6 (2008): 1666.

Peyret, Thomas, Patrick Poulin, and Kannan Krishnan. "A unified algorithm 
for predicting partition coefficients for PBPK modeling of drugs and 
environmental chemicals." Toxicology and applied pharmacology 249.3 (2010): 
197-207.

Pearce, Robert G., et al. "Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues." Journal of pharmacokinetics 
and pharmacodynamics 44.6 (2017): 549-565.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

parameterize_schmitt(chem.name='bisphenola')

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{parameterize\_steadystate}{Parameterize\_SteadyState}{parameterize.Rul.steadystate}
\keyword{3compss}{parameterize\_steadystate}
\keyword{Parameter}{parameterize\_steadystate}
%
\begin{Description}\relax
This function initializes the parameters needed in the functions
calc\_mc\_css, calc\_mc\_oral\_equiv, and calc\_analytic\_css for the three
compartment steady state model ('3compartmentss').
\end{Description}
%
\begin{Usage}
\begin{verbatim}
parameterize_steadystate(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  clint.pvalue.threshold = 0.05,
  default.to.human = FALSE,
  human.clint.fup = FALSE,
  adjusted.Funbound.plasma = TRUE,
  restrictive.clearance = TRUE,
  fup.lod.default = 0.005,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.cas}] Chemical Abstract Services Registry Number (CAS-RN) -- the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{chem.name}] Chemical name (spaces and capitalization ignored) --  the 
chemical must be identified by either CAS, name, or DTXISD

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
-- the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{clint.pvalue.threshold}] Hepatic clearances with clearance assays
having p-values greater than the threshold are set to zero.

\item[\code{default.to.human}] Substitutes missing rat values with human values if
true.

\item[\code{human.clint.fup}] Uses human hepatic intrinsic clearance and fraction
of unbound plasma in calculation of partition coefficients for rats if true.

\item[\code{adjusted.Funbound.plasma}] Returns adjusted Funbound.plasma when set to
TRUE.

\item[\code{restrictive.clearance}] In calculating hepatic.bioavailability, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.

\item[\code{fup.lod.default}] Default value used for fraction of unbound plasma for
chemicals where measured value was below the limit of detection. Default
value is 0.0005.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{Clint}] Hepatic Intrinsic Clearance, uL/min/10\textasciicircum{}6 cells.
\item[\code{Fgutabs}] Fraction of the oral dose absorbed, i.e. the fraction of the
dose that enters the gutlumen.
\item[\code{Funbound.plasma}] Fraction of plasma that is not bound.
\item[\code{Qtotal.liverc}] Flow rate of blood exiting the liver, L/h/kg BW\textasciicircum{}3/4.
\item[\code{Qgfrc}] Glomerular Filtration Rate, L/h/kg
BW\textasciicircum{}3/4, volume of fluid filtered from kidney and excreted.
\item[\code{BW}] Body Weight, kg
\item[\code{MW}] Molecular Weight, g/mol
\item[\code{million.cells.per.gliver}] Millions cells per gram of liver tissue.
\item[\code{Vliverc}] Volume of the liver per kg body weight, L/kg BW.
\item[\code{liver.density}] Liver tissue density, kg/L.
\item[\code{Fhep.assay.correction}] The fraction of chemical unbound in hepatocyte
assay using the method of Kilford et al. (2008)
\item[\code{hepatic.bioavailability}] Fraction of dose remaining after first pass
clearance, calculated from the corrected well-stirred model.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput 
toxicokinetics." Journal of statistical software 79.4 (2017): 1.

Kilford, P. J., Gertz, M., Houston, J. B. and Galetin, A.
(2008). Hepatocellular binding of drugs: correction for unbound fraction in
hepatocyte incubations using microsomal binding or drug lipophilicity data.
Drug Metabolism and Disposition 36(7), 1194-7, 10.1124/dmd.108.020834.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

 parameters <- parameterize_steadystate(chem.name='Bisphenol-A',species='Rat')
 parameters <- parameterize_steadystate(chem.cas='80-05-7')

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{pc.data}{Partition Coefficient Data}{pc.data}
\keyword{data}{pc.data}
%
\begin{Description}\relax
Measured rat in vivo partition coefficients and data for predicting them.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pc.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame.
\end{Format}
%
\begin{Author}\relax
Jimena Davis and Robert Pearce
\end{Author}
%
\begin{References}\relax
Schmitt, W., General approach for the calculation of tissue to
plasma partition coefficients. Toxicology in Vitro, 2008. 22(2): p. 457-467.

Schmitt, W., Corrigendum to:"General approach for the calculation of tissue
to plasma partition coefficients"[Toxicology in Vitro 22 (2008) 457-467].
Toxicology in Vitro, 2008. 22(6): p. 1666.

Poulin, P. and F.P. Theil, A priori prediction of tissue: plasma partition
coefficients of drugs to facilitate the use of physiologically based
pharmacokinetic models in drug discovery. Journal of pharmaceutical
sciences, 2000. 89(1): p. 16-35.

Rodgers, T. and M. Rowland, Physiologically based pharmacokinetic modelling
2: predicting the tissue distribution of acids, very weak bases, neutrals
and zwitterions. Journal of pharmaceutical sciences, 2006. 95(6): p.
1238-1257.

Rodgers, T., D. Leahy, and M. Rowland, Physiologically based pharmacokinetic
modeling 1: predicting the tissue distribution of moderate-to-strong bases.
Journal of pharmaceutical sciences, 2005. 94(6): p. 1259-1276.

Rodgers, T., D. Leahy, and M. Rowland, Tissue distribution of basic drugs:
Accounting for enantiomeric, compound and regional differences amongst
beta-blocking drugs in rat. Journal of pharmaceutical sciences, 2005. 94(6):
p. 1237-1248.

Gueorguieva, I., et al., Development of a whole body physiologically based
model to characterise the pharmacokinetics of benzodiazepines. 1: Estimation
of rat tissue-plasma partition ratios. Journal of pharmacokinetics and
pharmacodynamics, 2004. 31(4): p. 269-298.

Poulin, P., K. Schoenlein, and F.P. Theil, Prediction of adipose tissue:
plasma partition coefficients for structurally unrelated drugs. Journal of
pharmaceutical sciences, 2001. 90(4): p. 436-447.

Bjorkman, S., Prediction of the volume of distribution of a drug: which
tissue-plasma partition coefficients are needed? Journal of pharmacy and
pharmacology, 2002. 54(9): p. 1237-1245.

Yun, Y. and A. Edginton, Correlation-based prediction of tissue-to-plasma
partition coefficients using readily available input parameters.
Xenobiotica, 2013. 43(10): p. 839-852.

Uchimura, T., et al., Prediction of human blood-to-plasma drug concentration
ratio. Biopharmaceutics \& drug disposition, 2010. 31(5-6): p. 286-297.
\end{References}
\inputencoding{utf8}
\HeaderA{pearce2017regression}{Pearce et al. 2017 data}{pearce2017regression}
\aliasA{Pearce2017Regression}{pearce2017regression}{Pearce2017Regression}
\keyword{data}{pearce2017regression}
%
\begin{Description}\relax
This table includes the adjusted and unadjusted regression parameter
estimates for the chemical-specifc plasma
protein unbound fraction (fup) in 12 different tissue types.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pearce2017regression
\end{verbatim}
\end{Usage}
%
\begin{Format}
data.frame
\end{Format}
%
\begin{Details}\relax
Predictions were made with regression models, 
as reported in Pearce et al. (2017).
\end{Details}
%
\begin{Author}\relax
Robert G. Pearce
\end{Author}
%
\begin{Source}\relax
Pearce et al. 2017 Regression Models
\end{Source}
%
\begin{References}\relax
Pearce, Robert G., et al. "Evaluation and calibration of 
high-throughput predictions of chemical distribution to tissues."
Journal of pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.
\end{References}
\inputencoding{utf8}
\HeaderA{pharma}{DRUGS|NORMAN: Pharmaceutical List with EU, Swiss, US Consumption Data}{pharma}
\keyword{data}{pharma}
%
\begin{Description}\relax
SWISSPHARMA is a list of pharmaceuticals with consumption data from
Switzerland, France, Germany and the USA, used for a suspect
screening/exposure modelling approach described in
Singer et al 2016, DOI: 10.1021/acs.est.5b03332. The original data is
available on the NORMAN Suspect List Exchange.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pharma
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class \code{data.frame} with 954 rows and 14 columns.
\end{Format}
%
\begin{Source}\relax
\url{https://comptox.epa.gov/dashboard/chemical_lists/swisspharma}
\end{Source}
%
\begin{References}\relax
Wambaugh et al. (2019) "Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization", Toxicological Sciences, 172(2), 235-251.
\end{References}
\inputencoding{utf8}
\HeaderA{physiology.data}{Species-specific physiology parameters}{physiology.data}
\keyword{data}{physiology.data}
%
\begin{Description}\relax
This data set contains values from Davies and Morris (1993) necessary to
paramaterize a toxicokinetic model for human, mouse, rat, dog, or rabbit.
The temperature for each species are taken from Robertshaw et al. (2004),
Gordon (1993), and Stammers(1926).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
physiology.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 11 rows and 7 columns.
\end{Format}
%
\begin{Author}\relax
John Wambaugh and Nisha Sipes
\end{Author}
%
\begin{Source}\relax
Wambaugh, John F., et al. "Toxicokinetic triage for environmental
chemicals." Toxicological Sciences (2015): 228-237.
\end{Source}
%
\begin{References}\relax
Davies, B. and Morris, T. (1993). Physiological Parameters in
Laboratory Animals and Humans. Pharmaceutical Research 10(7), 1093-1095,
10.1023/a:1018943613122.  

Environment, in Dukes' Physiology of Domestic Animals, 12th ed., Reece W.O.,
Ed. Copyright 2004 by Cornell University.  Stammers (1926) The blood count
and body temperature in normal rats Gordon (1993) Temperature Regulation in
Laboratory Rodents
\end{References}
\inputencoding{utf8}
\HeaderA{predict\_partitioning\_schmitt}{Predict partition coefficients using the method from Schmitt (2008).}{predict.Rul.partitioning.Rul.schmitt}
\keyword{Parameter}{predict\_partitioning\_schmitt}
%
\begin{Description}\relax
This function implements the method from Schmitt (2008) in predicting the 
tissue to unbound plasma partition coefficients for the tissues contained 
in the tissue.data table.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
predict_partitioning_schmitt(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  species = "Human",
  model = "pbtk",
  default.to.human = FALSE,
  parameters = NULL,
  alpha = 0.001,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  regression.list = c("brain", "adipose", "gut", "heart", "kidney", "liver", "lung",
    "muscle", "skin", "spleen", "bone"),
  tissues = NULL,
  minimum.Funbound.plasma = 1e-04,
  suppress.messages = FALSE
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name or the CAS number must be
specified.

\item[\code{chem.cas}] Either the chemical name or the CAS number must be
specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{model}] Model for which partition coefficients are neeeded (for example,
"pbtk", "3compartment")

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{parameters}] Chemical parameters from \code{\LinkA{parameterize\_schmitt}{parameterize.Rul.schmitt}}
overrides chem.name, dtxsid, and chem.cas.

\item[\code{alpha}] Ratio of Distribution coefficient D of totally charged species
and that of the neutral form

\item[\code{adjusted.Funbound.plasma}] Whether or not to use Funbound.plasma
adjustment.

\item[\code{regression}] Whether or not to use the regressions.  Regressions are
used by default.

\item[\code{regression.list}] Tissues to use regressions on.

\item[\code{tissues}] Vector of desired partition coefficients.  Returns all by
default.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{suppress.messages}] Whether or not the output message is suppressed.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
A separate regression is used when adjusted.Funbound.plasma is FALSE.

A regression is used for membrane affinity when not provided.  The
regressions for correcting each tissue are performed on tissue plasma
partition coefficients (Ktissue2pu * Funbound.plasma) calculated with the
corrected Funbound.plasma value and divided by this value to get Ktissue2pu.
Thus the regressions should be used with the corrected Funbound.plasma.

The red blood cell regression can be used but is not by default because of
the span of the data used, reducing confidence in the regression for higher
and lower predicted values.

Human tissue volumes are used for species other than Rat.
\end{Details}
%
\begin{Value}
Returns tissue to unbound plasma partition coefficients for each
tissue.
\end{Value}
%
\begin{Author}\relax
Robert Pearce
\end{Author}
%
\begin{References}\relax
Schmitt, Walter. "General approach for the calculation of tissue to plasma 
partition coefficients." Toxicology in Vitro 22.2 (2008): 457-467.

Birnbaum, L., et al. "Physiological parameter values for PBPK models." 
International Life Sciences Institute, Risk Science Institute, Washington, 
DC (1994).

Pearce, Robert G., et al. "Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues." Journal of pharmacokinetics 
and pharmacodynamics 44.6 (2017): 549-565.

Yun, Y. E., and A. N. Edginton. "Correlation-based prediction of 
tissue-to-plasma partition coefficients using readily available input 
parameters." Xenobiotica 43.10 (2013): 839-852.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

predict_partitioning_schmitt(chem.name='ibuprofen',regression=FALSE)

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{propagate\_invitrouv\_1comp}{Propagates uncertainty and variability in in vitro HTTK data into one compartment model parameters}{propagate.Rul.invitrouv.Rul.1comp}
\keyword{1compartment}{propagate\_invitrouv\_1comp}
\keyword{monte-carlo}{propagate\_invitrouv\_1comp}
%
\begin{Description}\relax
Propagates uncertainty and variability in in vitro HTTK data into one
compartment model parameters
\end{Description}
%
\begin{Usage}
\begin{verbatim}
propagate_invitrouv_1comp(parameters.dt, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{parameters.dt}] The data table of parameters being used by the Monte
Carlo sampler

\item[\code{...}] Additional arguments passed to \code{\LinkA{calc\_elimination\_rate}{calc.Rul.elimination.Rul.rate}}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table whose columns are the parameters of the HTTK model
specified in \code{model}.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{propagate\_invitrouv\_3comp}{Propagates uncertainty and variability in in vitro HTTK data into three compartment model parameters}{propagate.Rul.invitrouv.Rul.3comp}
\keyword{3compartment}{propagate\_invitrouv\_3comp}
\keyword{monte-carlo}{propagate\_invitrouv\_3comp}
%
\begin{Description}\relax
Propagates uncertainty and variability in in vitro HTTK data into three
compartment model parameters
\end{Description}
%
\begin{Usage}
\begin{verbatim}
propagate_invitrouv_3comp(parameters.dt, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{parameters.dt}] The data table of parameters being used by the Monte
Carlo sampler

\item[\code{...}] Additional arguments passed to \code{\LinkA{calc\_hep\_clearance}{calc.Rul.hep.Rul.clearance}}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table whose columns are the parameters of the HTTK model
specified in \code{model}.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{propagate\_invitrouv\_pbtk}{Propagates uncertainty and variability in in vitro HTTK data into PBPK model parameters}{propagate.Rul.invitrouv.Rul.pbtk}
\keyword{monte-carlo}{propagate\_invitrouv\_pbtk}
\keyword{pbtk}{propagate\_invitrouv\_pbtk}
%
\begin{Description}\relax
Propagates uncertainty and variability in in vitro HTTK data into PBPK
model parameters
\end{Description}
%
\begin{Usage}
\begin{verbatim}
propagate_invitrouv_pbtk(parameters.dt, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{parameters.dt}] The data table of parameters being used by the Monte
Carlo sampler

\item[\code{...}] Additional arguments passed to \code{\LinkA{calc\_hep\_clearance}{calc.Rul.hep.Rul.clearance}}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A data.table whose columns are the parameters of the HTTK model
specified in \code{model}.
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{reset\_httk}{Reset HTTK to Default Data Tables}{reset.Rul.httk}
%
\begin{Description}\relax
This function returns an updated version of chem.physical\_and\_invitro.data
that includes data predicted with Simulations Plus' ADMET predictor that was
used in Sipes et al. 2017, included in admet.data.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
reset_httk(target.env = .GlobalEnv)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{target.env}] The environment where the new
chem.physical\_and\_invitro.data is loaded. Defaults to global environment.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
\begin{ldescription}
\item[\code{data.frame}] The package default version of 
chem.physical\_and\_invitro.data.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}


chem.physical_and_invitro.data <- load_sipes2017()
reset_httk()
                        

\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{rfun}{Randomly draws from a one-dimensional KDE}{rfun}
\keyword{httk-pop}{rfun}
%
\begin{Description}\relax
Randomly draws from a one-dimensional KDE
\end{Description}
%
\begin{Usage}
\begin{verbatim}
rfun(n, fhat)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{n}] Number of samples to draw

\item[\code{fhat}] A list with elements x, w, and h (h is the KDE bandwidth).
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of n samples from the KDE fhat
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{r\_left\_censored\_norm}{Returns draws from a normal distribution with a lower censoring limit of lod (limit of detection)}{r.Rul.left.Rul.censored.Rul.norm}
%
\begin{Description}\relax
Returns draws from a normal distribution with a lower censoring limit of lod
(limit of detection)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
r_left_censored_norm(n, mean = 0, sd = 1, lod = 0.005, lower = 0, upper = 1)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{n}] Number of samples to take

\item[\code{mean}] Mean of censored distribution. Default 0.

\item[\code{sd}] Standard deviation of censored distribution. Default 1.

\item[\code{lod}] Bound below which to censor. Default 0.005.

\item[\code{lower}] Lower bound on censored distribution. Default 0.

\item[\code{upper}] Upper bound on censored distribution. Default 1.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of samples from the specified censored distribution.
\end{Value}
\inputencoding{utf8}
\HeaderA{scale\_dosing}{Scale mg/kg body weight doses according to body weight and units}{scale.Rul.dosing}
\keyword{Dynamic}{scale\_dosing}
%
\begin{Description}\relax
This function transforms the dose (in mg/kg) into the appropriate units. It
handles single doses, matrices of doses, or daily repeated doses at varying
intervals. Gut absorption is also factored in through the parameter Fgutabs,
and scaling is currently avoided in the inhalation exposure case with a 
scale factor of 1
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scale_dosing(
  dosing,
  parameters,
  route,
  input.units = NULL,
  output.units = "uM",
  vol = NULL
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{dosing}] List of dosing metrics used in simulation, which must include
the general entries with names "initial.dose", "doses.per.day", 
"daily.dose", and "dosing.matrix". The "dosing.matrix" is used for more
precise dose regimen specification, and is a matrix consisting of two
columns or rows named "time" and "dose" containing the time and amount,
in mg/kg BW, of each dose. The minimal usage case involves all entries but
"initial.dose" set to NULL in value.

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk function,
overrides chem.name and chem.cas.

\item[\code{route}] String specification of route of exposure for simulation:
"oral", "iv", "inhalation", ...

\item[\code{output.units}] Desired units (either "mg/L", "mg", "umol", or default
"uM").

\item[\code{vol}] Volume for the target tissue of interest.
NOTE: Volume should not be in units of per BW, i.e. "kg".
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A list of numeric values for doses converted to output.units, potentially
(depending on argument dosing) including:
\begin{ldescription}
\item[\code{initial.dose}] The first dose given
\item[\code{dosing.matrix}] A 2xN matrix where the first column is dose time and
the second is dose amount for N doses
\item[\code{daily.dose}] The total cumulative daily dose
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Sarah E. Davidson
\end{Author}
\inputencoding{utf8}
\HeaderA{set\_httk\_precision}{set\_httk\_precision}{set.Rul.httk.Rul.precision}
%
\begin{Description}\relax
Although the ODE solver and other functions return very precise numbers,
we cannot (or at least do not spend enough computing time to) be sure of the 
precioion to an arbitrary level. This function both limits the number of
signficant figures reported and truncates the numerical precision.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
set_httk_precision(in.num, sig.fig = 4, num.prec = 9)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{in.num}] The numeric variable (or assembly of numerics) to be
processed.

\item[\code{sig.fig}] The number of significant figures reported. Defaults to 4.

\item[\code{num.prec}] The precision maintained, digits below 10\textasciicircum{}num.prec are 
dropped. Defaults to 9.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
numeric values
\end{Value}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{sipes2017}{Sipes et al. 2017 data}{sipes2017}
\aliasA{Sipes2017}{sipes2017}{Sipes2017}
\keyword{data}{sipes2017}
%
\begin{Description}\relax
This table includes in silico predicted chemical-specifc plasma protein 
unbound fraction (fup) and intrinsic hepatic clearance values for the entire
Tox21 library 
(see \url{https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21}). 
Predictions were made with Simulations Plus ADMET predictor,
as reported in Sipes et al. (2017).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sipes2017
\end{verbatim}
\end{Usage}
%
\begin{Format}
data.frame
\end{Format}
%
\begin{Author}\relax
Nisha Sipes
\end{Author}
%
\begin{Source}\relax
ADMET, Simulations Plus
\end{Source}
%
\begin{References}\relax
Sipes, Nisha S., et al. "An Intuitive Approach for Predicting
Potential Human Health Risk with the Tox21 10k Library." Environmental
Science \& Technology 51.18 (2017): 10786-10796.
\end{References}
\inputencoding{utf8}
\HeaderA{skeletal\_muscle\_mass}{Predict skeletal muscle mass}{skeletal.Rul.muscle.Rul.mass}
\keyword{httk-pop}{skeletal\_muscle\_mass}
%
\begin{Description}\relax
Predict skeletal muscle mass from age, height, and gender.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
skeletal_muscle_mass(smm, age_years, height, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{smm}] Vector of allometrically-scaled skeletal muscle masses.

\item[\code{age\_years}] Vector of ages in years.

\item[\code{height}] Vector of heights in cm.

\item[\code{gender}] Vector of genders, either 'Male' or 'Female.'
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
For individuals over age 18, use allometrically-scaled muscle mass with an
age-based scaling factor, to account for loss of muscle mass with age
(Janssen et al. 2000). For individuals under age 18, use
\code{\LinkA{skeletal\_muscle\_mass\_children}{skeletal.Rul.muscle.Rul.mass.Rul.children}}.
\end{Details}
%
\begin{Value}
Vector of skeletal muscle masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Janssen, Ian, et al. "Skeletal muscle mass and distribution in 468 men and
women aged 18-88 yer." Journal of Applied Physiology 89.1 (2000): 81-88

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{skeletal\_muscle\_mass\_children}{skeletal.Rul.muscle.Rul.mass.Rul.children}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{skeletal\_muscle\_mass\_children}{Predict skeletal muscle mass for children}{skeletal.Rul.muscle.Rul.mass.Rul.children}
\keyword{httk-pop}{skeletal\_muscle\_mass\_children}
%
\begin{Description}\relax
For individuals under age 18, predict skeletal muscle mass from gender and
age, using a nonlinear equation from Webber and Barr (2012)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
skeletal_muscle_mass_children(gender, age_years)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{gender}] Vector of genders (either 'Male' or 'Female').

\item[\code{age\_years}] Vector of ages in years.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Vector of skeletal muscle masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Webber, Colin E., and Ronald D. Barr. "Age-and gender-dependent values of 
skeletal muscle mass in healthy children and adolescents." Journal of 
cachexia, sarcopenia and muscle 3.1 (2012): 25-29.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{skin\_mass\_bosgra}{Predict skin mass}{skin.Rul.mass.Rul.bosgra}
\keyword{httk-pop}{skin\_mass\_bosgra}
%
\begin{Description}\relax
Using equation from Bosgra et al. 2012, predict skin mass from body surface
area.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
skin_mass_bosgra(BSA)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{BSA}] Vector of body surface areas in cm\textasciicircum{}2.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Vector of skin masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Bosgra, Sieto, et al. "An improved model to predict 
physiologically based model parameters and their inter-individual variability 
from anthropometry." Critical reviews in toxicology 42.9 (2012): 751-767.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\graphicspath{{"C:/Users/jwambaug/git/httk/httk/man/figures/"}}
\inputencoding{utf8}
\HeaderA{solve\_1comp}{Solve one compartment TK model}{solve.Rul.1comp}
\keyword{1compartment}{solve\_1comp}
\keyword{Solve}{solve\_1comp}
%
\begin{Description}\relax
This function solves for the amount or concentration of a chemical in plasma
for a one compartment model as a function of time based on the dose and
dosing frequency.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
solve_1comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  output.units = "uM",
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  dosing.matrix = NULL,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = T,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{times}] Optional time sequence for specified number of days.

\item[\code{parameters}] Chemical parameters from parameterize\_1comp function,
overrides chem.name and chem.cas.

\item[\code{days}] Length of the simulation.

\item[\code{tsteps}] The number time steps per hour.

\item[\code{daily.dose}] Total daily dose, mg/kg BW.

\item[\code{dose}] Amount of a single dose, mg/kg BW.

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{initial.values}] Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
output.units.  Defaults are zero.

\item[\code{plots}] Plots all outputs if true.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", or default
"Human").

\item[\code{iv.dose}] Simulates a single i.v. dose if true.

\item[\code{output.units}] Desired units (either "mg/L", "mg", "umol", or default
"uM").

\item[\code{method}] Method used by integrator (deSolve).

\item[\code{rtol}] Argument passed to integrator (deSolve).

\item[\code{atol}] Argument passed to integrator (deSolve).

\item[\code{default.to.human}] Substitutes missing rat values with human values if
true.

\item[\code{recalc.blood2plasma}] Whether or not to recalculate the blood:plasma
chemical concentrationr ratio

\item[\code{recalc.clearance}] Whether or not to recalculate the elimination
rate.

\item[\code{dosing.matrix}] Vector of dosing times or a matrix consisting of two
columns or rows named "dose" and "time" containing the time and amount, in
mg/kg BW, of each dose.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with volume of distribution calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients in volume of distribution calculation.

\item[\code{restrictive.clearance}] In calculating elimination rate, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{monitor.vars}] Which variables are returned as a function of time. 
Defaults value of NULL provides "Agutlumen", "Ccompartment", "Ametabolized",
"AUC"

\item[\code{...}] Additional arguments passed to the integrator.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Note that the model parameters have units of hours while the model output is
in days.

Default value of NULL for doses.per.day solves for a single dose.

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.

AUC is area under plasma concentration curve.

Model Figure 

\Figure{1comp.pdf}{width=12cm alt="Figure: OneCompartment Model Schematic"}
\end{Details}
%
\begin{Value}
A matrix with a column for time(in days) and a column for the
compartment and the area under the curve (concentration only).
\end{Value}
%
\begin{Author}\relax
Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput
toxicokinetics." Journal of statistical software 79.4 (2017): 1.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

solve_1comp(chem.name='Bisphenol-A',days=1)
params <- parameterize_1comp(chem.cas="80-05-7")
solve_1comp(parameters=params)

\end{ExampleCode}
\end{Examples}
\graphicspath{{"C:/Users/jwambaug/git/httk/httk/man/figures/"}}
\inputencoding{utf8}
\HeaderA{solve\_3comp}{Solve\_3comp}{solve.Rul.3comp}
\keyword{3compartment}{solve\_3comp}
\keyword{Solve}{solve\_3comp}
%
\begin{Description}\relax
This function solves for the amounts or concentrations of a chemical in
different tissues as functions of time based on the dose and dosing
frequency.  It uses a three compartment model with partition coefficients.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
solve_3comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  output.units = "uM",
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  dosing.matrix = NULL,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = T,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's 'DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{times}] Optional time sequence for specified number of days.  The
dosing sequence begins at the beginning of times.

\item[\code{parameters}] Chemical parameters from parameterize\_3comp function,
overrides chem.name and chem.cas.

\item[\code{days}] Length of the simulation.

\item[\code{tsteps}] The number time steps per hour.

\item[\code{daily.dose}] Total daily dose, mg/kg BW.

\item[\code{dose}] Amount of a single dose, mg/kg BW.

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{initial.values}] Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
output.units.  Defaults are zero.

\item[\code{plots}] Plots all outputs if true.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{iv.dose}] Simulates a single i.v. dose if true.

\item[\code{output.units}] Desired units (either "mg/L", "mg", "umol", or default
"uM").

\item[\code{method}] Method used by integrator (deSolve).

\item[\code{rtol}] Argument passed to integrator (deSolve).

\item[\code{atol}] Argument passed to integrator (deSolve).

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.

\item[\code{recalc.clearance}] Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.

\item[\code{dosing.matrix}] Vector of dosing times or a matrix consisting of two
columns or rows named "dose" and "time" containing the time and amount, in
mg/kg BW, of each dose.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{monitor.vars}] Which variables are returned as a function of time. 
Defaults value of NULL provides "Cliver", "Csyscomp", "Atubules", 
"Ametabolized", "AUC"

\item[\code{...}] Additional arguments passed to the integrator.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Note that the model parameters have units of hours while the model output is
in days.

Default of NULL for doses.per.day solves for a single dose.

The compartments used in this model are the gutlumen, gut, liver, and
rest-of-body, with the plasma equivalent to the liver plasma.

Model Figure 
 
\Figure{3comp.pdf}{width=12cm alt="Figure: Three CompartmentModel Schematic"}

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
A matrix of class deSolve with a column for time(in days) and each
compartment, the plasma concentration, area under the curve, and a row for
each time point.
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput
toxicokinetics." Journal of statistical software 79.4 (2017): 1.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

solve_3comp(chem.name='Bisphenol-A',doses.per.day=2,daily.dose=.5,days=1,tsteps=2)

params <-parameterize_3comp(chem.cas="80-05-7")
solve_3comp(parameters=params)

\end{ExampleCode}
\end{Examples}
\graphicspath{{"C:/Users/jwambaug/git/httk/httk/man/figures/"}}
\inputencoding{utf8}
\HeaderA{solve\_gas\_pbtk}{solve\_gas\_pbtk}{solve.Rul.gas.Rul.pbtk}
\keyword{Solve}{solve\_gas\_pbtk}
%
\begin{Description}\relax
This function solves for the amounts or concentrations of a chemical
in different tissues as functions of time as a result of inhalation 
exposure to an ideal gas.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
solve_gas_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  times = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  doses.per.day = NULL,
  dose = NULL,
  dosing.matrix = NULL,
  forcings = NULL,
  exp.start.time = 0,
  exp.conc = 1,
  period = 24,
  exp.duration = 12,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  input.units = "ppmv",
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = T,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  vmax = 0,
  km = 1,
  exercise = F,
  fR = 12,
  VT = 0.75,
  VD = 0.15,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{parameters}] Chemical parameters from parameterize\_gas\_pbtk (or other
bespoke) function, overrides chem.name and chem.cas.

\item[\code{times}] Optional time sequence for specified number of days.  Dosing
sequence begins at the beginning of times.

\item[\code{days}] Length of the simulation.

\item[\code{tsteps}] The number of time steps per hour.

\item[\code{daily.dose}] Total daily dose

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{dose}] Amount of a single dose

\item[\code{dosing.matrix}] Vector of dosing times or a matrix consisting of two
columns or rows named "dose" and "time" containing the time and amount of 
each dose.

\item[\code{forcings}] Manual input of 'forcings' data series argument for ode
integrator. If left unspecified, 'forcings' defaults to NULL, and then other 
input parameters (see exp.start.time, exp.conc, exp.duration, and period)
provide the necessary information to assemble a forcings data series.

\item[\code{exp.start.time}] Start time in specifying forcing exposure series,
default 0.

\item[\code{exp.conc}] Specified inhalation exposure concentration for use in 
assembling "forcings" data series argument for integrator. Defaults to
units of uM

\item[\code{period}] For use in assembling forcing function data series 'forcings'
argument, specified in hours

\item[\code{exp.duration}] For use in assembling forcing function data 
series 'forcings' argument, specified in hours

\item[\code{initial.values}] Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
those specified for the model outputs. Default values are zero.

\item[\code{plots}] Plots all outputs if true.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{input.units}] Input units of interest assigned to dosing, including 
forcings. Defaults to "ppmv" as applied to the default forcings scheme.

\item[\code{method}] Method used by integrator (deSolve).

\item[\code{rtol}] Argument passed to integrator (deSolve).

\item[\code{atol}] Argument passed to integrator (deSolve).

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.

\item[\code{recalc.clearance}] Recalculates the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{monitor.vars}] Which variables are returned as a function of time. 
Defaults value of NULL provides "Cgut", "Cliver", "Cven", "Clung", "Cart",
"Crest", "Ckidney", "Cplasma", "Calv", "Cendexh", "Cmixexh", "Cmuc", 
"Atubules", "Ametabolized", "AUC"

\item[\code{vmax}] Michaelis-Menten vmax value in reactions/min

\item[\code{km}] Michaelis-Menten concentration of half-maximal reaction velocity
in desired output concentration units.

\item[\code{exercise}] Logical indicator of whether to simulate an exercise-induced
heightened respiration rate

\item[\code{fR}] Respiratory frequency (breaths/minute), used especially to adjust
breathing rate in the case of exercise. This parameter, along with VT and VD
(below) gives another option for calculating Qalv (Alveolar ventilation) 
in case pulmonary ventilation rate is not known

\item[\code{VT}] Tidal volume (L), to be modulated especially as part of simulating
the state of exercise

\item[\code{VD}] Anatomical dead space (L), to be modulated especially as part of
simulating the state of exercise

\item[\code{...}] Additional arguments passed to the integrator.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
The default dosing scheme involves a specification of the start time
of exposure (exp.start.time), the concentration of gas inhaled (exp.conc),
the period of a cycle of exposure and non-exposure (period), the
duration of the exposure during that period (exp.duration), and the total
days simulated. Together,these arguments determine the "forcings" passed to
the ODE integrator. Forcings can also be specified manually, or effectively
turned off by setting exposure concentration to zero, if the user prefers to 
simulate dosing by other means. 

The "forcings" object is configured to be passed to the integrator with,
at the most, a basic unit conversion among ppmv, mg/L, and uM. No scaling by
BW is set to be performed on the forcings series.

Note that the model parameters have units of hours while the model output is
in days.

Default NULL value for doses.per.day solves for a single dose.

The compartments used in this model are the gut lumen, gut, liver, kidneys,
veins, arteries, lungs, and the rest of the body.

The extra compartments include the amounts or concentrations metabolized by
the liver and excreted by the kidneys through the tubules.

AUC is the area under the curve of the plasma concentration.

Model Figure from \bsl{}insertCitelinakis2020developmenthttk:

\Figure{gaspbtk.pdf}{width=12cm alt="Figure: Gas PBTK Model Schematic"}

Model parameters are named according to the following convention:
\Tabular{lrrrr}{
prefix & suffic & Meaning & units \\{}
K & & Partition coefficient for tissue to free plasma \bsl{} tab unitless \\{}
V & & Volume & L \\{}
Q & & Flow & L/h \\{}
k & & Rate & 1/h \\{}
& c & Parameter is proportional to body weight & 1 / kg for volumes
and 1/kg\textasciicircum{}(3/4) for flows \\{}}

When species is specified but chemical-specific in vitro data are not
available, the function uses the appropriate physiological data (volumes and 
flows) but default.to.human = TRUE must be used to substitute human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
\end{Value}
%
\begin{Author}\relax
Matt Linakis, John Wambaugh, Mark Sfeir, Miyuki Breen
\end{Author}
%
\begin{References}\relax
\bsl{}insertReflinakis2020developmenthttk

\bsl{}insertRefpearce2017httkhttk
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

solve_gas_pbtk(chem.name = 'pyrene', exp.conc = 1, period = 24, expduration = 24)


out <- solve_gas_pbtk(chem.name='pyrene',exp.conc = 0, doses.per.day = 2,
daily.dose = 3, plots=TRUE,initial.values=c(Aven=20))

out <- solve_gas_pbtk(chem.name = 'pyrene',exp.conc = 3, period = 24,
exp.duration = 6, exercise = TRUE)
                  
params <- parameterize_gas_pbtk(chem.cas="80-05-7")
solve_gas_pbtk(parameters=params)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{solve\_model}{Solve\_model}{solve.Rul.model}
\keyword{Solve}{solve\_model}
%
\begin{Description}\relax
solve\_model is designed to accept systematized metadata (provided by the
model.list defined in the modelinfo files) for a given 
toxicokinetic model, including names of variables, parameterization
functions, and key units, and use it along with chemical information
to prepare an ode system for numerical solution over time of the amounts
or concentrations of chemical in different bodily compartments of a given
species (either "Rat", "Rabbit", "Dog", "Mouse", or default "Human").
\end{Description}
%
\begin{Usage}
\begin{verbatim}
solve_model(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  model = NULL,
  route = "oral",
  dosing = NULL,
  days = 10,
  tsteps = 4,
  initial.values = NULL,
  initial.value.units = NULL,
  plots = FALSE,
  monitor.vars = NULL,
  suppress.messages = FALSE,
  species = "Human",
  input.units = "mg/kg",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  adjusted.Funbound.plasma = TRUE,
  minimum.Funbound.plasma = 1e-04,
  parameterize.arg.list = list(default.to.human = FALSE, clint.pvalue.threshold = 0.05,
    restrictive.clearance = T, regression = TRUE),
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{http://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{times}] Optional time sequence for specified number of days. Dosing
sequence begins at the beginning of times.

\item[\code{parameters}] List of chemical parameters, as output by 
parameterize\_pbtk function. Overrides chem.name and chem.cas.

\item[\code{model}] Specified model to use in simulation: "pbtk", "3compartment",
"3compartmentss", "1compartment", "schmitt", ...

\item[\code{route}] String specification of route of exposure for simulation:
"oral", "iv", "inhalation", ...

\item[\code{dosing}] List of dosing metrics used in simulation, which includes
the namesake entries of a model's associated dosing.params. In the case
of most httk models, these should include "initial.dose", "doses.per.day", 
"daily.dose", and "dosing.matrix". The "dosing.matrix" is used for more
precise dose regimen specification, and is a matrix consisting of two
columns or rows named "time" and "dose" containing the time and amount of 
each dose. If none of the namesake entries of the dosing list is set to a
non-NULL value, solve\_model uses a default dose of 1 mg/kg BW along with the 
dose type (add/multiply) specified for a given route (e.g. add the dose to gut
lumen for oral route)

\item[\code{days}] Simulated period. Default 10 days.

\item[\code{tsteps}] The number of time steps per hour. Default of 4.

\item[\code{initial.values}] Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
those specified for the model outputs. Default values are zero.

\item[\code{plots}] Plots all outputs if true.

\item[\code{monitor.vars}] Which variables are returned as a function of time. 
Default values of NULL looks up variables specified in modelinfo\_MODEL.R

\item[\code{suppress.messages}] Whether or not the output messages are suppressed.

\item[\code{species}] Species desired (models have been designed to be
parameterized for some subset of the following species: "Rat", "Rabbit", 
"Dog", "Mouse", or default "Human").

\item[\code{input.units}] Input units of interest assigned to dosing. Defaults
to mg/kg BW, in line with the default dosing scheme of a one-time dose of
1 mg/kg in which no other dosing parameters are specified.

\item[\code{output.units}] Output units of interest for the compiled components.
Defaults to NULL, and will provide values in model units if unspecified.

\item[\code{method}] Method used by integrator (deSolve).

\item[\code{rtol}] Argument passed to integrator (deSolve).

\item[\code{atol}] Argument passed to integrator (deSolve).

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.

\item[\code{recalc.clearance}] Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset)

\item[\code{parameterize.arg.list}] Additional parameterized passed to the model
parameterization function.

\item[\code{...}] Additional arguments passed to the integrator.

\item[\code{default.to.human}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Dosing values with certain acceptable associated input.units (like mg/kg BW)
are configured to undergo a unit conversion. All model simulations are 
intended to run with units as specifed by "compartment.units" in the 
model.list (as defined by the modelinfo files).

The 'dosing' argument includes all parameters needed to describe exposure
in terms of route of administration, frequency, and quantity short of 
scenarios that require use of a more precise forcing function. If the dosing
argument's namesake entries are left NULL, solve\_model defaults to a
single-time dose of 1 mg/kg BW according to the given dosing route and 
associated type (either add/multiply, for example we typically add a dose to 
gut lumen 
when oral route is specified).

AUC is the area under the curve of the plasma concentration.

Model parameters are named according to the following convention:

\Tabular{lrrrr}{
prefix & suffix & Meaning & units \\{}
K & & Partition coefficient for tissue to free plasma \bsl{} tab unitless \\{}
V & & Volume & L \\{}
Q & & Flow & L/h \\{}
k & & Rate & 1/h \\{}
& c & Parameter is proportional to body weight & 1 / kg for volumes
and 1/kg\textasciicircum{}(3/4) for flows \\{}}

When species is specified but chemical-specific in vitro data are not
available, the function uses the appropriate physiological data (volumes and
flows) but default.to.human = TRUE must be used to substitute human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
\end{Value}
%
\begin{Author}\relax
John Wambaugh, Robert Pearce, Miyuki Breen, Mark Sfeir, and
Sarah E. Davidson
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput
toxicokinetics." Journal of statistical software 79.4 (2017): 1.
\end{References}
\graphicspath{{"C:/Users/jwambaug/git/httk/httk/man/figures/"}}
\inputencoding{utf8}
\HeaderA{solve\_pbtk}{Solve\_PBTK}{solve.Rul.pbtk}
\keyword{Solve}{solve\_pbtk}
\keyword{pbtk}{solve\_pbtk}
%
\begin{Description}\relax
This function solves for the amounts or concentrations in uM of a chemical
in different tissues as functions of time based on the dose and dosing
frequency.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
solve_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  input.units = "mg/kg",
  output.units = "uM",
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  dosing.matrix = NULL,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = T,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  ...
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chem.name}] Either the chemical name, CAS number, or the parameters
must be specified.

\item[\code{chem.cas}] Either the chemical name, CAS number, or the parameters must
be specified.

\item[\code{dtxsid}] EPA's DSSTox Structure ID (\url{https://comptox.epa.gov/dashboard})
the chemical must be identified by either CAS, name, or DTXSIDs

\item[\code{times}] Optional time sequence for specified number of days.  Dosing
sequence begins at the beginning of times.

\item[\code{parameters}] Chemical parameters from parameterize\_pbtk function,
overrides chem.name and chem.cas.

\item[\code{days}] Length of the simulation.

\item[\code{tsteps}] The number of time steps per hour.

\item[\code{daily.dose}] Total daily dose, defaults to mg/kg BW.

\item[\code{dose}] Amount of a single dose, defaults to mg/kg BW.

\item[\code{doses.per.day}] Number of doses per day.

\item[\code{initial.values}] Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
output.units.  Defaults are zero.

\item[\code{plots}] Plots all outputs if true.

\item[\code{suppress.messages}] Whether or not the output message is suppressed.

\item[\code{species}] Species desired (either "Rat", "Rabbit", "Dog", "Mouse", or
default "Human").

\item[\code{iv.dose}] Simulates a single i.v. dose if true.

\item[\code{input.units}] Input units of interest assigned to dosing, defaults to
mg/kg BW

\item[\code{method}] Method used by integrator (deSolve).

\item[\code{rtol}] Argument passed to integrator (deSolve).

\item[\code{atol}] Argument passed to integrator (deSolve).

\item[\code{recalc.blood2plasma}] Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.

\item[\code{recalc.clearance}] Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.

\item[\code{dosing.matrix}] Vector of dosing times or a matrix consisting of two
columns or rows named "dose" and "time" containing the time and amount, in
mg/kg BW, of each dose.

\item[\code{adjusted.Funbound.plasma}] Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.

\item[\code{regression}] Whether or not to use the regressions in calculating
partition coefficients.

\item[\code{restrictive.clearance}] Protein binding not taken into account (set to
1) in liver clearance if FALSE.

\item[\code{minimum.Funbound.plasma}] Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 -- half the lowest measured Fup in our
dataset).

\item[\code{monitor.vars}] Which variables are returned as a function of time. 
The default value of NULL provides "Cgut", "Cliver", "Cven", "Clung", "Cart", 
"Crest", "Ckidney", "Cplasma", "Atubules", "Ametabolized", and "AUC"

\item[\code{...}] Additional arguments passed to the integrator.

\item[\code{3man}] Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Note that the model parameters have units of hours while the model output is
in days.

Default NULL value for doses.per.day solves for a single dose.

The compartments used in this model are the gutlumen, gut, liver, kidneys,
veins, arteries, lungs, and the rest of the body.

The extra compartments include the amounts or concentrations metabolized by
the liver and excreted by the kidneys through the tubules.

AUC is the area under the curve of the plasma concentration.

Model Figure 

\Figure{pbtk.pdf}{width=12cm alt="Figure: PBTK ModelSchematic"}

When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
\end{Details}
%
\begin{Value}
A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
\end{Value}
%
\begin{Author}\relax
John Wambaugh and Robert Pearce
\end{Author}
%
\begin{References}\relax
Pearce, Robert G., et al. "Httk: R package for high-throughput
toxicokinetics." Journal of statistical software 79.4 (2017): 1.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}

# Multiple doses per day:
head(solve_pbtk(
  chem.name='Bisphenol-A',
  daily.dose=.5,
  days=5,
  doses.per.day=2,
  tsteps=2))

# Starting with an initial concentration:
out <- solve_pbtk(
  chem.name='bisphenola',
  dose=0,
  output.units="mg/L", 
  initial.values=c(Agut=200))

# Working with parameters (rather than having solve_pbtk retrieve them):
params <- parameterize_pbtk(chem.cas="80-05-7")
head(solve_pbtk(parameters=params))
                  
# We can change the parameters given to us by parameterize_pbtk:
params <- parameterize_pbtk(dtxsid="DTXSID4020406", species = "rat")
params["Funbound.plasma"] <- 0.1
out <- solve_pbtk(parameters=params)


# A fifty day simulation:
out <- solve_pbtk(
  chem.name = "Bisphenol A", 
  days = 50, 
  daily.dose=1,
  doses.per.day = 3)
plot.data <- as.data.frame(out)
css <- calc_analytic_css(chem.name = "Bisphenol A")

library("ggplot2")
c.vs.t <- ggplot(plot.data, aes(time, Cplasma)) + 
  geom_line() +
  geom_hline(yintercept = css) + 
  ylab("Plasma Concentration (uM)") +
  xlab("Day") + 
  theme(
    axis.text = element_text(size = 16), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 17)) +
  ggtitle("Bisphenol A")
print(c.vs.t)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{spleen\_mass\_children}{Predict spleen mass for children}{spleen.Rul.mass.Rul.children}
\keyword{httk-pop}{spleen\_mass\_children}
%
\begin{Description}\relax
For individuals under 18, predict the spleen mass from height, weight, and
gender, using equations from Ogiu et al. (1997)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
spleen_mass_children(height, weight, gender)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{height}] Vector of heights in cm.

\item[\code{weight}] Vector of weights in kg.

\item[\code{gender}] Vector of genders (either 'Male' or 'Female').
\end{ldescription}
\end{Arguments}
%
\begin{Value}
A vector of spleen masses in kg.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ogiu, Nobuko, et al. "A statistical analysis of the internal 
organ weights of normal Japanese people." Health physics 72.3 (1997): 368-383.

Price, Paul S., et al. "Modeling interindividual variation in physiological 
factors used in PBPK models of humans." Critical reviews in toxicology 33.5 
(2003): 469-503.

Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability."
Environment International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{spline\_heightweight}{Smoothing splines for log height vs. age and log body weight vs. age, along with 2-D KDE residuals, by race and gender.}{spline.Rul.heightweight}
\keyword{data}{spline\_heightweight}
\keyword{httk-pop}{spline\_heightweight}
%
\begin{Description}\relax
\#'Smoothing splines and KDE fits to joint distribution of height and weight
residuals pre-calculated from NHANES height, weight, and age data by
race/ethnicity and gender.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
spline_heightweight
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table with 6 variables: \begin{description}
 \item[\code{g}] Gender: Male
or Female\item[\code{r}] Race/ethnicity: Mexican American, Other Hispanic,
Non-Hispanic White, Non-Hispanic Black, Other\item[\code{height\_spline}] A
list of smooth.spline objects, each giving a smoothed relationship between
log height in cm and age in months\item[\code{weight\_spline}] A list of
smooth.spline objects, each giving a smoothed relationship between log body
weight in kg and age in months\item[\code{hw\_kde}] A list of kde objects;
each is a 2-D KDE of the distribution of log height and log body weight
residuals about the smoothing splines.
\end{description}

\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{spline\_hematocrit}{Smoothing splines for log hematocrit vs. age in months, and KDE residuals, by race and gender.}{spline.Rul.hematocrit}
\keyword{data}{spline\_hematocrit}
\keyword{httk-pop}{spline\_hematocrit}
%
\begin{Description}\relax
Smoothing splines and KDE residuals pre-calculated from NHANES hematocrit and
age data by race/ethnicity and gender.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
spline_hematocrit
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table with 6 variables: \begin{description}
 \item[\code{gender}] Gender:
Male or Female\item[\code{reth}] Race/ethnicity: Mexican American, Other
Hispanic, Non-Hispanic White, Non-Hispanic Black, Other
\item[\code{hct\_spline}] A list of smooth.spline objects, each giving a
smoothed relationship between log hematocrit and age in months
\item[\code{hct\_kde}] A list of kde objects; each is a KDE of the
distribution of residuals about the smoothing spline.
\end{description}

\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{spline\_serumcreat}{Smoothing splines for log serum creatinine vs. age in months, along with KDE residuals, by race and gender.}{spline.Rul.serumcreat}
\keyword{data}{spline\_serumcreat}
\keyword{httk-pop}{spline\_serumcreat}
%
\begin{Description}\relax
\#'Smoothing splines and KDE residuals pre-calculated from NHANES serum creatinine and
age data by race/ethnicity and gender.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
spline_serumcreat
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table with 6 variables: \begin{description}

\item[\code{gender}] Gender:
Male or Female
\item[\code{reth}] Race/ethnicity: Mexican American, Other
Hispanic, Non-Hispanic White, Non-Hispanic Black, Other
\item[\code{sc\_spline}] A list of smooth.spline objects, each giving a
smoothed relationship between log serum creatinine and age in months
\item[\code{sc\_kde}] A list of kde
objects; each is a KDE of the distribution of residuals about the smoothing
spline.

\end{description}

\end{Format}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{supptab1\_Linakis2020}{Supplementary output from Linakis 2020 vignette analysis.}{supptab1.Rul.Linakis2020}
\keyword{data}{supptab1\_Linakis2020}
%
\begin{Description}\relax
Supplementary output from Linakis 2020 vignette analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
supptab1_Linakis2020
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing x rows and y columns.
\end{Format}
%
\begin{Author}\relax
Matt Linakis
\end{Author}
%
\begin{Source}\relax
Matt Linakis
\end{Source}
%
\begin{References}\relax
DSStox database (https:// www.epa.gov/ncct/dsstox
\end{References}
\inputencoding{utf8}
\HeaderA{supptab2\_Linakis2020}{More supplementary output from Linakis 2020 vignette analysis.}{supptab2.Rul.Linakis2020}
\keyword{data}{supptab2\_Linakis2020}
%
\begin{Description}\relax
More supplementary output from Linakis 2020 vignette analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
supptab2_Linakis2020
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing x rows and y columns.
\end{Format}
%
\begin{Author}\relax
Matt Linakis
\end{Author}
%
\begin{Source}\relax
Matt Linakis
\end{Source}
%
\begin{References}\relax
DSStox database (https:// www.epa.gov/ncct/dsstox
\end{References}
\inputencoding{utf8}
\HeaderA{Tables.Rdata.stamp}{A timestamp of table creation}{Tables.Rdata.stamp}
\keyword{datasets}{Tables.Rdata.stamp}
%
\begin{Description}\relax
The Tables.RData file is separately created as part of building a new
release of HTTK. This time stamp indicates the script used to build the file
and when it was run.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Tables.Rdata.stamp
\end{verbatim}
\end{Usage}
%
\begin{Format}
An object of class \code{character} of length 1.
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
\inputencoding{utf8}
\HeaderA{tissue.data}{Tissue composition and species-specific physiology parameters}{tissue.data}
\keyword{data}{tissue.data}
%
\begin{Description}\relax
This data set contains values from Schmitt (2008) and Ruark et al. (2014)
describing the composition of specific tissues and from Birnbaum et al.
(1994) describing volumes of and blood flows to those tissues, allowing
parameterization of toxicokinetic models for human, mouse, rat, dog, or
rabbit. Tissue volumes were calculated by converting the fractional mass of
each tissue with its density (both from ICRP), lumping the remaining tissues
into the rest-of-body, excluding the mass of the gastrointestinal contents
\end{Description}
%
\begin{Usage}
\begin{verbatim}
tissue.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 13 rows and 20 columns.
\end{Format}
%
\begin{Author}\relax
John Wambaugh, Robert Pearce, and Nisha Sipes
\end{Author}
%
\begin{Source}\relax
Pearce et al. (2017), in preparation,

Wambaugh, John F., et al. "Toxicokinetic triage for environmental
chemicals." Toxicological Sciences (2015): 228-237.
\end{Source}
%
\begin{References}\relax
Birnbaum, L and Brown, R and Bischoff, K and Foran, J and
Blancato, J and Clewell, H and Dedrick, R (1994). Physiological parameter
values for PBPK model. International Life Sciences Institute, Risk Science
Institute, Washington, DC

Ruark, Christopher D., et al. "Predicting passive and active tissue: plasma
partition coefficients: Interindividual and interspecies variability."
Journal of pharmaceutical sciences 103.7 (2014): 2189-2198.

Schmitt, W. (2008). General approach for the calculation of tissue to plasma
partition coefficients. Toxicology in vitro : an international journal
published in association with BIBRA 22(2), 457-67,
10.1016/j.tiv.2007.09.010.

ICRP. Report of the Task Group on Reference Man. ICRP Publication 23 1975
\end{References}
\inputencoding{utf8}
\HeaderA{tissue\_masses\_flows}{Given a data.table describing a virtual population by the NHANES quantities,  generates HTTK physiological parameters for each individual.}{tissue.Rul.masses.Rul.flows}
\keyword{httk-pop}{tissue\_masses\_flows}
%
\begin{Description}\relax
Given a data.table describing a virtual population by the NHANES quantities, 
generates HTTK physiological parameters for each individual.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
tissue_masses_flows(tmf_dt)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{tmf\_dt}] A data.table generated by
\code{gen\_age\_height\_weight()}, containing variables \code{gender},
\code{reth}, \code{age\_months}, \code{age\_years}, \code{weight}, and
\code{height}.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The same data.table, with aditional variables describing tissue masses
and flows.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Barter, Zoe E., et al. "Scaling factors for the extrapolation of in vivo 
metabolic drug clearance from in vitro data: reaching a consensus on values 
of human micro-somal protein and hepatocellularity per gram of liver." Current 
Drug Metabolism 8.1 (2007): 33-45.

Birnbaum, L., et al. "Physiological parameter values for PBPK models." 
International Life Sciences Institute, Risk Science Institute, Washington, 
DC (1994).

Geigy Pharmaceuticals, "Scientific Tables", 7th Edition, 
John Wiley and Sons (1970)

McNally, Kevin, et al. "PopGen: a virtual human population generator." 
Toxicology 315 (2014): 70-85.

Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{tissue\_scale}{Allometric scaling.}{tissue.Rul.scale}
\keyword{httk-pop}{tissue\_scale}
%
\begin{Description}\relax
Allometrically scale a tissue mass or flow based on height\textasciicircum{}3/4.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
tissue_scale(height_ref, height_indiv, tissue_mean_ref)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{height\_ref}] Reference height in cm.

\item[\code{height\_indiv}] Individual height in cm.

\item[\code{tissue\_mean\_ref}] Reference tissue mass or flow.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Allometrically scaled tissue mass or flow, in the same units as
\code{tissue\_mean\_ref}.
\end{Value}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to 
environmental chemicals by simulating toxicokinetic variability." Environment 
International 106 (2017): 105-118
\end{References}
\inputencoding{utf8}
\HeaderA{wambaugh2019}{in vitro Toxicokinetic Data from Wambaugh et al. (2019)}{wambaugh2019}
\keyword{data}{wambaugh2019}
%
\begin{Description}\relax
These data are the new HTTK in vitro data for chemicals reported in Wambaugh
et al. (2019) They
are the processed values used to make the figures in that manuscript.
These data summarize the results of Bayesian analysis of the in vitro
toxicokinetic experiments conducted by Cyprotex to characterize fraction
unbound in the presence of pooled human plasma protein and the intrnsic
hepatic clearance of the chemical by pooled human hepatocytes.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
wambaugh2019
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data frame with 496 rows and 17 variables:
\begin{description}

\item[Compound] The name of the chemical
\item[CAS] The Chemical Abstracts Service Registry Number
\item[Human.Clint] Median of Bayesian credible interval for intrinsic
hepatic clearance (uL/min/million hepatocytes)]
\item[Human.Clint.pValue] Probability that there is no clearance
\item[Human.Funbound.plasma] Median of Bayesian credibl interval for
fraction of chemical free in the presence of plasma
\item[pKa\_Accept] pH(s) at which hydrogen acceptor sites (if any) are at
equilibrium
\item[pKa\_Donor] pH(s) at which hydrogne donor sites (if any) are at
equilibrium
\item[DSSTox\_Substance\_Id] Identifier for CompTox Chemical Dashboard
\item[SMILES] Simplified Molecular-Input Line-Entry System structure
description
\item[Human.Clint.Low95] Lower 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance (uL/min/million hepatocytes)
\item[Human.Clint.High95] Uppper 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance (uL/min/million hepatocytes)
\item[Human.Clint.Point] Point estimate of intrinsic hepatic clearance
(uL/min/million hepatocytes)
\item[Human.Funbound.plasma.Low95] Lower 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma
\item[Human.Funbound.plasma.High95] Upper 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma
\item[Human.Funbound.plasma.Point] Point estimate of the fraction of
chemical free in the presence of plasma
\item[MW] Molecular weight (Daltons)
\item[logP] log base ten of octanol:water partiion coefficient

\end{description}

\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. (2019)
\end{Source}
%
\begin{References}\relax
Wambaugh et al. (2019) "Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization", Toxicological Sciences, 172(2), 235-251.
\end{References}
\inputencoding{utf8}
\HeaderA{wambaugh2019.nhanes}{NHANES Chemical Intake Rates for chemicals in Wambaugh et al. (2019)}{wambaugh2019.nhanes}
\keyword{data}{wambaugh2019.nhanes}
%
\begin{Description}\relax
These data are a subset of the Bayesian inferrences reported by Ring et al.
(2017) from the U.S. Centers for Disease Control and Prevention (CDC)
National Health and Nutrition Examination Survey (NHANES). They reflect the
populaton median intake rate (mg/kg body weight/day), with uncertainty.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
wambaugh2019.nhanes
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data frame with 20 rows and 4 variables:
\begin{description}

\item[lP] The median of the Bayesian credible interval for median population
intake rate (mg/kg bodyweight/day)
\item[lP.min] The lower 95th percentile of the Bayesian credible interval for median population
intake rate (mg/kg bodyweight/day)
\item[lP.max] The upper 95th percentile of the Bayesian credible interval for median population
intake rate (mg/kg bodyweight/day)
\item[CASRN] The Chemical Abstracts Service Registry Number

\end{description}

\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. (2019)
\end{Source}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
evironmental chemicals by simulating toxicokinetic variability." Environment
international 106 (2017): 105-118

Wambaugh et al. (2019) "Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization", Toxicological Sciences, 172(2), 235-251.
\end{References}
\inputencoding{utf8}
\HeaderA{wambaugh2019.raw}{Raw Bayesian in vitro Toxicokinetic Data Analysis from Wambaugh et al. (2019)}{wambaugh2019.raw}
\keyword{data}{wambaugh2019.raw}
%
\begin{Description}\relax
These data are the new HTTK in vitro data for chemicals reported in Wambaugh
et al. (2019) They
are the output of different Bayesian models evaluated to compare using a
single protein concentration vs. the new three concentration titration
protocol. These data summarize the results of Bayesian analysis of the in vitro
toxicokinetic experiments conducted by Cyprotex to characterize fraction
unbound in the presence of pooled human plasma protein and the intrnsic
hepatic clearance of the chemical by pooled human hepatocytes.
This file includes replicates (diferent CompoundName id's but same chemical')
\end{Description}
%
\begin{Usage}
\begin{verbatim}
wambaugh2019.raw
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data frame with 530 rows and 28 variables:
\begin{description}

\item[DTXSID] Identifier for CompTox Chemical Dashboard
\item[Name] The name of the chemical
\item[CAS] The Chemical Abstracts Service Registry Number
\item[CompoundName] Sample name provided by EPA to Cyprotex
\item[Fup.point] Point estimate of the fraction of
chemical free in the presence of plasma
\item[Base.Fup.Med] Median of Bayesian credible interval for
fraction of chemical free in the presence of plasma for analysis of 100
physiological plasma protein data only (base model)
\item[Base.Fup.Low] Lower 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of 100
physiological plasma protein data only (base model)
\item[Base.Fup.High] Upper 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of 100
physiological plasma protein data only (base model)
\item[Affinity.Fup.Med] Median of Bayesian credible interval for
fraction of chemical free in the presence of plasma for analysis of protein
titration protocol data (affinity model)
\item[Affinity.Fup.Low] Lower 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of protein
titration protocol data (affinity model)
\item[Affinity.Fup.High] Upper 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of protein
titration protocol data (affinity model)
\item[Affinity.Kd.Med] Median of Bayesian credible interval for
protein binding affinity from analysis of protein
titration protocol data (affinity model)
\item[Affinity.Kd.Low] Lower 95th percentile of Bayesian credible
interval for protein binding affinity from analysis of protein
titration protocol data (affinity model)
\item[Affinity.Kd.High] Upper 95th percentile of Bayesian credible
interval for protein binding affinity from analysis of protein
titration protocol data (affinity model)
\item[Decreases.Prob] Probability that the chemical concentration decreased
systematiclally during hepatic clearance assay.
\item[Saturates.Prob] Probability that the rate of chemical concentration
decrease varied between the 1 and 10 uM hepatic clearance experiments.
\item[Slope.1uM.Median] Estimated slope for chemcial concentration decrease
in the 1 uM hepatic clearance assay.
\item[Slope.10uM.Median] Estimated slope for chemcial concentration decrease
in the 10 uM hepatic clearance assay.
\item[CLint.1uM.Median] Median of Bayesian credible interval for intrinsic
hepatic clearance at 1 uM initital chemical concentration (uL/min/million hepatocytes)]
\item[CLint.1uM.Low95th] Lower 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 1 uM initital chemical
concentration (uL/min/million hepatocytes)
\item[CLint.1uM.High95th] Uppper 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 1 uM initital chemical
concentration(uL/min/million hepatocytes)
\item[CLint.10uM.Median] Median of Bayesian credible interval for intrinsic
hepatic clearance at 10 uM initital chemical concentration (uL/min/million hepatocytes)]
\item[CLint.10uM.Low95th] Lower 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 10 uM initital chemical
concentration (uL/min/million hepatocytes)
\item[CLint.10uM.High95th] Uppper 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 10 uM initital chemical
concentration(uL/min/million hepatocytes)
\item[CLint.1uM.Point] Point estimate of intrinsic hepatic clearance
(uL/min/million hepatocytes) for 1 uM initial chemical concentration
\item[CLint.10uM.Point] Point estimate of intrinsic hepatic clearance
(uL/min/million hepatocytes) for 10 uM initial chemical concentration
\item[Fit] Classification of clearance observed
\item[SMILES] Simplified Molecular-Input Line-Entry System structure
description

\end{description}

\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. (2019)
\end{Source}
%
\begin{References}\relax
Wambaugh et al. (2019) "Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization", Toxicological Sciences, 172(2), 235-251.
\end{References}
\inputencoding{utf8}
\HeaderA{wambaugh2019.seem3}{ExpoCast SEEM3 Consensus Exposure Model Predictions for Chemical Intake Rates}{wambaugh2019.seem3}
\keyword{data}{wambaugh2019.seem3}
%
\begin{Description}\relax
These data are a subset of the Bayesian inferrences reported by Ring et al.
(2019) for a consensus model of twelve exposue predictors. The predictors
were calibrated based upon their ability to predict intake rates inferred
National Health and Nutrition Examination Survey (NHANES). They reflect the
populaton median intake rate (mg/kg body weight/day), with uncertainty.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
wambaugh2019.seem3
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data frame with 385 rows and 38 variables:
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
Wambaugh et al. (2019)
\end{Source}
%
\begin{References}\relax
Ring, Caroline L., et al. "Consensus modeling of median chemical
intake for the US population based on predictions of exposure pathways."
Environmental science \& technology 53.2 (2018): 719-732.

Wambaugh et al. (2019) "Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization", Toxicological Sciences, 172(2), 235-251.
\end{References}
\inputencoding{utf8}
\HeaderA{wambaugh2019.tox21}{Tox21 2015 Active Hit Calls (EPA)}{wambaugh2019.tox21}
\keyword{data}{wambaugh2019.tox21}
%
\begin{Description}\relax
The ToxCast and Tox21 research programs employ batteries of high throughput
assays to assess chemical bioactivity in vitro. Not every chemical is tested
through every assay. Most assays are conducted in concentration response,
and each corresponding assay endpoint is analyzed statistically to determine
if there is a concentration-dependent response or "hit" using the ToxCast
Pipeline.  Most assay endpoint-chemical combinations are non-responsive.
Here, only the hits are treated as potential indicators of bioactivity. This
bioactivity does not have a direct toxicological interpretation. The October
2015 release (invitrodb\_v2) of the ToxCast and Tox21 data were used for this
analysis. This object contains just the chemicals in Wambaugh et al. (2019)
and only the quantiles across all assays for the ACC.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
wambaugh2019.tox21
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table with 401 rows and 6 columns
\end{Format}
%
\begin{Author}\relax
John Wambaugh
\end{Author}
%
\begin{Source}\relax
\url{ftp://newftp.epa.gov/COMPTOX/High_Throughput_Screening_Data/Previous_Data/ToxCast_Data_Release_Oct_2015/}
\end{Source}
%
\begin{References}\relax
Kavlock, Robert, et al. "Update on EPA's ToxCast program:
providing high throughput decision support tools for chemical risk
management." Chemical research in toxicology 25.7 (2012): 1287-1302.

Tice, Raymond R., et al. "Improving the human hazard characterization of
chemicals: a Tox21 update." Environmental health perspectives 121.7 (2013):
756-765.

Richard, Ann M., et al. "ToxCast chemical landscape: paving the road to 21st
century toxicology." Chemical research in toxicology 29.8 (2016): 1225-1251.

Filer, Dayne L., et al. "tcpl: the ToxCast pipeline for high-throughput
screening data." Bioinformatics 33.4 (2016): 618-620.

Wambaugh, John F., et al. "Assessing Toxicokinetic Uncertainty and 
Variability in Risk Prioritization." Toxicological Sciences 172.2 (2019): 
235-251.
\end{References}
\inputencoding{utf8}
\HeaderA{well\_param}{Microtiter Plate Well Descriptions for Armitage et al. (2014) Model}{well.Rul.param}
\keyword{data}{well\_param}
%
\begin{Description}\relax
Microtiter Plate Well Descriptions for Armitage et al. (2014) model from
Honda et al. (2019)
\end{Description}
%
\begin{Usage}
\begin{verbatim}
well_param
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data frame / data table with 11 rows and 8 variables:
\begin{description}

\item[sysID] Identifier for each multi-well plate system
\item[well\_desc] Well description
\item[well\_number] Number of wells on plate
\item[area\_bottom] Area of well bottom in mm\textasciicircum{}2
\item[cell\_yield] Number of cells
\item[diam] Diameter of well in mm
\item[v\_total] Total volume of well in uL)
\item[v\_working] Working volume of well in uL

\end{description}

\end{Format}
%
\begin{Author}\relax
Greg Honda
\end{Author}
%
\begin{Source}\relax
\url{https://www.corning.com/catalog/cls/documents/application-notes/CLS-AN-209.pdf}
\end{Source}
%
\begin{References}\relax
Armitage, J. M.; Wania, F.; Arnot, J. A. Environ. Sci. Technol.
2014, 48, 9770-9779. dx.doi.org/10.1021/es501955g

Honda, Gregory S., et al. "Using the Concordance of In Vitro and
In Vivo Data to Evaluate Extrapolation Assumptions", PloS ONE 14.5 (2019): e0217564.
\end{References}
\inputencoding{utf8}
\HeaderA{Wetmore.data}{Published toxicokinetic predictions based on in vitro data}{Wetmore.data}
\keyword{data}{Wetmore.data}
%
\begin{Description}\relax
This data set gives the chemical specific predictions for serum
concentration at steady state resulting from constant infusion exposure, as
published in a series of papers from Barbara Wetmore's group at the Hamner
Institutes for Life Sciences. Predictions include the median and 90\%
interval in uM and mg/L. Calculations were made using the 1 and 10 uM in
vitro measured clearances.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Wetmore.data
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 577 rows and 20 columns.
\end{Format}
%
\begin{Source}\relax
Wambaugh, John F., et al. "Toxicokinetic triage for environmental
chemicals." Toxicological Sciences (2015): 228-237.
\end{Source}
%
\begin{References}\relax
Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., "Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment," Toxicological Sciences 125
157-174 (2012)

Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., "The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays"
Toxicological Sciences, 132:327-346 (2013).

Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
"Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing" Toxicological
Sciences, kfv171.
\end{References}
\inputencoding{utf8}
\HeaderA{Wetmore2012}{Published toxicokinetic predictions based on in vitro data from Wetmore et al. 2012.}{Wetmore2012}
\keyword{data}{Wetmore2012}
%
\begin{Description}\relax
This data set overlaps with Wetmore.data and is used only in Vignette 4 for
steady state concentration.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Wetmore2012
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.frame containing 13 rows and 15 columns.
\end{Format}
%
\begin{References}\relax
Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., "Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment," Toxicological Sciences 125
157-174 (2012)
\end{References}
\inputencoding{utf8}
\HeaderA{wfl}{WHO weight-for-length charts}{wfl}
\keyword{data}{wfl}
\keyword{httk-pop}{wfl}
%
\begin{Description}\relax
Charts giving weight-for-length percentiles for boys and girls under age 2.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
wfl
\end{verbatim}
\end{Usage}
%
\begin{Format}
A data.table object with variables \begin{description}
 \item[\code{Sex}] 'Male'
or 'Female'\item[\code{Length}] length in cm\item[\code{L}, \code{M},
\code{S}] LMS parameters; see
\url{https://www.cdc.gov/growthcharts/percentile_data_files.htm}\item[\code{P2.3},
\code{P5}, \code{P10}, \code{P25}, \code{P50}, \code{P75}, \code{P90},
\code{P95}, and \code{P97.7}] weight percentiles
\end{description}

\end{Format}
%
\begin{Details}\relax
For infants under age 2, weight class depends on weight for length percentile.
\#'\begin{description}
 \item[Underweight] <2.3rd percentile\item[Normal
weight] 2.3rd-97.7th percentile\item[Obese] >=97.7th percentile
\end{description}

\end{Details}
%
\begin{Author}\relax
Caroline Ring
\end{Author}
%
\begin{Source}\relax
\url{https://www.cdc.gov/growthcharts/who/girls_weight_head_circumference.htm}
and
\url{https://www.cdc.gov/growthcharts/who/boys_weight_head_circumference.htm}
\end{Source}
%
\begin{References}\relax
Ring, Caroline L., et al. "Identifying populations sensitive to
environmental chemicals by simulating toxicokinetic variability." Environment
International 106 (2017): 105-118
\end{References}
\printindex{}
\end{document}
